// @primer/components/dist/browser.umd.js
!function(e,t){"object"==typeof exports&&"undefined"!=typeof module?t(exports,require("react"),require("styled-components"),require("react-dom")):"function"==typeof define&&define.amd?define(["exports","react","styled-components","react-dom"],t):t((e="undefined"!=typeof globalThis?globalThis:e||self).primer={},e.React,e.styled,e.ReactDOM)}(this,(function(e,t,r,o){"use strict";function a(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}var n=a(t),i=a(r),d=a(o),l={theme:{animation:{easeOutCubic:"cubic-bezier(0.33, 1, 0.68, 1)"},borderWidths:[0,"1px"],breakpoints:["544px","768px","1012px","1280px"],fonts:{normal:'-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"',mono:'SFMono-Regular, Consolas, "Liberation Mono", Menlo, Courier, monospace'},fontSizes:["12px","14px","16px","20px","24px","32px","40px","48px"],fontWeights:{light:300,normal:400,semibold:500,bold:600},lineHeights:{condensedUltra:"1",condensed:"1.25",default:"1.5"},radii:["0","3px","6px","100px"],sizes:{small:"544px",medium:"768px",large:"1012px",xlarge:"1280px"},space:["0","4px","8px","16px","24px","32px","40px","48px","64px","80px","96px","112px","128px"],colorSchemes:{light:{colors:{auto:{black:"#1b1f23",white:"#ffffff",gray:["#fafbfc","#f6f8fa","#e1e4e8","#d1d5da","#959da5","#6a737d","#586069","#444d56","#2f363d","#24292e"],blue:["#f1f8ff","#dbedff","#c8e1ff","#79b8ff","#2188ff","#0366d6","#005cc5","#044289","#032f62","#05264c"],green:["#f0fff4","#dcffe4","#bef5cb","#85e89d","#34d058","#28a745","#22863a","#176f2c","#165c26","#144620"],yellow:["#fffdef","#fffbdd","#fff5b1","#ffea7f","#ffdf5d","#ffd33d","#f9c513","#dbab09","#b08800","#735c0f"],orange:["#fff8f2","#ffebda","#ffd1ac","#ffab70","#fb8532","#f66a0a","#e36209","#d15704","#c24e00","#a04100"],red:["#ffeef0","#ffdce0","#fdaeb7","#f97583","#ea4a5a","#d73a49","#cb2431","#b31d28","#9e1c23","#86181d"],purple:["#f5f0ff","#e6dcfd","#d1bcf9","#b392f0","#8a63d2","#6f42c1","#5a32a3","#4c2889","#3a1d6e","#29134e"],pink:["#ffeef8","#fedbf0","#f9b3dd","#f692ce","#ec6cb9","#ea4aaa","#d03592","#b93a86","#99306f","#6d224f"]},text:{primary:"#24292e",secondary:"#586069",tertiary:"#6a737d",placeholder:"#6a737d",disabled:"#959da5",inverse:"#ffffff",link:"#0366d6",danger:"#cb2431",success:"#22863a",warning:"#b08800",white:"#ffffff"},icon:{primary:"#24292e",secondary:"#586069",tertiary:"#959da5",info:"#0366d6",danger:"#d73a49",success:"#22863a",warning:"#b08800"},border:{primary:"#e1e4e8",secondary:"#eaecef",tertiary:"#d1d5da",overlay:"#e1e4e8",inverse:"#ffffff",info:"#0366d6",danger:"#d73a49",success:"#34d058",warning:"#f9c513"},bg:{canvas:"#ffffff",canvasMobile:"#ffffff",canvasInverse:"#24292e",canvasInset:"#f6f8fa",primary:"#ffffff",secondary:"#fafbfc",tertiary:"#f6f8fa",overlay:"#ffffff",backdrop:"rgba(27,31,35,0.5)",info:"#f1f8ff",infoInverse:"#0366d6",danger:"#ffeef0",dangerInverse:"#d73a49",success:"#dcffe4",successInverse:"#28a745",warning:"#fff5b1",warningInverse:"#ffd33d"},state:{hover:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff",secondaryBg:"#f6f8fa",secondaryBorder:"#f6f8fa"},selected:{primaryBg:"#0366d6",primaryBorder:"#0366d6",primaryText:"#ffffff",primaryIcon:"#ffffff"},focus:{border:"#0366d6"}},fade:{fg10:"rgba(27,31,35,0.1)",fg15:"rgba(27,31,35,0.15)",fg30:"rgba(27,31,35,0.3)",fg50:"rgba(27,31,35,0.5)",fg70:"rgba(27,31,35,0.7)",fg85:"rgba(27,31,35,0.85)",black10:"rgba(27,31,35,0.1)",black15:"rgba(27,31,35,0.15)",black30:"rgba(27,31,35,0.3)",black50:"rgba(27,31,35,0.5)",black70:"rgba(27,31,35,0.7)",black85:"rgba(27,31,35,0.85)",white10:"rgba(255,255,255,0.1)",white15:"rgba(255,255,255,0.15)",white30:"rgba(255,255,255,0.3)",white50:"rgba(255,255,255,0.5)",white70:"rgba(255,255,255,0.7)",white85:"rgba(255,255,255,0.85)"},alert:{info:{text:"#24292e",icon:"rgba(4,66,137,0.6)",bg:"#dbedff",border:"rgba(4,66,137,0.2)"},warn:{text:"#24292e",icon:"#b08800",bg:"#fffbdd",border:"rgba(176,136,0,0.2)"},error:{text:"#24292e",icon:"rgba(158,28,35,0.6)",bg:"#ffe3e6",border:"rgba(158,28,35,0.2)"},success:{text:"#24292e",icon:"rgba(23,111,44,0.8)",bg:"#dcffe4",border:"rgba(23,111,44,0.2)"}},autocomplete:{rowBorder:"#eaecef"},blankslate:{icon:"#a3aab1"},btn:{text:"#24292e",bg:"#fafbfc",border:"rgba(27,31,35,0.15)",hoverBg:"#f3f4f6",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"#edeff2",focusBg:"#fafbfc",focusBorder:"rgba(27,31,35,0.15)",primary:{text:"#ffffff",bg:"#2ea44f",border:"rgba(27,31,35,0.15)",hoverBg:"#2c974b",hoverBorder:"rgba(27,31,35,0.15)",selectedBg:"#2a8f47",disabledText:"rgba(255,255,255,0.8)",disabledBg:"#94d3a2",disabledBorder:"rgba(27,31,35,0.1)",focusBg:"#2ea44f",focusBorder:"rgba(27,31,35,0.15)",icon:"rgba(255,255,255,0.8)",counterBg:"rgba(255,255,255,0.2)"},outline:{text:"#0366d6",hoverText:"#ffffff",hoverBg:"#0366d6",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#035fc7",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(3,102,214,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(3,102,214,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(3,102,214,0.1)"},danger:{text:"#d73a49",hoverText:"#ffffff",hoverBg:"#cb2431",hoverBorder:"rgba(27,31,35,0.15)",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#d42d3d",selectedBorder:"rgba(27,31,35,0.15)",disabledText:"rgba(215,58,73,0.5)",disabledBg:"#fafbfc",disabledCounterBg:"rgba(215,58,73,0.05)",focusBorder:"rgba(27,31,35,0.15)",counterBg:"rgba(215,58,73,0.1)",icon:"#d73a49",hoverIcon:"#ffffff"},counterBg:"rgba(27,31,35,0.08)"},counter:{text:"#24292e",bg:"rgba(209,213,218,0.5)",primary:{text:"#ffffff",bg:"#6a737d"},secondary:{text:"#6a737d"}},label:{border:"#e1e4e8",primary:{text:"#24292e",border:"#6a737d"},secondary:{text:"#586069",border:"#e1e4e8"},info:{text:"#0366d6",border:"#0366d6"},success:{text:"#22863a",border:"#28a745"},warning:{text:"#735c0f",border:"#b08800"},danger:{text:"#d73a49",border:"#cb2431"},orange:{text:"#c24e00",border:"#f66a0a"}},input:{bg:"#ffffff",contrastBg:"#fafbfc",border:"#e1e4e8",disabledBg:"#f6f8fa",disabledBorder:"#e1e4e8",warningBorder:"#f9c513",errorBorder:"#cb2431",tooltip:{success:{text:"#144620",bg:"#dcffe4",border:"#34d058"},warning:{text:"#735c0f",bg:"#fff5b1",border:"#f9c513"},error:{text:"#86181d",bg:"#ffeef0",border:"#f97583"}}},avatar:{bg:"#ffffff",border:"rgba(0,0,0,0)",stackFade:"#d1d5da",stackFadeMore:"#e1e4e8"},toast:{text:"#24292e",bg:"#ffffff",border:"#e1e4e8",icon:"#ffffff",iconBg:"#0366d6",iconBorder:"rgba(0,0,0,0)",success:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#28a745",iconBorder:"rgba(0,0,0,0)"},warning:{text:"#24292e",border:"#e1e4e8",icon:"#24292e",iconBg:"#ffd33d",iconBorder:"rgba(0,0,0,0)"},danger:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#d73a49",iconBorder:"rgba(0,0,0,0)"},loading:{text:"#24292e",border:"#e1e4e8",icon:"#ffffff",iconBg:"#586069",iconBorder:"rgba(0,0,0,0)"}},timeline:{text:"#444d56",badgeBg:"#e1e4e8",targetBadgeBorder:"#2188ff",targetBadgeShadow:"#c8e1ff"},selectMenu:{borderSecondary:"#eaecef",backdropBg:"rgba(27,31,35,0.5)",backdropBorder:"rgba(0,0,0,0)",tapHighlight:"rgba(209,213,218,0.5)",tapFocusBg:"#dbedff"},box:{blueBorder:"#c8e1ff",rowYellowBg:"#fffbdd",rowBlueBg:"#f1f8ff",headerBlueBg:"#f1f8ff",headerBlueBorder:"#c8e1ff",borderInfo:"rgba(3,102,214,0.2)",bgInfo:"#f1f8ff",borderWarning:"rgba(255,211,61,0.4)",bgWarning:"#fffdef"},branchName:{text:"#586069",icon:"#a8bbd0",bg:"#eaf5ff",link:{text:"#0366d6",icon:"#a8bbd0",bg:"#eaf5ff"}},markdown:{codeBg:"rgba(27,31,35,0.05)",frameBorder:"#dfe2e5",blockquoteBorder:"#dfe2e5",tableBorder:"#dfe2e5",tableTrBorder:"#c6cbd1"},menu:{headingText:"#24292e",borderActive:"#f9826c",bgActive:"rgba(0,0,0,0)"},sidenav:{selectedBg:"#ffffff",borderActive:"#f9826c"},header:{text:"rgba(255,255,255,0.7)",bg:"#24292e",logo:"#ffffff"},filterItem:{barBg:"#eff3f6"},hiddenTextExpander:{bg:"#dfe2e5",bgHover:"#c6cbd1"},dragAndDrop:{border:"#c3c8cf"},uploadEnabled:{border:"#dfe2e5",borderFocused:"#4a9eff"},previewableCommentForm:{border:"#c3c8cf"},underlinenav:{border:"rgba(209,213,218,0)",borderHover:"#d1d5da",borderActive:"#f9826c",text:"#24292e",textHover:"#24292e",textActive:"#24292e",icon:"#959da5",iconHover:"#959da5",iconActive:"#24292e",counterText:"#24292e"},verifiedBadge:{text:"#22863a",bg:"#ffffff",border:"#e1e4e8"},socialCount:{bg:"#ffffff"},tooltip:{text:"#ffffff",bg:"#24292e"},headerSearch:{bg:"#24292e",border:"#444d56"},searchKeyword:{hl:"#fffbdd"},diffstat:{neutralBg:"#d1d5da",neutralBorder:"#d1d5da",deletionBg:"#d73a49",deletionBorder:"#d73a49",additionBg:"#28a745",additionBorder:"#28a745"},mktg:{success:"#2ebc4f",info:"#1074e7",bgShadeGradient:{top:"rgba(27,31,35,0.065)",bottom:"rgba(27,31,35,0)"},btn:{bg:"#3076df",border:"#3076df",text:"#ffffff",icon:"#ffffff",hover:{bg:"#0366d6",border:"#0366d6"},disabled:{bg:"rgba(48,118,223,0.5)",border:"rgba(0,0,0,0)",text:"#ffffff",icon:"#ffffff"},primary:{bg:"#5eb85c",border:"#5eb85c",text:"#ffffff",icon:"#ffffff",hover:{bg:"#28a745",border:"#28a745"},disabled:{bg:"rgba(94,184,92,0.5)",border:"rgba(0,0,0,0)",text:"#ffffff",icon:"#ffffff"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(42,103,195,0.5)",text:"#3076df",icon:"#3076df",hover:{bg:"rgba(0,0,0,0)",border:"#0366d6",text:"#0366d6",icon:"#0366d6"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(42,103,195,0.2)",text:"rgba(48,118,223,0.5)",icon:"rgba(48,118,223,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(255,255,255,0.5)",text:"#ffffff",icon:"#ffffff",hover:{bg:"#ffffff",border:"#ffffff",text:"#444d56",icon:"#444d56"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(255,255,255,0.2)",text:"rgba(255,255,255,0.5)",icon:"rgba(255,255,255,0.5)"}}}},filesExplorerIcon:"#79b8ff",hlAuthorBg:"#f1f8ff",hlAuthorBorder:"#c8e1ff",logoSubdued:"#d1d5da",discussionBorder:"#a2cbac",discussionBgSuccess:"#28a745",actionsWorkflowTableStickyBg:"rgba(255,255,255,0.95)",repoLanguageColorBorder:"rgba(27,31,35,0.1)",codeSelectionBg:"#c8e1ff",highlight:{text:"#442c12",bg:"#fff0bb"},blob:{lineHighlightBg:"#fffbdd",lineHighlightBorder:"rgba(0,0,0,0)"},diff:{addition:{text:"#22863a",bg:"#e6ffed",border:"#34d058"},deletion:{text:"#cb2431",bg:"#ffeef0",border:"#d73a49"},change:{text:"#b08800",bg:"#fff5b1",border:"#f9c513"}},diffBlob:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",addition:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#cdffd8",lineBg:"#e6ffed",wordBg:"#acf2bd"},deletion:{numText:"rgba(27,31,35,0.3)",numHoverText:"rgba(27,31,35,0.6)",numBg:"#ffdce0",lineBg:"#ffeef0",wordBg:"#fdb8c0"},hunk:{text:"rgba(27,31,35,0.7)",numBg:"#dbedff",lineBg:"#f1f8ff"},emptyBlockBg:"#fafbfc",selectedLineHighlightBg:"rgba(255,223,93,0.2)",selectedLineHighlightBorder:"#ffd33d",expander:{icon:"#586069",hoverIcon:"#ffffff",hoverBg:"#0366d6"},commentButton:{icon:"#ffffff",bg:"#0366d6",gradientBg:"#0372ef"}},globalNav:{logo:"#ffffff",bg:"#24292e",text:"#ffffff",icon:"#ffffff",inputBg:"#fafbfc",inputBorder:"#fafbfc",inputIcon:"#d1d5da",inputPlaceholder:"#959da5"},calendarGraph:{dayBg:"#ebedf0",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#9be9a8",dayL2Bg:"#40c463",dayL3Bg:"#30a14e",dayL4Bg:"#216e39",dayL4Border:"rgba(27,31,35,0.06)",dayL3Border:"rgba(27,31,35,0.06)",dayL2Border:"rgba(27,31,35,0.06)",dayL1Border:"rgba(27,31,35,0.06)"},footerInvertocat:{octicon:"#d1d5da",octiconHover:"#6a737d"},prState:{draft:{text:"#ffffff",bg:"#6a737d",border:"rgba(0,0,0,0)"},open:{text:"#ffffff",bg:"#28a745",border:"rgba(0,0,0,0)"},merged:{text:"#ffffff",bg:"#6f42c1",border:"rgba(0,0,0,0)"},closed:{text:"#ffffff",bg:"#d73a49",border:"rgba(0,0,0,0)"}},topicTag:{text:"#0366d6",bg:"#f1f8ff",hoverBg:"#ddeeff",activeBg:"#e7f3ff"},mergeBox:{successIconBg:"#28a745",successIconText:"#ffffff",successIconBorder:"rgba(0,0,0,0)",successIndicatorBg:"#28a745",successIndicatorBorder:"rgba(0,0,0,0)",mergedIconBg:"#6f42c1",mergedIconText:"#ffffff",mergedIconBorder:"rgba(0,0,0,0)",mergedBoxBorder:"#6f42c1",neutralIconBg:"#6a737d",neutralIconText:"#ffffff",neutralIconBorder:"rgba(0,0,0,0)",neutralIndicatorBg:"#6a737d",neutralIndicatorBorder:"rgba(0,0,0,0)",warningIconBg:"#dbab09",warningIconText:"#ffffff",warningIconBorder:"rgba(0,0,0,0)",warningBoxBorder:"#ffd33d",warningMergeHighlight:"rgba(0,0,0,0)",errorIconBg:"#d73a49",errorIconText:"#ffffff",errorIconBorder:"rgba(0,0,0,0)",errorIndicatorBg:"#d73a49",errorIndicatorBorder:"rgba(0,0,0,0)"},project:{cardBg:"#ffffff",headerBg:"#24292e",sidebarBg:"#ffffff",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},checks:{bg:"#24292e",textPrimary:"#fafbfc",textSecondary:"#959da5",textLink:"#79b8ff",btnIcon:"#d1d5da",btnHoverIcon:"#fafbfc",btnHoverBg:"rgba(255,255,255,0.125)",inputText:"#f6f8fa",inputPlaceholderText:"#959da5",inputFocusText:"#959da5",inputBg:"#2f363d",dropdownText:"#d1d5da",dropdownBg:"#2f363d",dropdownBorder:"#444d56",dropdownHoverText:"#fafbfc",dropdownHoverBg:"#444d56",dropdownBtnHoverText:"#fafbfc",dropdownBtnHoverBg:"#2f363d",scrollbarThumbBg:"#586069",headerLabelText:"#e1e4e8",headerLabelOpenText:"#fafbfc",headerBorder:"#2f363d",headerIcon:"#959da5",lineText:"#e1e4e8",lineNumText:"rgba(149,157,165,0.75)",lineTimestampText:"#959da5",lineHoverBg:"#2f363d",lineSelectedBg:"rgba(33,136,255,0.15)",lineSelectedNumText:"#79b8ff",lineDtFmText:"#24292e",lineDtFmBg:"#ffd33d",gateBg:"rgba(249,197,19,0.15)",gateText:"#e1e4e8",gateWaitingText:"#d1d5da",stepHeaderOpenBg:"#2f363d",stepErrorText:"#f97583",stepWarningText:"#ffea7f",loglineText:"#959da5",loglineNumText:"rgba(149,157,165,0.75)",loglineDebugText:"#b392f0",loglineErrorText:"#e1e4e8",loglineErrorNumText:"#f97583",loglineErrorBg:"rgba(203,36,49,0.15)",loglineWarningText:"#e1e4e8",loglineWarningNumText:"#ffea7f",loglineWarningBg:"rgba(249,197,19,0.15)",loglineCommandText:"#79b8ff",loglineSectionText:"#85e89d"},introShelf:{gradientLeft:"#f1f8ff",gradientRight:"#dcffe4",gradientIn:"#ffffff",gradientOut:"rgba(255,255,255,0)"},marketingIcon:{primary:"#2188ff",secondary:"#79b8ff"},prettylights:{syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",storageModifierImport:"#24292e",entityTag:"#22863a",keyword:"#d73a49",string:"#032f62",variable:"#e36209",brackethighlighterUnmatched:"#b31d28",invalidIllegalText:"#fafbfc",invalidIllegalBg:"#b31d28",carriageReturnText:"#fafbfc",carriageReturnBg:"#d73a49",stringRegexp:"#22863a",markupList:"#735c0f",markupHeading:"#005cc5",markupItalic:"#24292e",markupBold:"#24292e",markupDeletedText:"#b31d28",markupDeletedBg:"#ffeef0",markupInsertedText:"#22863a",markupInsertedBg:"#f0fff4",markupChangedText:"#e36209",markupChangedBg:"#ffebda",markupIgnoredText:"#f6f8fa",markupIgnoredBg:"#005cc5",metaDiffRange:"#6f42c1",brackethighlighterAngle:"#586069",sublimelinterGutterMark:"#959da5",constantOtherReferenceLink:"#032f62"}},codemirror:{text:"#24292e",bg:"#ffffff",guttersBg:"#ffffff",guttermarkerText:"#ffffff",guttermarkerSubtleText:"#d1d5da",linenumberText:"#959da5",cursor:"#24292e",selectionBg:"#c8e1ff",activelineBg:"#fafbfc",matchingbracketText:"#24292e",linesBg:"#ffffff",syntax:{comment:"#6a737d",constant:"#005cc5",entity:"#6f42c1",keyword:"#d73a49",storage:"#d73a49",string:"#032f62",support:"#005cc5",variable:"#e36209"}},ansi:{black:"#24292e",blackBright:"#2f363d",white:"#e1e4e8",whiteBright:"#e1e4e8",gray:"#959da5",red:"#f97583",redBright:"#fdaeb7",green:"#85e89d",greenBright:"#bef5cb",yellow:"#ffea7f",yellowBright:"#fff5b1",blue:"#79b8ff",blueBright:"#c8e1ff",magenta:"#b392f0",magentaBright:"#d1bcf9",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 1px 0 rgba(27,31,35,0.04)",medium:"0 3px 6px rgba(149,157,165,0.15)",large:"0 8px 24px rgba(149,157,165,0.2)",extraLarge:"0 12px 48px rgba(149,157,165,0.3)",highlight:"inset 0 1px 0 rgba(255,255,255,0.25)",inset:"inset 0 1px 0 rgba(225,228,232,0.2)"},state:{focus:{shadow:"0 0 0 3px rgba(3,102,214,0.3)"}},autocomplete:{shadow:"0 3px 6px rgba(149,157,165,0.15)"},btn:{shadow:"0 1px 0 rgba(27,31,35,0.04)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.25)",focusShadow:"0 0 0 3px rgba(3,102,214,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(27,31,35,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(3,102,214,0.3)",primary:{shadow:"0 1px 0 rgba(27,31,35,0.1)",insetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(20,70,32,0.2)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(5,38,76,0.2)",focusShadow:"0 0 0 3px rgba(0,92,197,0.4)"},danger:{hoverShadow:"0 1px 0 rgba(27,31,35,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(255,255,255,0.03)",selectedShadow:"inset 0 1px 0 rgba(134,24,29,0.2)",focusShadow:"0 0 0 3px rgba(203,36,49,0.4)"}},dropdown:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},input:{shadow:"inset 0 1px 2px rgba(27,31,35,0.075)"},avatar:{childShadow:"-2px -2px 0 rgba(255,255,255,0.8)"},toast:{shadow:"0 8px 24px rgba(149,157,165,0.2)"},selectMenu:{shadow:"0 0 18px rgba(27,31,35,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(48,118,223,0.3)",primary:{focusShadow:"0 0 0 3px rgba(94,184,92,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(42,103,195,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(255,255,255,0.3)"}}}}},dark:{colors:{auto:{black:"#f0f6fc",white:"#010409",gray:["#0d1117","#161b22","#21262d","#30363d","#484f58","#6e7681","#8b949e","#b1bac4","#c9d1d9","#f0f6fc"],blue:["#051d4d","#0c2d6b","#0d419d","#1158c7","#1f6feb","#388bfd","#58a6ff","#79c0ff","#a5d6ff","#cae8ff"],green:["#04260f","#033a16","#0f5323","#196c2e","#238636","#2ea043","#3fb950","#56d364","#7ee787","#aff5b4"],yellow:["#341a00","#4b2900","#693e00","#845306","#9e6a03","#bb8009","#d29922","#e3b341","#f2cc60","#f8e3a1"],orange:["#3d1300","#5a1e02","#762d0a","#9b4215","#bd561d","#db6d28","#f0883e","#ffa657","#ffc680","#ffdfb6"],red:["#490202","#67060c","#8e1519","#b62324","#da3633","#f85149","#ff7b72","#ffa198","#ffc1ba","#ffdcd7"],purple:["#271052","#3c1e70","#553098","#6e40c9","#8957e5","#a371f7","#bc8cff","#d2a8ff","#e2c5ff","#eddeff"],pink:["#42062a","#5e103e","#7d2457","#9e3670","#bf4b8a","#db61a2","#f778ba","#ff9bce","#ffbedd","#ffdaec"]},text:{primary:"#c9d1d9",secondary:"#8b949e",tertiary:"#8b949e",placeholder:"#484f58",disabled:"#484f58",inverse:"#0d1117",link:"#58a6ff",danger:"#f85149",success:"#56d364",warning:"#e3b341",white:"#f0f6fc"},icon:{primary:"#c9d1d9",secondary:"#6e7681",tertiary:"#484f58",info:"#79c0ff",danger:"#f85149",success:"#56d364",warning:"#e3b341"},border:{primary:"#30363d",secondary:"#21262d",tertiary:"#6e7681",overlay:"#30363d",inverse:"#f0f6fc",info:"rgba(56,139,253,0.4)",danger:"rgba(248,81,73,0.4)",success:"rgba(63,185,80,0.4)",warning:"rgba(187,128,9,0.4)"},bg:{canvas:"#0d1117",canvasMobile:"#010409",canvasInverse:"#f0f6fc",canvasInset:"#090c10",primary:"#0d1117",secondary:"#0d1117",tertiary:"#161b22",overlay:"#21262d",backdrop:"rgba(1,4,9,0.8)",info:"rgba(56,139,253,0.1)",infoInverse:"#388bfd",danger:"rgba(248,81,73,0.1)",dangerInverse:"#da3633",success:"rgba(46,160,67,0.1)",successInverse:"#2ea043",warning:"rgba(187,128,9,0.1)",warningInverse:"#bb8009"},state:{hover:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc",secondaryBg:"#161b22",secondaryBorder:"#161b22"},selected:{primaryBg:"#1f6feb",primaryBorder:"#388bfd",primaryText:"#f0f6fc",primaryIcon:"#f0f6fc"},focus:{border:"#388bfd"}},fade:{fg10:"rgba(240,246,252,0.1)",fg15:"rgba(240,246,252,0.15)",fg30:"rgba(240,246,252,0.3)",fg50:"rgba(240,246,252,0.5)",fg70:"rgba(240,246,252,0.7)",fg85:"rgba(240,246,252,0.85)",black10:"rgba(1,4,9,0.1)",black15:"rgba(1,4,9,0.15)",black30:"rgba(1,4,9,0.3)",black50:"rgba(1,4,9,0.5)",black70:"rgba(1,4,9,0.7)",black85:"rgba(1,4,9,0.85)",white10:"rgba(240,246,252,0.1)",white15:"rgba(240,246,252,0.15)",white30:"rgba(240,246,252,0.3)",white50:"rgba(240,246,252,0.5)",white70:"rgba(240,246,252,0.7)",white85:"rgba(240,246,252,0.85)"},alert:{info:{text:"#79c0ff",icon:"#79c0ff",bg:"rgba(56,139,253,0.1)",border:"rgba(56,139,253,0.4)"},warn:{text:"#e3b341",icon:"#e3b341",bg:"rgba(187,128,9,0.1)",border:"rgba(187,128,9,0.4)"},error:{text:"#ff7b72",icon:"#ff7b72",bg:"rgba(248,81,73,0.1)",border:"rgba(248,81,73,0.4)"},success:{text:"#56d364",icon:"#56d364",bg:"rgba(46,160,67,0.1)",border:"rgba(46,160,67,0.4)"}},autocomplete:{rowBorder:"#30363d"},blankslate:{icon:"#535c66"},btn:{text:"#c9d1d9",bg:"#21262d",border:"#30363d",hoverBg:"#30363d",hoverBorder:"#8b949e",selectedBg:"#161b22",focusBg:"#21262d",focusBorder:"#8b949e",primary:{text:"#ffffff",bg:"#238636",border:"#2ea043",hoverBg:"#2ea043",hoverBorder:"#3fb950",selectedBg:"#238636",disabledText:"rgba(240,246,252,0.5)",disabledBg:"rgba(35,134,54,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#238636",focusBorder:"#3fb950",icon:"#f0f6fc",counterBg:"rgba(240,246,252,0.2)"},outline:{text:"#58a6ff",hoverText:"#58a6ff",hoverBg:"#30363d",hoverBorder:"#58a6ff",hoverCounterBg:"rgba(240,246,252,0.2)",selectedText:"#f0f6fc",selectedBg:"#0d419d",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(88,166,255,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(31,111,235,0.05)",focusBorder:"#58a6ff",counterBg:"rgba(31,111,235,0.1)"},danger:{text:"#f85149",hoverText:"#ffffff",hoverBg:"#da3633",hoverBorder:"#f85149",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#b62324",selectedBorder:"rgba(240,246,252,0.1)",disabledText:"rgba(248,81,73,0.5)",disabledBg:"#0d1117",disabledCounterBg:"rgba(218,54,51,0.05)",focusBorder:"#f85149",counterBg:"rgba(218,54,51,0.1)",icon:"#f85149",hoverIcon:"#f0f6fc"},counterBg:"#30363d"},counter:{text:"#c9d1d9",bg:"#30363d",primary:{text:"#c9d1d9",bg:"#6e7681"},secondary:{text:"#8b949e"}},label:{border:"#30363d",primary:{text:"#b1bac4",border:"#6e7681"},secondary:{text:"#8b949e",border:"#30363d"},info:{text:"#388bfd",border:"rgba(56,139,253,0.4)"},success:{text:"#3fb950",border:"rgba(46,160,67,0.4)"},warning:{text:"#e3b341",border:"rgba(242,211,91,0.4)"},danger:{text:"#f85149",border:"rgba(248,81,73,0.4)"},orange:{text:"#db6d28",border:"rgba(219,109,40,0.4)"}},input:{bg:"#0d1117",contrastBg:"rgba(1,4,9,0.5)",border:"#21262d",disabledBg:"#161b22",disabledBorder:"#30363d",warningBorder:"#d29922",errorBorder:"#f85149",tooltip:{success:{text:"#56d364",bg:"#101f1b",border:"#1c532b"},warning:{text:"#e3b341",bg:"#1e1c16",border:"#5d4411"},error:{text:"#ff7b72",bg:"#25171c",border:"#792e2e"}}},avatar:{bg:"rgba(240,246,252,0.1)",border:"rgba(240,246,252,0.1)",stackFade:"#30363d",stackFadeMore:"#21262d"},toast:{text:"#79c0ff",bg:"#010409",border:"#214981",icon:"#79c0ff",iconBg:"#121d2e",iconBorder:"#214981",success:{text:"#56d364",border:"#1c532c",icon:"#56d364",iconBg:"#111f1c",iconBorder:"#1c532c"},warning:{text:"#e3b341",border:"#5e4411",icon:"#e3b341",iconBg:"#1f1c16",iconBorder:"#5e4411"},danger:{text:"#ff7b72",border:"#792e2e",icon:"#ff7b72",iconBg:"#25171c",iconBorder:"#792e2e"},loading:{text:"#c9d1d9",border:"#30363d",icon:"#f0f6fc",iconBg:"#30363d",iconBorder:"#30363d"}},timeline:{text:"#b1bac4",badgeBg:"#0d1117",targetBadgeBorder:"#1f6feb",targetBadgeShadow:"#0d419d"},selectMenu:{borderSecondary:"#30363d",backdropBg:"rgba(1,4,9,0.5)",backdropBorder:"#484f58",tapHighlight:"rgba(48,54,61,0.5)",tapFocusBg:"#0c2d6b"},box:{blueBorder:"#0d419d",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(121,192,255,0.1)",headerBlueBg:"#0d1117",headerBlueBorder:"#30363d",borderInfo:"rgba(56,139,253,0.4)",bgInfo:"rgba(56,139,253,0.1)",borderWarning:"rgba(187,128,9,0.4)",bgWarning:"rgba(187,128,9,0.1)"},branchName:{text:"#c9d1d9",icon:"#b1bac4",bg:"rgba(88,166,255,0.1)",link:{text:"#58a6ff",icon:"#58a6ff",bg:"rgba(88,166,255,0.1)"}},markdown:{codeBg:"rgba(240,246,252,0.15)",frameBorder:"#3b434b",blockquoteBorder:"#3b434b",tableBorder:"#3b434b",tableTrBorder:"#272c32"},menu:{headingText:"#8b949e",borderActive:"#f78166",bgActive:"#161b22"},sidenav:{selectedBg:"#21262d",borderActive:"#f78166"},header:{text:"rgba(240,246,252,0.7)",bg:"#161b22",logo:"#f0f6fc"},filterItem:{barBg:"#12161c"},hiddenTextExpander:{bg:"#21262d",bgHover:"#30363d"},dragAndDrop:{border:"#25292f"},uploadEnabled:{border:"#3b434b",borderFocused:"#4487ee"},previewableCommentForm:{border:"#25292f"},underlinenav:{border:"rgba(48,54,61,0)",borderHover:"#30363d",borderActive:"#f78166",text:"#8b949e",textHover:"#c9d1d9",textActive:"#c9d1d9",icon:"#6e7681",iconHover:"#c9d1d9",iconActive:"#c9d1d9",counterText:"#8b949e"},verifiedBadge:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},socialCount:{bg:"#21262d"},tooltip:{text:"#f0f6fc",bg:"#21262d"},headerSearch:{bg:"#0d1117",border:"#21262d"},searchKeyword:{hl:"rgba(187,128,9,0.4)"},diffstat:{neutralBg:"#30363d",neutralBorder:"rgba(240,246,252,0.1)",deletionBg:"#da3633",deletionBorder:"#f85149",additionBg:"#238636",additionBorder:"#2ea043"},mktg:{success:"#29933d",info:"#2a7bf3",bgShadeGradient:{top:"rgba(1,4,9,0.065)",bottom:"rgba(1,4,9,0)"},btn:{bg:"#1f6feb",border:"#1f6feb",text:"#f0f6fc",icon:"#f0f6fc",hover:{bg:"#388bfd",border:"#388bfd"},disabled:{bg:"rgba(56,139,253,0.5)",border:"rgba(0,0,0,0)",text:"rgba(240,246,252,0.5)",icon:"rgba(240,246,252,0.5)"},primary:{bg:"#238636",border:"#238636",text:"#f0f6fc",icon:"#f0f6fc",hover:{bg:"#2ea043",border:"#2ea043"},disabled:{bg:"rgba(46,160,67,0.5)",border:"rgba(0,0,0,0)",text:"rgba(240,246,252,0.5)",icon:"rgba(240,246,252,0.5)"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(56,139,253,0.5)",text:"#388bfd",icon:"#388bfd",hover:{bg:"rgba(0,0,0,0)",border:"#388bfd",text:"#58a6ff",icon:"#58a6ff"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(56,139,253,0.2)",text:"rgba(56,139,253,0.5)",icon:"rgba(56,139,253,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(201,209,217,0.5)",text:"#c9d1d9",icon:"#c9d1d9",hover:{bg:"rgba(201,209,217,0.5)",border:"rgba(201,209,217,0.5)",text:"#0d1117",icon:"#0d1117"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(201,209,217,0.2)",text:"rgba(201,209,217,0.5)",icon:"rgba(201,209,217,0.5)"}}}},filesExplorerIcon:"#6e7681",hlAuthorBg:"#051d4d",hlAuthorBorder:"#0d419d",logoSubdued:"#30363d",discussionBorder:"#34513b",discussionBgSuccess:"rgba(46,160,67,0.1)",actionsWorkflowTableStickyBg:"rgba(13,17,23,0.95)",repoLanguageColorBorder:"rgba(240,246,252,0.2)",codeSelectionBg:"rgba(121,192,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(210,153,34,0.15)",lineHighlightBorder:"#e3b341"},diff:{addition:{text:"#56d364",bg:"rgba(46,160,67,0.2)",border:"#196c2e"},deletion:{text:"#f85149",bg:"rgba(218,54,51,0.2)",border:"#b62324"},change:{text:"#e3b341",bg:"#341a00",border:"#9e6a03"}},diffBlob:{numText:"rgba(240,246,252,0.3)",numHoverText:"rgba(240,246,252,0.6)",addition:{numText:"#3fb950",numHoverText:"#7ee787",numBg:"rgba(46,160,67,0.1)",lineBg:"rgba(46,160,67,0.2)",wordBg:"rgba(46,160,67,0.55)"},deletion:{numText:"#f85149",numHoverText:"#ffa198",numBg:"rgba(218,54,51,0.1)",lineBg:"rgba(218,54,51,0.2)",wordBg:"rgba(218,54,51,0.5)"},hunk:{text:"#8b949e",numBg:"rgba(88,166,255,0.15)",lineBg:"rgba(88,166,255,0.1)"},emptyBlockBg:"#161b22",selectedLineHighlightBg:"rgba(187,128,9,0.1)",selectedLineHighlightBorder:"#bb8009",expander:{icon:"#8b949e",hoverIcon:"#f0f6fc",hoverBg:"#1f6feb"},commentButton:{icon:"#f0f6fc",bg:"#1f6feb",gradientBg:"#367eed"}},globalNav:{logo:"#f0f6fc",bg:"#161b22",text:"#c9d1d9",icon:"#c9d1d9",inputBg:"#0d1117",inputBorder:"#21262d",inputIcon:"#21262d",inputPlaceholder:"#484f58"},calendarGraph:{dayBg:"#161b22",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#003820",dayL2Bg:"#00602d",dayL3Bg:"#10983d",dayL4Bg:"#27d545",dayL4Border:"rgba(255,255,255,0.05)",dayL3Border:"rgba(255,255,255,0.05)",dayL2Border:"rgba(255,255,255,0.05)",dayL1Border:"rgba(255,255,255,0.05)"},footerInvertocat:{octicon:"#30363d",octiconHover:"#6e7681"},prState:{draft:{text:"#8b949e",bg:"rgba(139,148,158,0.1)",border:"rgba(139,148,158,0.4)"},open:{text:"#3fb950",bg:"rgba(63,185,80,0.1)",border:"rgba(63,185,80,0.4)"},merged:{text:"#a371f7",bg:"rgba(188,140,255,0.1)",border:"rgba(188,140,255,0.4)"},closed:{text:"#f85149",bg:"rgba(218,54,51,0.1)",border:"rgba(218,54,51,0.4)"}},topicTag:{text:"#58a6ff",bg:"rgba(56,139,253,0.1)",hoverBg:"rgba(56,139,253,0.2)",activeBg:"rgba(56,139,253,0.15)"},mergeBox:{successIconBg:"rgba(46,160,67,0.1)",successIconText:"#3fb950",successIconBorder:"rgba(46,160,67,0.4)",successIndicatorBg:"#238636",successIndicatorBorder:"#2ea043",mergedIconBg:"rgba(188,140,255,0.1)",mergedIconText:"#a371f7",mergedIconBorder:"rgba(188,140,255,0.4)",mergedBoxBorder:"rgba(188,140,255,0.4)",neutralIconBg:"rgba(201,209,217,0.1)",neutralIconText:"#8b949e",neutralIconBorder:"rgba(201,209,217,0.4)",neutralIndicatorBg:"#484f58",neutralIndicatorBorder:"#6e7681",warningIconBg:"rgba(187,128,9,0.1)",warningIconText:"#e3b341",warningIconBorder:"rgba(187,128,9,0.4)",warningBoxBorder:"rgba(187,128,9,0.4)",warningMergeHighlight:"rgba(187,128,9,0.1)",errorIconBg:"rgba(248,81,73,0.1)",errorIconText:"#f85149",errorIconBorder:"rgba(248,81,73,0.4)",errorIndicatorBg:"#da3633",errorIndicatorBorder:"#f85149"},project:{cardBg:"#161b22",headerBg:"#0d1117",sidebarBg:"#161b22",gradientIn:"#161b22",gradientOut:"rgba(22,27,34,0)"},checks:{bg:"#090c10",textPrimary:"#c9d1d9",textSecondary:"#8b949e",textLink:"#58a6ff",btnIcon:"#6e7681",btnHoverIcon:"#c9d1d9",btnHoverBg:"#30363d",inputText:"#8b949e",inputPlaceholderText:"#484f58",inputFocusText:"#c9d1d9",inputBg:"#0d1117",dropdownText:"#c9d1d9",dropdownBg:"#21262d",dropdownBorder:"#30363d",dropdownHoverText:"#f0f6fc",dropdownHoverBg:"#1f6feb",dropdownBtnHoverText:"#f0f6fc",dropdownBtnHoverBg:"#161b22",scrollbarThumbBg:"#30363d",headerLabelText:"#8b949e",headerLabelOpenText:"#c9d1d9",headerBorder:"#21262d",headerIcon:"#6e7681",lineText:"#8b949e",lineNumText:"#8b949e",lineTimestampText:"#8b949e",lineHoverBg:"#161b22",lineSelectedBg:"rgba(56,139,253,0.1)",lineSelectedNumText:"#58a6ff",lineDtFmText:"#0d1117",lineDtFmBg:"#d29922",gateBg:"rgba(132,83,6,0.15)",gateText:"#8b949e",gateWaitingText:"#e3b341",stepHeaderOpenBg:"#161b22",stepErrorText:"#f85149",stepWarningText:"#e3b341",loglineText:"#6e7681",loglineNumText:"#8b949e",loglineDebugText:"#bc8cff",loglineErrorText:"#8b949e",loglineErrorNumText:"#8b949e",loglineErrorBg:"rgba(248,81,73,0.1)",loglineWarningText:"#8b949e",loglineWarningNumText:"#e3b341",loglineWarningBg:"rgba(187,128,9,0.1)",loglineCommandText:"#58a6ff",loglineSectionText:"#56d364"},introShelf:{gradientLeft:"rgba(56,139,253,0.1)",gradientRight:"rgba(46,160,67,0.1)",gradientIn:"#0d1117",gradientOut:"rgba(13,17,23,0)"},marketingIcon:{primary:"#79c0ff",secondary:"#1f6feb"},prettylights:{syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",storageModifierImport:"#c9d1d9",entityTag:"#7ee787",keyword:"#ff7b72",string:"#a5d6ff",variable:"#ffa657",brackethighlighterUnmatched:"#f85149",invalidIllegalText:"#f0f6fc",invalidIllegalBg:"#8e1519",carriageReturnText:"#f0f6fc",carriageReturnBg:"#b62324",stringRegexp:"#7ee787",markupList:"#f2cc60",markupHeading:"#1f6feb",markupItalic:"#c9d1d9",markupBold:"#c9d1d9",markupDeletedText:"#ffdcd7",markupDeletedBg:"#67060c",markupInsertedText:"#aff5b4",markupInsertedBg:"#033a16",markupChangedText:"#ffdfb6",markupChangedBg:"#5a1e02",markupIgnoredText:"#c9d1d9",markupIgnoredBg:"#1158c7",metaDiffRange:"#d2a8ff",brackethighlighterAngle:"#8b949e",sublimelinterGutterMark:"#484f58",constantOtherReferenceLink:"#a5d6ff"}},codemirror:{text:"#c9d1d9",bg:"#0d1117",guttersBg:"#0d1117",guttermarkerText:"#0d1117",guttermarkerSubtleText:"#6e7681",linenumberText:"#8b949e",cursor:"#f0f6fc",selectionBg:"rgba(121,192,255,0.3)",activelineBg:"#161b22",matchingbracketText:"#c9d1d9",linesBg:"#0d1117",syntax:{comment:"#8b949e",constant:"#79c0ff",entity:"#d2a8ff",keyword:"#ff7b72",storage:"#ff7b72",string:"#a5d6ff",support:"#79c0ff",variable:"#ffa657"}},ansi:{black:"#0d1117",blackBright:"#161b22",white:"#b1bac4",whiteBright:"#b1bac4",gray:"#6e7681",red:"#ff7b72",redBright:"#ffa198",green:"#3fb950",greenBright:"#56d364",yellow:"#d29922",yellowBright:"#e3b341",blue:"#58a6ff",blueBright:"#79c0ff",magenta:"#bc8cff",magentaBright:"#d2a8ff",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 0 rgba(0,0,0,0)",highlight:"0 0 rgba(0,0,0,0)",inset:"0 0 rgba(0,0,0,0)"},autocomplete:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},btn:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(139,148,158,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(1,4,9,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(31,111,235,0.3)",primary:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(1,4,9,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(240,246,252,0.03)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(17,88,199,0.4)"},danger:{hoverShadow:"0 0 rgba(0,0,0,0)",hoverInsetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(182,35,36,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(1,4,9,0.85)"},input:{shadow:"0 0 rgba(0,0,0,0)"},selectMenu:{shadow:"0 0 18px rgba(1,4,9,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(31,111,235,0.3)",primary:{focusShadow:"0 0 0 3px rgba(35,134,54,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(56,139,253,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(201,209,217,0.3)"}}}}},dark_dimmed:{colors:{auto:{black:"#cdd9e5",white:"#1c2128",gray:["#22272e","#2d333b","#373e47","#444c56","#545d68","#636e7b","#768390","#909dab","#adbac7","#cdd9e5"],blue:["#0f2d5c","#143d79","#1b4b91","#255ab2","#316dca","#4184e4","#539bf5","#6cb6ff","#96d0ff","#c6e6ff"],green:["#113417","#1b4721","#245829","#2b6a30","#347d39","#46954a","#57ab5a","#6bc46d","#8ddb8c","#b4f1b4"],yellow:["#452700","#593600","#6c4400","#805400","#966600","#ae7c14","#c69026","#daaa3f","#eac55f","#fbe090"],orange:["#4d210c","#682d0f","#7f3913","#94471b","#ae5622","#cc6b2c","#e0823d","#f69d50","#ffbc6f","#ffddb0"],red:["#78191b","#78191b","#922323","#ad2e2c","#c93c37","#e5534b","#f47067","#ff938a","#ffb8b0","#ffd8d3"],purple:["#352160","#472c82","#5936a2","#6b44bc","#8256d0","#986ee2","#b083f0","#dcbdfb","#dcbdfb","#eedcff"],pink:["#551639","#69264a","#7e325a","#983b6e","#ae4c82","#c96198","#e275ad","#fc8dc7","#ffb3d8","#ffd7eb"]},text:{primary:"#adbac7",secondary:"#768390",tertiary:"#768390",placeholder:"#545d68",disabled:"#545d68",inverse:"#22272e",link:"#539bf5",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f",white:"#cdd9e5"},icon:{primary:"#adbac7",secondary:"#636e7b",tertiary:"#545d68",info:"#6cb6ff",danger:"#e5534b",success:"#6bc46d",warning:"#daaa3f"},border:{primary:"#444c56",secondary:"#373e47",tertiary:"#636e7b",overlay:"#444c56",inverse:"#cdd9e5",info:"rgba(65,132,228,0.4)",danger:"rgba(229,83,75,0.4)",success:"rgba(87,171,90,0.4)",warning:"rgba(174,124,20,0.4)"},bg:{canvas:"#22272e",canvasMobile:"#1c2128",canvasInverse:"#cdd9e5",canvasInset:"#1e2228",primary:"#22272e",secondary:"#22272e",tertiary:"#2d333b",overlay:"#373e47",backdrop:"rgba(28,33,40,0.8)",info:"rgba(65,132,228,0.1)",infoInverse:"#4184e4",danger:"rgba(229,83,75,0.1)",dangerInverse:"#c93c37",success:"rgba(70,149,74,0.1)",successInverse:"#46954a",warning:"rgba(174,124,20,0.1)",warningInverse:"#ae7c14"},state:{hover:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5",secondaryBg:"#2d333b",secondaryBorder:"#2d333b"},selected:{primaryBg:"#316dca",primaryBorder:"#4184e4",primaryText:"#cdd9e5",primaryIcon:"#cdd9e5"},focus:{border:"#4184e4"}},fade:{fg10:"rgba(205,217,229,0.1)",fg15:"rgba(205,217,229,0.15)",fg30:"rgba(205,217,229,0.3)",fg50:"rgba(205,217,229,0.5)",fg70:"rgba(205,217,229,0.7)",fg85:"rgba(205,217,229,0.85)",black10:"rgba(28,33,40,0.1)",black15:"rgba(28,33,40,0.15)",black30:"rgba(28,33,40,0.3)",black50:"rgba(28,33,40,0.5)",black70:"rgba(28,33,40,0.7)",black85:"rgba(28,33,40,0.85)",white10:"rgba(205,217,229,0.1)",white15:"rgba(205,217,229,0.15)",white30:"rgba(205,217,229,0.3)",white50:"rgba(205,217,229,0.5)",white70:"rgba(205,217,229,0.7)",white85:"rgba(205,217,229,0.85)"},alert:{info:{text:"#6cb6ff",icon:"#6cb6ff",bg:"rgba(65,132,228,0.1)",border:"rgba(65,132,228,0.4)"},warn:{text:"#daaa3f",icon:"#daaa3f",bg:"rgba(174,124,20,0.1)",border:"rgba(174,124,20,0.4)"},error:{text:"#f47067",icon:"#f47067",bg:"rgba(229,83,75,0.1)",border:"rgba(229,83,75,0.4)"},success:{text:"#6bc46d",icon:"#6bc46d",bg:"rgba(70,149,74,0.1)",border:"rgba(70,149,74,0.4)"}},autocomplete:{rowBorder:"#444c56"},blankslate:{icon:"#5f6a76"},btn:{text:"#adbac7",bg:"#373e47",border:"#444c56",hoverBg:"#444c56",hoverBorder:"#768390",selectedBg:"#2d333b",focusBg:"#373e47",focusBorder:"#768390",primary:{text:"#ffffff",bg:"#347d39",border:"#46954a",hoverBg:"#46954a",hoverBorder:"#57ab5a",selectedBg:"#347d39",disabledText:"rgba(205,217,229,0.5)",disabledBg:"rgba(52,125,57,0.6)",disabledBorder:"rgba(0,0,0,0)",focusBg:"#347d39",focusBorder:"#57ab5a",icon:"#cdd9e5",counterBg:"rgba(205,217,229,0.2)"},outline:{text:"#539bf5",hoverText:"#539bf5",hoverBg:"#444c56",hoverBorder:"#539bf5",hoverCounterBg:"rgba(205,217,229,0.2)",selectedText:"#cdd9e5",selectedBg:"#1b4b91",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(83,155,245,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(49,109,202,0.05)",focusBorder:"#539bf5",counterBg:"rgba(49,109,202,0.1)"},danger:{text:"#e5534b",hoverText:"#ffffff",hoverBg:"#c93c37",hoverBorder:"#e5534b",hoverCounterBg:"rgba(255,255,255,0.2)",selectedText:"#ffffff",selectedBg:"#ad2e2c",selectedBorder:"rgba(205,217,229,0.1)",disabledText:"rgba(229,83,75,0.5)",disabledBg:"#22272e",disabledCounterBg:"rgba(201,60,55,0.05)",focusBorder:"#e5534b",counterBg:"rgba(201,60,55,0.1)",icon:"#e5534b",hoverIcon:"#cdd9e5"},counterBg:"#444c56"},counter:{text:"#adbac7",bg:"#444c56",primary:{text:"#adbac7",bg:"#636e7b"},secondary:{text:"#768390"}},label:{border:"#444c56",primary:{text:"#909dab",border:"#636e7b"},secondary:{text:"#768390",border:"#444c56"},info:{text:"#4184e4",border:"rgba(65,132,228,0.4)"},success:{text:"#57ab5a",border:"rgba(70,149,74,0.4)"},warning:{text:"#daaa3f",border:"rgba(242,211,91,0.4)"},danger:{text:"#e5534b",border:"rgba(229,83,75,0.4)"},orange:{text:"#cc6b2c",border:"rgba(204,107,44,0.4)"}},input:{bg:"#22272e",contrastBg:"rgba(28,33,40,0.5)",border:"#373e47",disabledBg:"#2d333b",disabledBorder:"#444c56",warningBorder:"#c69026",errorBorder:"#e5534b",tooltip:{success:{text:"#6bc46d",bg:"#263231",border:"#335a3b"},warning:{text:"#daaa3f",bg:"#30302b",border:"#624e22"},error:{text:"#f47067",bg:"#362b31",border:"#7c3b3b"}}},avatar:{bg:"rgba(205,217,229,0.1)",border:"rgba(205,217,229,0.1)",stackFade:"#444c56",stackFadeMore:"#373e47"},toast:{text:"#6cb6ff",bg:"#1c2128",border:"#214981",icon:"#6cb6ff",iconBg:"#121d2e",iconBorder:"#214981",success:{text:"#6bc46d",border:"#1c532c",icon:"#6bc46d",iconBg:"#111f1c",iconBorder:"#1c532c"},warning:{text:"#daaa3f",border:"#5e4411",icon:"#daaa3f",iconBg:"#1f1c16",iconBorder:"#5e4411"},danger:{text:"#f47067",border:"#792e2e",icon:"#f47067",iconBg:"#25171c",iconBorder:"#792e2e"},loading:{text:"#adbac7",border:"#444c56",icon:"#cdd9e5",iconBg:"#444c56",iconBorder:"#444c56"}},timeline:{text:"#909dab",badgeBg:"#22272e",targetBadgeBorder:"#316dca",targetBadgeShadow:"#1b4b91"},selectMenu:{borderSecondary:"#444c56",backdropBg:"rgba(28,33,40,0.5)",backdropBorder:"#545d68",tapHighlight:"rgba(68,76,86,0.5)",tapFocusBg:"#143d79"},box:{blueBorder:"#1b4b91",rowYellowBg:"rgba(235,196,64,0.1)",rowBlueBg:"rgba(108,182,255,0.1)",headerBlueBg:"#22272e",headerBlueBorder:"#444c56",borderInfo:"rgba(65,132,228,0.4)",bgInfo:"rgba(65,132,228,0.1)",borderWarning:"rgba(174,124,20,0.4)",bgWarning:"rgba(174,124,20,0.1)"},branchName:{text:"#adbac7",icon:"#909dab",bg:"rgba(83,155,245,0.1)",link:{text:"#539bf5",icon:"#539bf5",bg:"rgba(83,155,245,0.1)"}},markdown:{codeBg:"rgba(205,217,229,0.15)",frameBorder:"#4f5964",blockquoteBorder:"#4f5964",tableBorder:"#4f5964",tableTrBorder:"#3b424b"},menu:{headingText:"#768390",borderActive:"#f78166",bgActive:"#2d333b"},sidenav:{selectedBg:"#373e47",borderActive:"#f78166"},header:{text:"rgba(205,217,229,0.7)",bg:"#2d333b",logo:"#cdd9e5"},filterItem:{barBg:"#292e35"},hiddenTextExpander:{bg:"#373e47",bgHover:"#444c56"},dragAndDrop:{border:"#393f48"},uploadEnabled:{border:"#4f5964",borderFocused:"#4f84d4"},previewableCommentForm:{border:"#393f48"},underlinenav:{border:"rgba(68,76,86,0)",borderHover:"#444c56",borderActive:"#f78166",text:"#768390",textHover:"#adbac7",textActive:"#adbac7",icon:"#636e7b",iconHover:"#adbac7",iconActive:"#adbac7",counterText:"#768390"},verifiedBadge:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},socialCount:{bg:"#373e47"},tooltip:{text:"#cdd9e5",bg:"#373e47"},headerSearch:{bg:"#22272e",border:"#373e47"},searchKeyword:{hl:"rgba(174,124,20,0.4)"},diffstat:{neutralBg:"#444c56",neutralBorder:"rgba(205,217,229,0.1)",deletionBg:"#c93c37",deletionBorder:"#e5534b",additionBg:"#347d39",additionBorder:"#46954a"},mktg:{success:"#3d8942",info:"#3877d5",bgShadeGradient:{top:"rgba(28,33,40,0.065)",bottom:"rgba(28,33,40,0)"},btn:{bg:"#316dca",border:"#316dca",text:"#cdd9e5",icon:"#cdd9e5",hover:{bg:"#4184e4",border:"#4184e4"},disabled:{bg:"rgba(65,132,228,0.5)",border:"rgba(0,0,0,0)",text:"rgba(205,217,229,0.5)",icon:"rgba(205,217,229,0.5)"},primary:{bg:"#347d39",border:"#347d39",text:"#cdd9e5",icon:"#cdd9e5",hover:{bg:"#46954a",border:"#46954a"},disabled:{bg:"rgba(70,149,74,0.5)",border:"rgba(0,0,0,0)",text:"rgba(205,217,229,0.5)",icon:"rgba(205,217,229,0.5)"}},outline:{bg:"rgba(0,0,0,0)",border:"rgba(65,132,228,0.5)",text:"#4184e4",icon:"#4184e4",hover:{bg:"rgba(0,0,0,0)",border:"#4184e4",text:"#539bf5",icon:"#539bf5"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(65,132,228,0.2)",text:"rgba(65,132,228,0.5)",icon:"rgba(65,132,228,0.5)"}},dark:{bg:"rgba(0,0,0,0)",border:"rgba(173,186,199,0.5)",text:"#adbac7",icon:"#adbac7",hover:{bg:"rgba(173,186,199,0.5)",border:"rgba(173,186,199,0.5)",text:"#22272e",icon:"#22272e"},disabled:{bg:"rgba(0,0,0,0)",border:"rgba(173,186,199,0.2)",text:"rgba(173,186,199,0.5)",icon:"rgba(173,186,199,0.5)"}}}},filesExplorerIcon:"#636e7b",hlAuthorBg:"#0f2d5c",hlAuthorBorder:"#1b4b91",logoSubdued:"#444c56",discussionBorder:"#494c49",discussionBgSuccess:"rgba(70,149,74,0.1)",actionsWorkflowTableStickyBg:"rgba(34,39,46,0.95)",repoLanguageColorBorder:"rgba(205,217,229,0.2)",codeSelectionBg:"rgba(108,182,255,0.3)",highlight:{text:"#ffd467",bg:"rgba(204,143,44,0.38)"},blob:{lineHighlightBg:"rgba(198,144,38,0.15)",lineHighlightBorder:"#daaa3f"},diff:{addition:{text:"#6bc46d",bg:"rgba(70,149,74,0.2)",border:"#2b6a30"},deletion:{text:"#e5534b",bg:"rgba(201,60,55,0.2)",border:"#ad2e2c"},change:{text:"#daaa3f",bg:"#452700",border:"#966600"}},diffBlob:{numText:"rgba(205,217,229,0.3)",numHoverText:"rgba(205,217,229,0.6)",addition:{numText:"#57ab5a",numHoverText:"#8ddb8c",numBg:"rgba(70,149,74,0.1)",lineBg:"rgba(70,149,74,0.2)",wordBg:"rgba(70,149,74,0.55)"},deletion:{numText:"#e5534b",numHoverText:"#ff938a",numBg:"rgba(201,60,55,0.1)",lineBg:"rgba(201,60,55,0.2)",wordBg:"rgba(201,60,55,0.5)"},hunk:{text:"#768390",numBg:"rgba(83,155,245,0.15)",lineBg:"rgba(83,155,245,0.1)"},emptyBlockBg:"#2d333b",selectedLineHighlightBg:"rgba(174,124,20,0.1)",selectedLineHighlightBorder:"#ae7c14",expander:{icon:"#768390",hoverIcon:"#cdd9e5",hoverBg:"#316dca"},commentButton:{icon:"#cdd9e5",bg:"#316dca",gradientBg:"#437bd1"}},globalNav:{logo:"#cdd9e5",bg:"#2d333b",text:"#adbac7",icon:"#adbac7",inputBg:"#22272e",inputBorder:"#373e47",inputIcon:"#373e47",inputPlaceholder:"#545d68"},calendarGraph:{dayBg:"#2d333b",dayBorder:"rgba(27,31,35,0.06)",dayL1Bg:"#003820",dayL2Bg:"#00602d",dayL3Bg:"#10983d",dayL4Bg:"#27d545",dayL4Border:"rgba(255,255,255,0.05)",dayL3Border:"rgba(255,255,255,0.05)",dayL2Border:"rgba(255,255,255,0.05)",dayL1Border:"rgba(255,255,255,0.05)"},footerInvertocat:{octicon:"#444c56",octiconHover:"#636e7b"},prState:{draft:{text:"#768390",bg:"rgba(118,131,144,0.1)",border:"rgba(118,131,144,0.4)"},open:{text:"#57ab5a",bg:"rgba(87,171,90,0.1)",border:"rgba(87,171,90,0.4)"},merged:{text:"#986ee2",bg:"rgba(176,131,240,0.1)",border:"rgba(176,131,240,0.4)"},closed:{text:"#e5534b",bg:"rgba(201,60,55,0.1)",border:"rgba(201,60,55,0.4)"}},topicTag:{text:"#539bf5",bg:"rgba(65,132,228,0.1)",hoverBg:"rgba(65,132,228,0.2)",activeBg:"rgba(65,132,228,0.15)"},mergeBox:{successIconBg:"rgba(70,149,74,0.1)",successIconText:"#57ab5a",successIconBorder:"rgba(70,149,74,0.4)",successIndicatorBg:"#347d39",successIndicatorBorder:"#46954a",mergedIconBg:"rgba(176,131,240,0.1)",mergedIconText:"#986ee2",mergedIconBorder:"rgba(176,131,240,0.4)",mergedBoxBorder:"rgba(176,131,240,0.4)",neutralIconBg:"rgba(173,186,199,0.1)",neutralIconText:"#768390",neutralIconBorder:"rgba(173,186,199,0.4)",neutralIndicatorBg:"#545d68",neutralIndicatorBorder:"#636e7b",warningIconBg:"rgba(174,124,20,0.1)",warningIconText:"#daaa3f",warningIconBorder:"rgba(174,124,20,0.4)",warningBoxBorder:"rgba(174,124,20,0.4)",warningMergeHighlight:"rgba(174,124,20,0.1)",errorIconBg:"rgba(229,83,75,0.1)",errorIconText:"#e5534b",errorIconBorder:"rgba(229,83,75,0.4)",errorIndicatorBg:"#c93c37",errorIndicatorBorder:"#e5534b"},project:{cardBg:"#2d333b",headerBg:"#22272e",sidebarBg:"#2d333b",gradientIn:"#2d333b",gradientOut:"rgba(45,51,59,0)"},checks:{bg:"#1e2228",textPrimary:"#adbac7",textSecondary:"#768390",textLink:"#539bf5",btnIcon:"#636e7b",btnHoverIcon:"#adbac7",btnHoverBg:"#444c56",inputText:"#768390",inputPlaceholderText:"#545d68",inputFocusText:"#adbac7",inputBg:"#22272e",dropdownText:"#adbac7",dropdownBg:"#373e47",dropdownBorder:"#444c56",dropdownHoverText:"#cdd9e5",dropdownHoverBg:"#316dca",dropdownBtnHoverText:"#cdd9e5",dropdownBtnHoverBg:"#2d333b",scrollbarThumbBg:"#444c56",headerLabelText:"#768390",headerLabelOpenText:"#adbac7",headerBorder:"#373e47",headerIcon:"#636e7b",lineText:"#768390",lineNumText:"#768390",lineTimestampText:"#768390",lineHoverBg:"#2d333b",lineSelectedBg:"rgba(65,132,228,0.1)",lineSelectedNumText:"#539bf5",lineDtFmText:"#22272e",lineDtFmBg:"#c69026",gateBg:"rgba(128,84,0,0.15)",gateText:"#768390",gateWaitingText:"#daaa3f",stepHeaderOpenBg:"#2d333b",stepErrorText:"#e5534b",stepWarningText:"#daaa3f",loglineText:"#636e7b",loglineNumText:"#768390",loglineDebugText:"#b083f0",loglineErrorText:"#768390",loglineErrorNumText:"#768390",loglineErrorBg:"rgba(229,83,75,0.1)",loglineWarningText:"#768390",loglineWarningNumText:"#daaa3f",loglineWarningBg:"rgba(174,124,20,0.1)",loglineCommandText:"#539bf5",loglineSectionText:"#6bc46d"},introShelf:{gradientLeft:"rgba(65,132,228,0.1)",gradientRight:"rgba(70,149,74,0.1)",gradientIn:"#22272e",gradientOut:"rgba(34,39,46,0)"},marketingIcon:{primary:"#6cb6ff",secondary:"#316dca"},prettylights:{syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",storageModifierImport:"#adbac7",entityTag:"#8ddb8c",keyword:"#f47067",string:"#96d0ff",variable:"#f69d50",brackethighlighterUnmatched:"#e5534b",invalidIllegalText:"#cdd9e5",invalidIllegalBg:"#922323",carriageReturnText:"#cdd9e5",carriageReturnBg:"#ad2e2c",stringRegexp:"#8ddb8c",markupList:"#eac55f",markupHeading:"#316dca",markupItalic:"#adbac7",markupBold:"#adbac7",markupDeletedText:"#ffd8d3",markupDeletedBg:"#78191b",markupInsertedText:"#b4f1b4",markupInsertedBg:"#1b4721",markupChangedText:"#ffddb0",markupChangedBg:"#682d0f",markupIgnoredText:"#adbac7",markupIgnoredBg:"#255ab2",metaDiffRange:"#dcbdfb",brackethighlighterAngle:"#768390",sublimelinterGutterMark:"#545d68",constantOtherReferenceLink:"#96d0ff"}},codemirror:{text:"#adbac7",bg:"#22272e",guttersBg:"#22272e",guttermarkerText:"#22272e",guttermarkerSubtleText:"#636e7b",linenumberText:"#768390",cursor:"#cdd9e5",selectionBg:"rgba(108,182,255,0.3)",activelineBg:"#2d333b",matchingbracketText:"#adbac7",linesBg:"#22272e",syntax:{comment:"#768390",constant:"#6cb6ff",entity:"#dcbdfb",keyword:"#f47067",storage:"#f47067",string:"#96d0ff",support:"#6cb6ff",variable:"#f69d50"}},ansi:{black:"#22272e",blackBright:"#2d333b",white:"#909dab",whiteBright:"#909dab",gray:"#636e7b",red:"#f47067",redBright:"#ff938a",green:"#57ab5a",greenBright:"#6bc46d",yellow:"#c69026",yellowBright:"#daaa3f",blue:"#539bf5",blueBright:"#6cb6ff",magenta:"#b083f0",magentaBright:"#dcbdfb",cyan:"#76e3ea",cyanBright:"#b3f0ff"}},shadows:{shadow:{small:"0 0 rgba(0,0,0,0)",highlight:"0 0 rgba(0,0,0,0)",inset:"0 0 rgba(0,0,0,0)"},autocomplete:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},btn:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(118,131,144,0.3)",shadowActive:"inset 0 0.15em 0.3em rgba(28,33,40,0.15)",shadowInputFocus:"0 0 0 0.2em rgba(49,109,202,0.3)",primary:{shadow:"0 0 rgba(0,0,0,0)",insetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(46,164,79,0.4)"},outline:{hoverShadow:"0 1px 0 rgba(28,33,40,0.1)",hoverInsetShadow:"inset 0 1px 0 rgba(205,217,229,0.03)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(37,90,178,0.4)"},danger:{hoverShadow:"0 0 rgba(0,0,0,0)",hoverInsetShadow:"0 0 rgba(0,0,0,0)",selectedShadow:"0 0 rgba(0,0,0,0)",focusShadow:"0 0 0 3px rgba(173,46,44,0.4)"}},dropdown:{shadow:"0 16px 32px rgba(28,33,40,0.85)"},input:{shadow:"0 0 rgba(0,0,0,0)"},selectMenu:{shadow:"0 0 18px rgba(28,33,40,0.4)"},mktg:{btn:{focusShadow:"0 0 0 3px rgba(49,109,202,0.3)",primary:{focusShadow:"0 0 0 3px rgba(52,125,57,0.3)"},outline:{focusShadow:"0 0 0 3px rgba(65,132,228,0.3)"},dark:{focusShadow:"0 0 0 3px rgba(173,186,199,0.3)"}}}}}}}},c=Object.getOwnPropertySymbols,s=Object.prototype.hasOwnProperty,f=Object.prototype.propertyIsEnumerable;function b(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}var g=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var t={},r=0;r<10;r++)t["_"+String.fromCharCode(r)]=r;if("0123456789"!==Object.getOwnPropertyNames(t).map((function(e){return t[e]})).join(""))return!1;var o={};return"abcdefghijklmnopqrst".split("").forEach((function(e){o[e]=e})),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},o)).join("")}catch(e){return!1}}()?Object.assign:function(e,t){for(var r,o,a=b(e),n=1;n<arguments.length;n++){for(var i in r=Object(arguments[n]))s.call(r,i)&&(a[i]=r[i]);if(c){o=c(r);for(var d=0;d<o.length;d++)f.call(r,o[d])&&(a[o[d]]=r[o[d]])}}return a},u=function(e,t){var r=g({},e,t);for(var o in e){var a;e[o]&&"object"==typeof t[o]&&g(r,((a={})[o]=g(e[o],t[o]),a))}return r},p={breakpoints:[40,52,64].map((function(e){return e+"em"}))},m=function(e){return"@media screen and (min-width: "+e+")"},h=function(e,t){return x(t,e,e)},x=function(e,t,r,o,a){for(t=t&&t.split?t.split("."):[t],o=0;o<t.length;o++)e=e?e[t[o]]:a;return e===a?r:e},v=function e(t){var r={},o=function(e){var o,a,n={},i=!1,d=e.theme&&e.theme.disableStyledSystemCache;for(var l in e)if(t[l]){var c=t[l],s=e[l],f=x(e.theme,c.scale,c.defaults);if("object"!=typeof s)g(n,c(s,f,e));else{if(r.breakpoints=!d&&r.breakpoints||x(e.theme,"breakpoints",p.breakpoints),Array.isArray(s)){r.media=!d&&r.media||[null].concat(r.breakpoints.map(m)),n=u(n,y(r.media,c,f,s,e));continue}null!==s&&(n=u(n,w(r.breakpoints,c,f,s,e)),i=!0)}}return i&&(o=n,a={},Object.keys(o).sort((function(e,t){return e.localeCompare(t,void 0,{numeric:!0,sensitivity:"base"})})).forEach((function(e){a[e]=o[e]})),n=a),n};o.config=t,o.propNames=Object.keys(t),o.cache=r;var a=Object.keys(t).filter((function(e){return"config"!==e}));return a.length>1&&a.forEach((function(r){var a;o[r]=e(((a={})[r]=t[r],a))})),o},y=function(e,t,r,o,a){var n={};return o.slice(0,e.length).forEach((function(o,i){var d,l=e[i],c=t(o,r,a);l?g(n,((d={})[l]=g({},n[l],c),d)):g(n,c)})),n},w=function(e,t,r,o,a){var n={};for(var i in o){var d=e[i],l=t(o[i],r,a);if(d){var c,s=m(d);g(n,((c={})[s]=g({},n[s],l),c))}else g(n,l)}return n},B=function(e){var t=e.properties,r=e.property,o=e.scale,a=e.transform,n=void 0===a?h:a,i=e.defaultScale;t=t||[r];var d=function(e,r,o){var a={},i=n(e,r,o);if(null!==i)return t.forEach((function(e){a[e]=i})),a};return d.scale=o,d.defaults=i,d},$=function(e){void 0===e&&(e={});var t={};return Object.keys(e).forEach((function(r){var o=e[r];t[r]=!0!==o?"function"!=typeof o?B(o):o:B({property:r,scale:r})})),v(t)},k=function(){for(var e={},t=arguments.length,r=new Array(t),o=0;o<t;o++)r[o]=arguments[o];r.forEach((function(t){t&&t.config&&g(e,t.config)}));var a=v(e);return a},S=$({width:{property:"width",scale:"sizes",transform:function(e,t){return x(t,e,!function(e){return"number"==typeof e&&!isNaN(e)}(e)||e>1?e:100*e+"%")}},height:{property:"height",scale:"sizes"},minWidth:{property:"minWidth",scale:"sizes"},minHeight:{property:"minHeight",scale:"sizes"},maxWidth:{property:"maxWidth",scale:"sizes"},maxHeight:{property:"maxHeight",scale:"sizes"},size:{properties:["width","height"],scale:"sizes"},overflow:!0,overflowX:!0,overflowY:!0,display:!0,verticalAlign:!0}),T={color:{property:"color",scale:"colors"},backgroundColor:{property:"backgroundColor",scale:"colors"},opacity:!0};T.bg=T.backgroundColor;var E=$(T),C=$({fontFamily:{property:"fontFamily",scale:"fonts"},fontSize:{property:"fontSize",scale:"fontSizes",defaultScale:[12,14,16,20,24,32,48,64,72]},fontWeight:{property:"fontWeight",scale:"fontWeights"},lineHeight:{property:"lineHeight",scale:"lineHeights"},letterSpacing:{property:"letterSpacing",scale:"letterSpacings"},textAlign:!0,fontStyle:!0}),I=$({alignItems:!0,alignContent:!0,justifyItems:!0,justifyContent:!0,flexWrap:!0,flexDirection:!0,flex:!0,flexGrow:!0,flexShrink:!0,flexBasis:!0,justifySelf:!0,alignSelf:!0,order:!0}),N={space:[0,4,8,16,32,64,128,256,512]},A=$({gridGap:{property:"gridGap",scale:"space",defaultScale:N.space},gridColumnGap:{property:"gridColumnGap",scale:"space",defaultScale:N.space},gridRowGap:{property:"gridRowGap",scale:"space",defaultScale:N.space},gridColumn:!0,gridRow:!0,gridAutoFlow:!0,gridAutoColumns:!0,gridAutoRows:!0,gridTemplateColumns:!0,gridTemplateRows:!0,gridTemplateAreas:!0,gridArea:!0}),z={border:{property:"border",scale:"borders"},borderWidth:{property:"borderWidth",scale:"borderWidths"},borderStyle:{property:"borderStyle",scale:"borderStyles"},borderColor:{property:"borderColor",scale:"colors"},borderRadius:{property:"borderRadius",scale:"radii"},borderTop:{property:"borderTop",scale:"borders"},borderTopLeftRadius:{property:"borderTopLeftRadius",scale:"radii"},borderTopRightRadius:{property:"borderTopRightRadius",scale:"radii"},borderRight:{property:"borderRight",scale:"borders"},borderBottom:{property:"borderBottom",scale:"borders"},borderBottomLeftRadius:{property:"borderBottomLeftRadius",scale:"radii"},borderBottomRightRadius:{property:"borderBottomRightRadius",scale:"radii"},borderLeft:{property:"borderLeft",scale:"borders"},borderX:{properties:["borderLeft","borderRight"],scale:"borders"},borderY:{properties:["borderTop","borderBottom"],scale:"borders"},borderTopWidth:{property:"borderTopWidth",scale:"borderWidths"},borderTopColor:{property:"borderTopColor",scale:"colors"},borderTopStyle:{property:"borderTopStyle",scale:"borderStyles"}};z.borderTopLeftRadius={property:"borderTopLeftRadius",scale:"radii"},z.borderTopRightRadius={property:"borderTopRightRadius",scale:"radii"},z.borderBottomWidth={property:"borderBottomWidth",scale:"borderWidths"},z.borderBottomColor={property:"borderBottomColor",scale:"colors"},z.borderBottomStyle={property:"borderBottomStyle",scale:"borderStyles"},z.borderBottomLeftRadius={property:"borderBottomLeftRadius",scale:"radii"},z.borderBottomRightRadius={property:"borderBottomRightRadius",scale:"radii"},z.borderLeftWidth={property:"borderLeftWidth",scale:"borderWidths"},z.borderLeftColor={property:"borderLeftColor",scale:"colors"},z.borderLeftStyle={property:"borderLeftStyle",scale:"borderStyles"},z.borderRightWidth={property:"borderRightWidth",scale:"borderWidths"},z.borderRightColor={property:"borderRightColor",scale:"colors"},z.borderRightStyle={property:"borderRightStyle",scale:"borderStyles"};var R=$(z),L={background:!0,backgroundImage:!0,backgroundSize:!0,backgroundPosition:!0,backgroundRepeat:!0};L.bgImage=L.backgroundImage,L.bgSize=L.backgroundSize,L.bgPosition=L.backgroundPosition,L.bgRepeat=L.backgroundRepeat;var M=$(L),O={space:[0,4,8,16,32,64,128,256,512]},H=$({position:!0,zIndex:{property:"zIndex",scale:"zIndices"},top:{property:"top",scale:"space",defaultScale:O.space},right:{property:"right",scale:"space",defaultScale:O.space},bottom:{property:"bottom",scale:"space",defaultScale:O.space},left:{property:"left",scale:"space",defaultScale:O.space}}),P={space:[0,4,8,16,32,64,128,256,512]},D=function(e){return"number"==typeof e&&!isNaN(e)},W=function(e,t){if(!D(e))return x(t,e,e);var r=e<0,o=Math.abs(e),a=x(t,o,o);return D(a)?a*(r?-1:1):r?"-"+a:a},F={};F.margin={margin:{property:"margin",scale:"space",transform:W,defaultScale:P.space},marginTop:{property:"marginTop",scale:"space",transform:W,defaultScale:P.space},marginRight:{property:"marginRight",scale:"space",transform:W,defaultScale:P.space},marginBottom:{property:"marginBottom",scale:"space",transform:W,defaultScale:P.space},marginLeft:{property:"marginLeft",scale:"space",transform:W,defaultScale:P.space},marginX:{properties:["marginLeft","marginRight"],scale:"space",transform:W,defaultScale:P.space},marginY:{properties:["marginTop","marginBottom"],scale:"space",transform:W,defaultScale:P.space}},F.margin.m=F.margin.margin,F.margin.mt=F.margin.marginTop,F.margin.mr=F.margin.marginRight,F.margin.mb=F.margin.marginBottom,F.margin.ml=F.margin.marginLeft,F.margin.mx=F.margin.marginX,F.margin.my=F.margin.marginY,F.padding={padding:{property:"padding",scale:"space",defaultScale:P.space},paddingTop:{property:"paddingTop",scale:"space",defaultScale:P.space},paddingRight:{property:"paddingRight",scale:"space",defaultScale:P.space},paddingBottom:{property:"paddingBottom",scale:"space",defaultScale:P.space},paddingLeft:{property:"paddingLeft",scale:"space",defaultScale:P.space},paddingX:{properties:["paddingLeft","paddingRight"],scale:"space",defaultScale:P.space},paddingY:{properties:["paddingTop","paddingBottom"],scale:"space",defaultScale:P.space}},F.padding.p=F.padding.padding,F.padding.pt=F.padding.paddingTop,F.padding.pr=F.padding.paddingRight,F.padding.pb=F.padding.paddingBottom,F.padding.pl=F.padding.paddingLeft,F.padding.px=F.padding.paddingX,F.padding.py=F.padding.paddingY;var j=$(F.margin),U=$(F.padding),_=k(j,U),G=$({boxShadow:{property:"boxShadow",scale:"shadows"},textShadow:{property:"textShadow",scale:"shadows"}});function K(){return(K=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e}).apply(this,arguments)}var V=function(e,t,r,o,a){for(t=t&&t.split?t.split("."):[t],o=0;o<t.length;o++)e=e?e[t[o]]:a;return e===a?r:e},X=[40,52,64].map((function(e){return e+"em"})),q={space:[0,4,8,16,32,64,128,256,512],fontSizes:[12,14,16,20,24,32,48,64,72]},Y={bg:"backgroundColor",m:"margin",mt:"marginTop",mr:"marginRight",mb:"marginBottom",ml:"marginLeft",mx:"marginX",my:"marginY",p:"padding",pt:"paddingTop",pr:"paddingRight",pb:"paddingBottom",pl:"paddingLeft",px:"paddingX",py:"paddingY"},J={marginX:["marginLeft","marginRight"],marginY:["marginTop","marginBottom"],paddingX:["paddingLeft","paddingRight"],paddingY:["paddingTop","paddingBottom"],size:["width","height"]},Z={color:"colors",backgroundColor:"colors",borderColor:"colors",margin:"space",marginTop:"space",marginRight:"space",marginBottom:"space",marginLeft:"space",marginX:"space",marginY:"space",padding:"space",paddingTop:"space",paddingRight:"space",paddingBottom:"space",paddingLeft:"space",paddingX:"space",paddingY:"space",top:"space",right:"space",bottom:"space",left:"space",gridGap:"space",gridColumnGap:"space",gridRowGap:"space",gap:"space",columnGap:"space",rowGap:"space",fontFamily:"fonts",fontSize:"fontSizes",fontWeight:"fontWeights",lineHeight:"lineHeights",letterSpacing:"letterSpacings",border:"borders",borderTop:"borders",borderRight:"borders",borderBottom:"borders",borderLeft:"borders",borderWidth:"borderWidths",borderStyle:"borderStyles",borderRadius:"radii",borderTopRightRadius:"radii",borderTopLeftRadius:"radii",borderBottomRightRadius:"radii",borderBottomLeftRadius:"radii",borderTopWidth:"borderWidths",borderTopColor:"colors",borderTopStyle:"borderStyles",borderBottomWidth:"borderWidths",borderBottomColor:"colors",borderBottomStyle:"borderStyles",borderLeftWidth:"borderWidths",borderLeftColor:"colors",borderLeftStyle:"borderStyles",borderRightWidth:"borderWidths",borderRightColor:"colors",borderRightStyle:"borderStyles",outlineColor:"colors",boxShadow:"shadows",textShadow:"shadows",zIndex:"zIndices",width:"sizes",minWidth:"sizes",maxWidth:"sizes",height:"sizes",minHeight:"sizes",maxHeight:"sizes",flexBasis:"sizes",size:"sizes",fill:"colors",stroke:"colors"},Q=function(e,t){if("number"!=typeof t||t>=0)return V(e,t,t);var r=Math.abs(t),o=V(e,r,r);return"string"==typeof o?"-"+o:-1*o},ee=["margin","marginTop","marginRight","marginBottom","marginLeft","marginX","marginY","top","bottom","left","right"].reduce((function(e,t){var r;return K({},e,((r={})[t]=Q,r))}),{}),te=function e(t){return function(r){void 0===r&&(r={});var o=K({},q,{},r.theme||r),a={},n=function(e){return function(t){var r={},o=V(t,"breakpoints",X),a=[null].concat(o.map((function(e){return"@media screen and (min-width: "+e+")"})));for(var n in e){var i="function"==typeof e[n]?e[n](t):e[n];if(null!=i)if(Array.isArray(i))for(var d=0;d<i.slice(0,a.length).length;d++){var l=a[d];l?(r[l]=r[l]||{},null!=i[d]&&(r[l][n]=i[d])):r[n]=i[d]}else r[n]=i}return r}}("function"==typeof t?t(o):t)(o);for(var i in n){var d=n[i],l="function"==typeof d?d(o):d;if("variant"!==i)if(l&&"object"==typeof l)a[i]=e(l)(o);else{var c=V(Y,i,i),s=V(Z,c),f=V(o,s,V(o,c,{})),b=V(ee,c,V)(f,l,l);if(J[c])for(var g=J[c],u=0;u<g.length;u++)a[g[u]]=b;else a[c]=b}else a=K({},a,{},e(V(o,l))(o))}return a}},re=function(e){var t,r,o=e.scale,a=e.prop,n=void 0===a?"variant":a,i=e.variants,d=void 0===i?{}:i,l=e.key;(r=Object.keys(d).length?function(e,t,r){return te(x(t,e,null))(r.theme)}:function(e,t){return x(t,e,null)}).scale=o||l,r.defaults=d;var c=((t={})[n]=r,t);return v(c)},oe=re({key:"buttons"}),ae=re({key:"textStyles",prop:"textStyle"}),ne=re({key:"colorStyles",prop:"colors"}),ie=S.width,de=S.height,le=S.minWidth,ce=S.minHeight,se=S.maxWidth,fe=S.maxHeight,be=S.size,ge=S.verticalAlign,ue=S.display,pe=S.overflow,me=S.overflowX,he=S.overflowY,xe=E.opacity,ve=C.fontSize,ye=C.fontFamily,we=C.fontWeight,Be=C.lineHeight,$e=C.textAlign,ke=C.fontStyle,Se=C.letterSpacing,Te=I.alignItems,Ee=I.alignContent,Ce=I.justifyItems,Ie=I.justifyContent,Ne=I.flexWrap,Ae=I.flexDirection,ze=I.flex,Re=I.flexGrow,Le=I.flexShrink,Me=I.flexBasis,Oe=I.justifySelf,He=I.alignSelf,Pe=I.order,De=A.gridGap,We=A.gridColumnGap,Fe=A.gridRowGap,je=A.gridColumn,Ue=A.gridRow,_e=A.gridAutoFlow,Ge=A.gridAutoColumns,Ke=A.gridAutoRows,Ve=A.gridTemplateColumns,Xe=A.gridTemplateRows,qe=A.gridTemplateAreas,Ye=A.gridArea,Je=R.borderWidth,Ze=R.borderStyle,Qe=R.borderColor,et=R.borderTop,tt=R.borderRight,rt=R.borderBottom,ot=R.borderLeft,at=R.borderRadius,nt=M.backgroundImage,it=M.backgroundSize,dt=M.backgroundPosition,lt=M.backgroundRepeat,ct=H.zIndex,st=H.top,ft=H.right,bt=H.bottom,gt=H.left,ut=function(e){var t=e.prop,r=e.cssProperty,o=e.alias,a=e.key,n=e.transformValue,i=e.scale,d=e.properties,l={};return l[t]=B({properties:d,property:r||t,scale:a,defaultScale:i,transform:n}),o&&(l[o]=l[t]),v(l)},pt=Object.freeze({__proto__:null,width:ie,height:de,minWidth:le,minHeight:ce,maxWidth:se,maxHeight:fe,size:be,verticalAlign:ge,display:ue,overflow:pe,overflowX:me,overflowY:he,opacity:xe,fontSize:ve,fontFamily:ye,fontWeight:we,lineHeight:Be,textAlign:$e,fontStyle:ke,letterSpacing:Se,alignItems:Te,alignContent:Ee,justifyItems:Ce,justifyContent:Ie,flexWrap:Ne,flexDirection:Ae,flex:ze,flexGrow:Re,flexShrink:Le,flexBasis:Me,justifySelf:Oe,alignSelf:He,order:Pe,gridGap:De,gridColumnGap:We,gridRowGap:Fe,gridColumn:je,gridRow:Ue,gridAutoFlow:_e,gridAutoColumns:Ge,gridAutoRows:Ke,gridTemplateColumns:Ve,gridTemplateRows:Xe,gridTemplateAreas:qe,gridArea:Ye,borderWidth:Je,borderStyle:Ze,borderColor:Qe,borderTop:et,borderRight:tt,borderBottom:rt,borderLeft:ot,borderRadius:at,backgroundImage:nt,backgroundSize:it,backgroundPosition:dt,backgroundRepeat:lt,zIndex:ct,top:st,right:ft,bottom:bt,left:gt,style:ut,get:x,createParser:v,createStyleFunction:B,compose:k,system:$,margin:j,padding:U,space:_,color:E,layout:S,typography:C,flexbox:I,border:R,background:M,position:H,grid:A,shadow:G,boxShadow:G,textShadow:G,variant:re,buttonStyle:oe,textStyle:ae,colorStyle:ne,borders:R});const{get:mt,compose:ht,system:xt}=pt,vt=e=>{return t=e,void 0===(r=mt(l.theme,e))&&(r=null),function(e){return x(e.theme,t,r)};var t,r},yt=ht(_,E,ue),wt=ht(C,xt({whiteSpace:{property:"whiteSpace"}})),Bt=ht(R,G),$t=S,kt=H,St=I,Tt=A;function Et(){return(Et=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e}).apply(this,arguments)}const Ct=r.createGlobalStyle`
  * { box-sizing: border-box; }
  body { margin: 0; }
  table { border-collapse: collapse; }
  
  [role="button"]:focus:not(:focus-visible):not(.focus-visible),
  [role="tabpanel"][tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  button:focus:not(:focus-visible):not(.focus-visible),
  summary:focus:not(:focus-visible):not(.focus-visible),
  a:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
    box-shadow: none;
  }

  [tabindex="0"]:focus:not(:focus-visible):not(.focus-visible),
  details-dialog:focus:not(:focus-visible):not(.focus-visible) {
    outline: none;
  }
`,It=i.default.div`
  ${wt};
  ${yt};
`;function Nt(e){const{children:t,...r}=e;return require("focus-visible"),n.default.createElement(It,Et({},r,{"data-portal-root":!0}),n.default.createElement(Ct,null),t)}Nt.displayName="BaseStyles",Nt.defaultProps={color:"text.primary",fontFamily:"normal",lineHeight:"default"};var At=function(e){return function(e){return!!e&&"object"==typeof e}(e)&&!function(e){var t=Object.prototype.toString.call(e);return"[object RegExp]"===t||"[object Date]"===t||function(e){return e.$$typeof===zt}(e)}(e)};var zt="function"==typeof Symbol&&Symbol.for?Symbol.for("react.element"):60103;function Rt(e,t){return!1!==t.clone&&t.isMergeableObject(e)?Pt((r=e,Array.isArray(r)?[]:{}),e,t):e;var r}function Lt(e,t,r){return e.concat(t).map((function(e){return Rt(e,r)}))}function Mt(e){return Object.keys(e).concat(function(e){return Object.getOwnPropertySymbols?Object.getOwnPropertySymbols(e).filter((function(t){return e.propertyIsEnumerable(t)})):[]}(e))}function Ot(e,t){try{return t in e}catch(e){return!1}}function Ht(e,t,r){var o={};return r.isMergeableObject(e)&&Mt(e).forEach((function(t){o[t]=Rt(e[t],r)})),Mt(t).forEach((function(a){(function(e,t){return Ot(e,t)&&!(Object.hasOwnProperty.call(e,t)&&Object.propertyIsEnumerable.call(e,t))})(e,a)||(Ot(e,a)&&r.isMergeableObject(t[a])?o[a]=function(e,t){if(!t.customMerge)return Pt;var r=t.customMerge(e);return"function"==typeof r?r:Pt}(a,r)(e[a],t[a],r):o[a]=Rt(t[a],r))})),o}function Pt(e,t,r){(r=r||{}).arrayMerge=r.arrayMerge||Lt,r.isMergeableObject=r.isMergeableObject||At,r.cloneUnlessOtherwiseSpecified=Rt;var o=Array.isArray(t);return o===Array.isArray(e)?o?r.arrayMerge(e,t,r):Ht(e,t,r):Rt(t,r)}Pt.all=function(e,t){if(!Array.isArray(e))throw new Error("first argument should be an array");return e.reduce((function(e,r){return Pt(e,r,t)}),{})};var Dt=Pt;const Wt="light",Ft="dark",jt=n.default.createContext({setColorMode:()=>null,setDayScheme:()=>null,setNightScheme:()=>null}),Ut=({children:e,...t})=>{var o,a,i,d,c,s,f,b;const{theme:g,colorMode:u,dayScheme:p,nightScheme:m}=_t(),h=null!==(o=null!==(a=t.theme)&&void 0!==a?a:g)&&void 0!==o?o:l.theme,[x,v]=n.default.useState(null!==(i=null!==(d=t.colorMode)&&void 0!==d?d:u)&&void 0!==i?i:"day"),[y,w]=n.default.useState(null!==(c=null!==(s=t.dayScheme)&&void 0!==s?s:p)&&void 0!==c?c:Wt),[B,$]=n.default.useState(null!==(f=null!==(b=t.nightScheme)&&void 0!==b?b:m)&&void 0!==f?f:Ft),k=function(e,t){switch(e){case"auto":return t;default:return e}}(x,function(){const[e,t]=n.default.useState(Kt);return n.default.useEffect((()=>{var e,r;const o=null===(e=window)||void 0===e||null===(r=e.matchMedia)||void 0===r?void 0:r.call(e,"(prefers-color-scheme: dark)");function a(e){const r=e.matches;t(r?"night":"day")}return null==o||o.addEventListener("change",a),function(){null==o||o.removeEventListener("change",a)}}),[]),e}()),S=function(e,t,r){switch(e){case"day":return t;case"night":return r}}(k,y,B),T=n.default.useMemo((()=>function(e,t){if(!e.colorSchemes)return e;if(!e.colorSchemes[t]){console.error(`\`${t}\` scheme not defined in \`theme.colorSchemes\``);const r=Object.keys(e.colorSchemes)[0];return Dt(e,e.colorSchemes[r])}return Dt(e,e.colorSchemes[t])}(h,S)),[h,S]);return n.default.useEffect((()=>{var e,r;v(null!==(e=null!==(r=t.colorMode)&&void 0!==r?r:u)&&void 0!==e?e:"day")}),[t.colorMode,u]),n.default.useEffect((()=>{var e,r;w(null!==(e=null!==(r=t.dayScheme)&&void 0!==r?r:p)&&void 0!==e?e:Wt)}),[t.dayScheme,p]),n.default.useEffect((()=>{var e,r;$(null!==(e=null!==(r=t.nightScheme)&&void 0!==r?r:m)&&void 0!==e?e:Ft)}),[t.nightScheme,m]),n.default.createElement(jt.Provider,{value:{theme:T,colorScheme:S,colorMode:x,resolvedColorMode:k,dayScheme:y,nightScheme:B,setColorMode:v,setDayScheme:w,setNightScheme:$}},n.default.createElement(r.ThemeProvider,{theme:T},e))};function _t(){return n.default.useContext(jt)}function Gt(e,t){var r;const{colorScheme:o=""}=_t();return null!==(r=e[o])&&void 0!==r?r:t}function Kt(){var e,t,r;return"undefined"!=typeof window&&null!==(e=(t=window).matchMedia)&&void 0!==e&&null!==(r=e.call(t,"(prefers-color-scheme: dark)"))&&void 0!==r&&r.matches?"night":"day"}Ut.displayName="ThemeProvider";const Vt=e=>te(e.sx),Xt=i.default.div`
  ${yt}
  ${St}
  ${$t}
  ${Vt};
`,qt=i.default(Xt)`
  ${Bt};
  ${Vt};
`;qt.defaultProps={borderWidth:"1px",borderStyle:"solid",borderColor:"border.primary",borderRadius:2};const Yt=i.default(Xt)``;Yt.defaultProps={display:"flex"};const Jt=i.default(Xt)`
  ${Tt};
`;Jt.defaultProps={display:"grid"};const Zt=i.default(Xt)`
  ${kt};
  ${Vt};
`,Qt=n.default.forwardRef(((e,t)=>n.default.createElement(Zt,Et({},e,{position:"absolute",ref:t}))));Qt.displayName="Absolute";const er=n.default.forwardRef(((e,t)=>n.default.createElement(Zt,Et({},e,{position:"fixed",ref:t}))));er.displayName="Fixed";const tr=n.default.forwardRef(((e,t)=>n.default.createElement(Zt,Et({},e,{position:"relative",ref:t}))));tr.displayName="Relative";const rr=n.default.forwardRef(((e,t)=>n.default.createElement(Zt,Et({},e,{position:"sticky",ref:t}))));function or({ref:e,closeOnOutsideClick:r,defaultOpen:o,onClickOutside:a}){const[n,i]=t.useState(o),d=t.useRef(null),l=null!=e?e:d,c=t.useCallback((e=>{const{current:t}=l;e.target.closest("details")!==t&&(a&&a(e),e.defaultPrevented||i(!1))}),[l,i,a]);t.useEffect((()=>{if(n&&r)return document.addEventListener("click",c),()=>{document.removeEventListener("click",c)}}),[n,r,c]);const s=e=>{if(!e.defaultPrevented){const t=e.target;i(t.open)}};return{open:n,setOpen:i,getDetailsProps:()=>({onToggle:s,open:n,ref:l})}}rr.defaultProps={top:0,zIndex:1},rr.displayName="Sticky";const ar=({containerRef:e,ignoreClickRefs:r,onClickOutside:o})=>{const a=t.useCallback((t=>{(({ignoreClickRefs:e,containerRef:t,e:r})=>{var o;let a=!0;if(r instanceof MouseEvent&&r.button>0&&(a=!1),null!==(o=t.current)&&void 0!==o&&o.contains(r.target))a=!1;else if(e)for(const t of e){var n;if(null!==(n=t.current)&&void 0!==n&&n.contains(r.target)){a=!1;break}}return a})({ignoreClickRefs:r,containerRef:e,e:t})&&o(t)}),[o,e,r]);t.useEffect((()=>(document.addEventListener("mousedown",a,{capture:!0}),()=>{document.removeEventListener("mousedown",a,{capture:!0})})),[a])};function*nr(e,t={}){var r,o;const a=null!==(r=t.strict)&&void 0!==r&&r,n=null!==(o=t.onlyTabbable)&&void 0!==o&&o?dr:ir,i=document.createTreeWalker(e,NodeFilter.SHOW_ELEMENT,{acceptNode:e=>e instanceof HTMLElement&&n(e,a)?NodeFilter.FILTER_ACCEPT:NodeFilter.FILTER_SKIP});let d=null;if(!t.reverse&&n(e,a)&&(yield e),t.reverse){let e=i.lastChild();for(;e;)d=e,e=i.lastChild()}else d=i.firstChild();for(;d instanceof HTMLElement;)yield d,d=t.reverse?i.previousNode():i.nextNode();t.reverse&&n(e,a)&&(yield e)}function ir(e,t=!1){const r=["BUTTON","INPUT","SELECT","TEXTAREA","OPTGROUP","OPTION","FIELDSET"].includes(e.tagName)&&e.disabled,o=e.hidden,a=e instanceof HTMLInputElement&&"hidden"===e.type;if(r||o||a)return!1;if(t){const t=0===e.offsetWidth||0===e.offsetHeight,r=["hidden","collapse"].includes(getComputedStyle(e).visibility),o=0===e.getClientRects().length;if(t||r||o)return!1}return null!=e.getAttribute("tabindex")||!(e instanceof HTMLAnchorElement&&null==e.getAttribute("href"))&&-1!==e.tabIndex}function dr(e,t=!1){return ir(e,t)&&"-1"!==e.getAttribute("tabindex")}function lr({initialFocusRef:e,returnFocusRef:r,containerRef:o}){t.useEffect((()=>{const t=r.current;if(e&&e.current)e.current.focus();else if(o.current){const e=nr(o.current).next().value;null==e||e.focus()}return function(){null==t||t.focus()}}),[e,r,o])}const cr=[];function sr(e){if("Escape"===e.key&&!e.defaultPrevented)for(let t=cr.length-1;t>=0&&(cr[t](e),!e.defaultPrevented);--t);}const fr=(e,r=[e])=>{const o=t.useCallback(e,r);t.useEffect((()=>(0===cr.length&&document.addEventListener("keydown",sr),cr.push(o),()=>{cr.splice(cr.findIndex((e=>e===o)),1),0===cr.length&&document.removeEventListener("keydown",sr)})),[o])};function br(e){const t=n.default.useRef(null);return null!=e?e:t}const gr=({overlayRef:e,returnFocusRef:t,initialFocusRef:r,onEscape:o,ignoreClickRefs:a,onClickOutside:n})=>{const i=br(e);return lr({containerRef:i,returnFocusRef:t,initialFocusRef:r}),ar({containerRef:i,ignoreClickRefs:a,onClickOutside:n}),fr(o),{ref:i}};let ur,pr=!1;function mr(){}try{const e=Object.create({},{signal:{get(){pr=!0}}});window.addEventListener("test",mr,e),window.removeEventListener("test",mr,e)}catch(e){}function hr(){pr||(!function(){if("undefined"==typeof window)return;const e=EventTarget.prototype.addEventListener;EventTarget.prototype.addEventListener=function(t,r,o){return"object"==typeof o&&"signal"in o&&o.signal instanceof AbortSignal&&e.call(o.signal,"abort",(()=>{this.removeEventListener(t,r,o)})),e.call(this,t,r,o)}}(),pr=!0)}function xr(){return void 0===ur&&(ur=/^mac/i.test(window.navigator.platform)),ur}let vr,yr=1e4;function wr(){return"__primer_id_"+yr++}hr(),function(e){e[e.ArrowHorizontal=1]="ArrowHorizontal",e[e.ArrowVertical=2]="ArrowVertical",e[e.JK=4]="JK",e[e.HL=8]="HL",e[e.HomeAndEnd=16]="HomeAndEnd",e[e.PageUpDown=256]="PageUpDown",e[e.WS=32]="WS",e[e.AD=64]="AD",e[e.Tab=128]="Tab",e[e.ArrowAll=e.ArrowHorizontal|e.ArrowVertical]="ArrowAll",e[e.HJKL=e.HL|e.JK]="HJKL",e[e.WASD=e.WS|e.AD]="WASD",e[e.All=e.ArrowAll|e.HJKL|e.HomeAndEnd|e.PageUpDown|e.WASD|e.Tab]="All"}(vr||(vr={}));const Br={ArrowLeft:vr.ArrowHorizontal,ArrowDown:vr.ArrowVertical,ArrowUp:vr.ArrowVertical,ArrowRight:vr.ArrowHorizontal,h:vr.HL,j:vr.JK,k:vr.JK,l:vr.HL,a:vr.AD,s:vr.WS,w:vr.WS,d:vr.AD,Tab:vr.Tab,Home:vr.HomeAndEnd,End:vr.HomeAndEnd,PageUp:vr.PageUpDown,PageDown:vr.PageUpDown},$r={ArrowLeft:"previous",ArrowDown:"next",ArrowUp:"previous",ArrowRight:"next",h:"previous",j:"next",k:"previous",l:"next",a:"previous",s:"next",w:"previous",d:"next",Tab:"next",Home:"start",End:"end",PageUp:"start",PageDown:"end"};function kr(e){const t=$r[e.key];if("Tab"===e.key&&e.shiftKey)return"previous";const r=xr();if(r&&e.metaKey||!r&&e.ctrlKey){if("ArrowLeft"===e.key||"ArrowUp"===e.key)return"start";if("ArrowRight"===e.key||"ArrowDown"===e.key)return"end"}return t}const Sr=[];function Tr(e){for(const t of Sr)t(e)}function Er(e,t){var r,o,a,n;const i=[],d=new WeakMap,l=null!==(r=null==t?void 0:t.bindKeys)&&void 0!==r?r:(null!=t&&t.getNextFocusable?vr.ArrowAll:vr.ArrowVertical)|vr.HomeAndEnd,c=null!==(o=null==t?void 0:t.focusOutBehavior)&&void 0!==o?o:"stop",s=null!==(a=null==t?void 0:t.focusInStrategy)&&void 0!==a?a:"previous",f=null==t?void 0:t.activeDescendantControl,b=null==t?void 0:t.onActiveDescendantChanged;let g,u=Boolean(f);function p(){return i[0]}function m(e){const t=g;g=e,f||(t&&t!==e&&d.has(t)&&t.setAttribute("tabindex","-1"),null==e||e.setAttribute("tabindex","0"))}function h(e,t){t.id||t.setAttribute("id",wr()),g=t,null==f||f.setAttribute("aria-activedescendant",t.id),Tr(t),null==b||b(t,e===t?void 0:e)}function x(e=g){null==f||f.removeAttribute("aria-activedescendant"),u=!0,null==b||b(void 0,e),"first"===s&&(g=void 0)}function v(...e){const r=e.filter((e=>{var r,o;return null===(r=null==t||null===(o=t.focusableElementFilter)||void 0===o?void 0:o.call(t,e))||void 0===r||r}));if(0===r.length)return;const o=i.findIndex((e=>(e.compareDocumentPosition(r[0])&Node.DOCUMENT_POSITION_PRECEDING)>0));i.splice(-1===o?i.length:o,0,...r);for(const e of r)d.has(e)||d.set(e,e.getAttribute("tabindex")),e.setAttribute("tabindex","-1");g||m(p())}function y(...e){for(const t of e){const e=i.indexOf(t);if(e>=0&&(i.splice(e,1),t===g)){const e=p();m(e),f&&!u&&(e?h(t,e):x(t))}const r=d.get(t);void 0!==r&&(null===r?t.removeAttribute("tabindex"):t.setAttribute("tabindex",r),d.delete(t))}}v(...nr(e));const w=(B=e=>{if("previous"===s){const t=i.indexOf(e);t>=0&&m(i[t])}},Sr.push(B),()=>{const e=Sr.indexOf(B);Sr.splice(e,1)});var B;m(p());new MutationObserver((e=>{for(const t of e)for(const e of t.removedNodes)e instanceof HTMLElement&&y(...nr(e));for(const t of e)for(const e of t.addedNodes)e instanceof HTMLElement&&v(...nr(e))})).observe(e,{subtree:!0,childList:!0});const $=new AbortController,k=null!==(n=null==t?void 0:t.abortSignal)&&void 0!==n?n:$.signal;let S;k.addEventListener("abort",(()=>{y(...i),w()})),e.addEventListener("mousedown",(e=>{e.target instanceof HTMLElement&&e.target!==document.activeElement&&(S=i.indexOf(e.target))}),{signal:k}),f||e.addEventListener("focusin",(t=>{if(t.target instanceof HTMLElement){if(void 0!==S)S>=0&&i[S]!==g&&m(i[S]),S=void 0;else if("previous"===s)m(t.target);else if("closest"===s||"first"===s){if(t.relatedTarget instanceof Element&&!e.contains(t.relatedTarget)){const e="previous"===E?i.length-1:0,t=i[e];return void(null==t||t.focus())}m(t.target)}else if("function"==typeof s)if(t.relatedTarget instanceof Element&&!e.contains(t.relatedTarget)){const e=s(t.relatedTarget);if((e?i.indexOf(e):-1)>=0&&e instanceof HTMLElement)return void e.focus();console.warn("Element requested is not a known focusable element.")}else m(t.target);Tr(t.target)}E=void 0}),{signal:k});const T=null!=f?f:e;let E;return"closest"===s&&document.addEventListener("keydown",(e=>{"Tab"===e.key&&(E=kr(e))}),{signal:k,capture:!0}),T.addEventListener("keydown",(e=>{if(e.key in $r){const o=Br[e.key];if(!e.defaultPrevented&&(o&l)>0&&!function(e,t){const r=e.key,o=[...r].length,a=t instanceof HTMLInputElement&&"text"===t.type||t instanceof HTMLTextAreaElement;if(a&&(1===o||"Home"===r||"End"===r))return!0;if(t instanceof HTMLSelectElement){if(1===o)return!0;if("ArrowDown"===r&&xr()&&!e.metaKey)return!0;if("ArrowDown"===r&&!xr()&&e.altKey)return!0}if(t instanceof HTMLTextAreaElement&&("PageUp"===r||"PageDown"===r))return!0;if(a){const e=t,o=0===e.selectionStart&&0===e.selectionEnd,a=e.selectionStart===e.value.length&&e.selectionEnd===e.value.length;if("ArrowLeft"===r&&!o)return!0;if("ArrowRight"===r&&!a)return!0;if(e instanceof HTMLTextAreaElement){if("ArrowUp"===r&&!o)return!0;if("ArrowDown"===r&&!a)return!0}}return!1}(e,document.activeElement)){const o=kr(e);let a;if(u)a=g||p(),a&&(u=!1);else{var r;if(null!=t&&t.getNextFocusable)a=t.getNextFocusable(o,null!==(r=document.activeElement)&&void 0!==r?r:void 0,e);if(!a){const t=function(){if(!g)return 0;const e=i.indexOf(g);return-1===e?0:e}();let r=t;"previous"===o?r-=1:"start"===o?r=0:"next"===o?r+=1:r=i.length-1,r<0&&("wrap"===c&&"Tab"!==e.key?r=i.length-1:(f&&x(),r=0)),r>=i.length&&(r="wrap"===c&&"Tab"!==e.key?0:i.length-1),t!==r&&(a=i[r])}}a&&(f?h(g,a):(E=o,a.focus())),("Tab"!==e.key||a)&&e.preventDefault()}}"Escape"===e.key&&!u&&f&&x()}),{signal:k}),f&&f.addEventListener("focusout",(()=>{x()})),$}var Cr=r.css(["position:relative;display:inline-block;padding:6px 16px;font-family:inherit;font-weight:",";line-height:20px;white-space:nowrap;vertical-align:middle;cursor:pointer;user-select:none;border-radius:",";appearance:none;text-decoration:none;text-align:center;&:hover{text-decoration:none;}&:focus{outline:none;}&:disabled{cursor:default;}&:disabled svg{opacity:0.6;}"],vt("fontWeights.bold"),vt("radii.2"));const Ir=k(ve,yt,$t),Nr=re({variants:{small:{p:"4px 12px",fontSize:0},medium:{fontSize:1},large:{fontSize:2,p:"10px 20px"}}}),Ar=i.default.button.attrs((({disabled:e,onClick:t})=>({onClick:e?void 0:t})))`
  ${Cr}
  ${Nr}
`;Ar.defaultProps={variant:"medium"};const zr=i.default(Ar)`
  color: ${vt("colors.btn.text")};
  background-color: ${vt("colors.btn.bg")};
  border: 1px solid ${vt("colors.btn.border")};
  box-shadow: ${vt("shadows.btn.shadow")}, ${vt("shadows.btn.insetShadow")}};

  &:hover {
    background-color: ${vt("colors.btn.hoverBg")};
    border-color: ${vt("colors.btn.hoverBorder")};
  }

  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.focusBorder")};
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  &:active {
    background-color: ${vt("colors.btn.selectedBg")};
    box-shadow: ${vt("shadows.btn.shadowActive")};
  }

  &:disabled {
    color: ${vt("colors.text.disabled")};
    background-color: ${vt("colors.btn.bg")};
    border-color: ${vt("colors.btn.border")};
  }

  ${Ir};
  ${Vt};
`,Rr=i.default(Ar)`
  color: ${vt("colors.btn.danger.text")};
  border: 1px solid ${vt("colors.btn.border")};
  background-color: ${vt("colors.btn.bg")};
  box-shadow: ${vt("shadows.btn.shadow")};

  &:hover {
    color: ${vt("colors.btn.danger.hoverText")};
    background-color: ${vt("colors.btn.danger.hoverBg")};
    border-color: ${vt("colors.btn.danger.hoverBorder")};
    box-shadow: ${vt("shadows.btn.danger.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.danger.focusBorder")};
    box-shadow: ${vt("shadows.btn.danger.focusShadow")};
  }

  &:active {
    color: ${vt("colors.btn.danger.selectedText")};
    background-color: ${vt("colors.btn.danger.selectedBg")};
    box-shadow: ${vt("shadows.btn.danger.selectedShadow")};
    border-color: ${vt("colors.btn.danger.selectedBorder")};
  }

  &:disabled {
    color: ${vt("colors.btn.danger.disabledText")};
    background-color: ${vt("colors.btn.danger.disabledBg")};
    border-color: ${vt("colors.btn.danger.disabledBorder")};
  }

  ${Ir};
  ${Vt};
`,Lr=i.default(Xt)`
  vertical-align: middle;

  && > * {
    position: relative;
    border-right-width: 0;
    border-radius: 0;

    :first-child {
      border-top-left-radius: ${vt("radii.2")};
      border-bottom-left-radius: ${vt("radii.2")};
      margin-right: 0;
    }

    :not(:first-child) {
      margin-left: 0;
      margin-right: 0;
    }

    :last-child {
      border-right-width: 1px;
      border-top-right-radius: ${vt("radii.2")};
      border-bottom-right-radius: ${vt("radii.2")};
    }

    :focus,
    :active,
    :hover {
      border-right-width: 1px;
      + * {
        border-left-width: 0;
      }
    }

    :focus,
    :active {
      z-index: 1;
    }
  }

  ${Vt};
`;Lr.defaultProps={display:"inline-block"};const Mr=i.default(Ar)`
  color: ${vt("colors.btn.outline.text")};
  border: 1px solid ${vt("colors.btn.border")};
  background-color: ${vt("colors.btn.bg")};
  box-shadow: ${vt("shadows.btn.shadow")};

  &:hover {
    color: ${vt("colors.btn.outline.hoverText")};
    background-color: ${vt("colors.btn.outline.hoverBg")};
    border-color: ${vt("colors.btn.outline.hoverBorder")};
    box-shadow: ${vt("shadows.btn.outline.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.outline.focusBorder")};
    box-shadow: ${vt("shadows.btn.outline.focusShadow")};
  }

  &:active {
    color: ${vt("colors.btn.outline.selectedText")};
    background-color: ${vt("colors.btn.outline.selectedBg")};
    box-shadow: ${vt("shadows.btn.outline.selectedShadow")};
    border-color: ${vt("colors.btn.outline.selectedBorder")};
  }

  &:disabled {
    color: ${vt("colors.btn.outline.disabledText")};
    background-color: ${vt("colors.btn.outline.disabledBg")};
    border-color: ${vt("colors.btn.border")};
  }

  ${Ir};
  ${Vt};
`,Or=i.default(Ar)`
  color: ${vt("colors.btn.primary.text")};
  border: 1px solid ${vt("colors.btn.primary.border")};
  background-color: ${vt("colors.btn.primary.bg")};
  box-shadow: ${vt("shadows.btn.primary.shadow")};

  &:hover {
    color: ${vt("colors.btn.primary.hoverText")};
    background-color: ${vt("colors.btn.primary.hoverBg")};
    border-color: ${vt("colors.btn.primary.hoverBorder")};
    box-shadow: ${vt("shadows.btn.primary.hoverShadow")};
  }
  // focus must come before :active so that the active box shadow overrides
  &:focus {
    border-color: ${vt("colors.btn.primary.focusBorder")};
    box-shadow: ${vt("shadows.btn.primary.focusShadow")};
  }

  &:active {
    background-color: ${vt("colors.btn.primary.selectedBg")};
    box-shadow: ${vt("shadows.btn.primary.selectedShadow")};
  }

  &:disabled {
    color: ${vt("colors.btn.primary.disabledText")};
    background-color: ${vt("colors.btn.primary.disabledBg")};
    border-color: ${vt("colors.btn.primary.disabledBorder")};
  }

  ${Ir};
  ${Vt};
`,Hr=i.default(Ar)`
  color: ${vt("colors.text.link")};
  background-color: transparent;
  border: 0;
  border-radius: ${vt("radii.2")};
  box-shadow: none;

  &:disabled {
    color: ${vt("colors.text.disabled")};
  }

  &:focus {
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  ${Ir};
  ${Vt}
`,Pr=i.default.summary`
  display: inline-block;
  padding: 0;
  font-size: ${vt("fontSizes.1")};
  color: ${vt("colors.text.secondary")};
  text-decoration: none;
  white-space: nowrap;
  cursor: pointer;
  user-select: none;
  background-color: transparent;
  border: 0;
  appearance: none; // Corrects inability to style clickable input types in iOS.

  &:hover {
    text-decoration: underline;
  }

  &:disabled {
    &,
    &:hover {
      color: rgba(${vt("colors.text.secondary")}, 0.5);
      cursor: default;
    }
  }

  &:after {
    display: inline-block;
    margin-left: ${vt("space.1")};
    width: 0;
    height: 0;
    vertical-align: -2px;
    content: '';
    border: 4px solid transparent;
    border-top-color: currentcolor;
  }
  ${yt}
  ${wt}
  ${$t}
  ${Vt};
`;var Dr={small:16,medium:32,large:64};function Wr(e){var t=e["aria-label"],r=e.className,o=e.fill,a=void 0===o?"currentColor":o,n=e.size,i=e.verticalAlign,d=e.svgDataByHeight,l=Dr[n]||n,c=function(e,t){return e.map((function(e){return parseInt(e,10)})).reduce((function(e,r){return r<=t?r:e}),e[0])}(Object.keys(d),l),s=d[c].width;return{"aria-hidden":t?"false":"true","aria-label":t,role:"img",className:r,viewBox:"0 0 "+s+" "+c,width:l*(s/c),height:l,fill:a,style:{display:"inline-block",userSelect:"none",verticalAlign:i},dangerouslySetInnerHTML:{__html:d[c].path}}}var Fr=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var r=arguments[t];for(var o in r)Object.prototype.hasOwnProperty.call(r,o)&&(e[o]=r[o])}return e};function jr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M21.03 5.72a.75.75 0 010 1.06l-11.5 11.5a.75.75 0 01-1.072-.012l-5.5-5.75a.75.75 0 111.084-1.036l4.97 5.195L19.97 5.72a.75.75 0 011.06 0z"></path>'}}})))}function Ur(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M5 3.254V3.25v.005a.75.75 0 110-.005v.004zm.45 1.9a2.25 2.25 0 10-1.95.218v5.256a2.25 2.25 0 101.5 0V7.123A5.735 5.735 0 009.25 9h1.378a2.251 2.251 0 100-1.5H9.25a4.25 4.25 0 01-3.8-2.346zM12.75 9a.75.75 0 100-1.5.75.75 0 000 1.5zm-8.5 4.5a.75.75 0 100-1.5.75.75 0 000 1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.75 21a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 19.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM5.75 6.5a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM2.5 4.75a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0zM18.25 15a1.75 1.75 0 110-3.5 1.75 1.75 0 010 3.5zM15 13.25a3.25 3.25 0 106.5 0 3.25 3.25 0 00-6.5 0z"></path><path fill-rule="evenodd" d="M6.5 7.25c0 2.9 2.35 5.25 5.25 5.25h4.5V14h-4.5A6.75 6.75 0 015 7.25h1.5z"></path><path fill-rule="evenodd" d="M5.75 16.75A.75.75 0 006.5 16V8A.75.75 0 005 8v8c0 .414.336.75.75.75z"></path>'}}})))}function _r(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M7.177 3.073L9.573.677A.25.25 0 0110 .854v4.792a.25.25 0 01-.427.177L7.177 3.427a.25.25 0 010-.354zM3.75 2.5a.75.75 0 100 1.5.75.75 0 000-1.5zm-2.25.75a2.25 2.25 0 113 2.122v5.256a2.251 2.251 0 11-1.5 0V5.372A2.25 2.25 0 011.5 3.25zM11 2.5h-1V4h1a1 1 0 011 1v5.628a2.251 2.251 0 101.5 0V5A2.5 2.5 0 0011 2.5zm1 10.25a.75.75 0 111.5 0 .75.75 0 01-1.5 0zM3.75 12a.75.75 0 100 1.5.75.75 0 000-1.5z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M4.75 3a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 4.75a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zM4.75 17.5a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM1.5 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0zm17.75-1.75a1.75 1.75 0 100 3.5 1.75 1.75 0 000-3.5zM16 19.25a3.25 3.25 0 116.5 0 3.25 3.25 0 01-6.5 0z"></path><path fill-rule="evenodd" d="M4.75 7.25A.75.75 0 015.5 8v8A.75.75 0 014 16V8a.75.75 0 01.75-.75zm8.655-5.53a.75.75 0 010 1.06L12.185 4h4.065A3.75 3.75 0 0120 7.75v8.75a.75.75 0 01-1.5 0V7.75a2.25 2.25 0 00-2.25-2.25h-4.064l1.22 1.22a.75.75 0 01-1.061 1.06l-2.5-2.5a.75.75 0 010-1.06l2.5-2.5a.75.75 0 011.06 0z"></path>'}}})))}function Gr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.5 8a6.5 6.5 0 0110.65-5.003.75.75 0 00.959-1.153 8 8 0 102.592 8.33.75.75 0 10-1.444-.407A6.5 6.5 0 011.5 8zM8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path>'},24:{width:24,path:'<path d="M2.5 12c0-5.24 4.288-9.5 9.593-9.5a9.608 9.608 0 017.197 3.219.75.75 0 001.12-.998A11.108 11.108 0 0012.093 1C5.973 1 1 5.919 1 12s4.973 11 11.093 11c5.403 0 9.91-3.832 10.893-8.915a.75.75 0 10-1.472-.285c-.848 4.381-4.74 7.7-9.421 7.7C6.788 21.5 2.5 17.24 2.5 12z"></path><path d="M12 17a1 1 0 100-2 1 1 0 000 2zm0-10a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm11.28.78a.75.75 0 00-1.06-1.06l-3.47 3.47-1.47-1.47a.75.75 0 10-1.06 1.06l2 2a.75.75 0 001.06 0l4-4z"></path>'}}})))}function Kr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path>'},24:{width:24,path:'<path d="M12 7a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0112 7zm1 9a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function Vr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M1.326 1.973a1.2 1.2 0 011.49-.832c.387.112.977.307 1.575.602.586.291 1.243.71 1.7 1.296.022.027.042.056.061.084A13.22 13.22 0 018 3c.67 0 1.289.037 1.861.108l.051-.07c.457-.586 1.114-1.004 1.7-1.295a9.654 9.654 0 011.576-.602 1.2 1.2 0 011.49.832c.14.493.356 1.347.479 2.29.079.604.123 1.28.07 1.936.541.977.773 2.11.773 3.301C16 13 14.5 15 8 15s-8-2-8-5.5c0-1.034.238-2.128.795-3.117-.08-.712-.034-1.46.052-2.12.122-.943.34-1.797.479-2.29zM8 13.065c6 0 6.5-2 6-4.27C13.363 5.905 11.25 5 8 5s-5.363.904-6 3.796c-.5 2.27 0 4.27 6 4.27z"></path><path d="M4 8a1 1 0 012 0v1a1 1 0 01-2 0V8zm2.078 2.492c-.083-.264.146-.492.422-.492h3c.276 0 .505.228.422.492C9.67 11.304 8.834 12 8 12c-.834 0-1.669-.696-1.922-1.508zM10 8a1 1 0 112 0v1a1 1 0 11-2 0V8z"></path>'},24:{width:24,path:'<path d="M7.75 11c-.69 0-1.25.56-1.25 1.25v1.5a1.25 1.25 0 102.5 0v-1.5C9 11.56 8.44 11 7.75 11zm1.27 4.5a.469.469 0 01.48-.5h5a.47.47 0 01.48.5c-.116 1.316-.759 2.5-2.98 2.5s-2.864-1.184-2.98-2.5zm7.23-4.5c-.69 0-1.25.56-1.25 1.25v1.5a1.25 1.25 0 102.5 0v-1.5c0-.69-.56-1.25-1.25-1.25z"></path><path fill-rule="evenodd" d="M21.255 3.82a1.725 1.725 0 00-2.141-1.195c-.557.16-1.406.44-2.264.866-.78.386-1.647.93-2.293 1.677A18.442 18.442 0 0012 5c-.93 0-1.784.059-2.569.17-.645-.74-1.505-1.28-2.28-1.664a13.876 13.876 0 00-2.265-.866 1.725 1.725 0 00-2.141 1.196 23.645 23.645 0 00-.69 3.292c-.125.97-.191 2.07-.066 3.112C1.254 11.882 1 13.734 1 15.527 1 19.915 3.13 23 12 23c8.87 0 11-3.053 11-7.473 0-1.794-.255-3.647-.99-5.29.127-1.046.06-2.15-.066-3.125a23.652 23.652 0 00-.689-3.292zM20.5 14c.5 3.5-1.5 6.5-8.5 6.5s-9-3-8.5-6.5c.583-4 3-6 8.5-6s7.928 2 8.5 6z"></path>'}}})))}function Xr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zM6.92 6.085c.081-.16.19-.299.34-.398.145-.097.371-.187.74-.187.28 0 .553.087.738.225A.613.613 0 019 6.25c0 .177-.04.264-.077.318a.956.956 0 01-.277.245c-.076.051-.158.1-.258.161l-.007.004a7.728 7.728 0 00-.313.195 2.416 2.416 0 00-.692.661.75.75 0 001.248.832.956.956 0 01.276-.245 6.3 6.3 0 01.26-.16l.006-.004c.093-.057.204-.123.313-.195.222-.149.487-.355.692-.662.214-.32.329-.702.329-1.15 0-.76-.36-1.348-.863-1.725A2.76 2.76 0 008 4c-.631 0-1.155.16-1.572.438-.413.276-.68.638-.849.977a.75.75 0 101.342.67z"></path>'},24:{width:24,path:'<path d="M10.97 8.265a1.45 1.45 0 00-.487.57.75.75 0 01-1.341-.67c.2-.402.513-.826.997-1.148C10.627 6.69 11.244 6.5 12 6.5c.658 0 1.369.195 1.934.619a2.45 2.45 0 011.004 2.006c0 1.033-.513 1.72-1.027 2.215-.19.183-.399.358-.579.508l-.147.123a4.329 4.329 0 00-.435.409v1.37a.75.75 0 11-1.5 0v-1.473c0-.237.067-.504.247-.736.22-.28.486-.517.718-.714l.183-.153.001-.001c.172-.143.324-.27.47-.412.368-.355.569-.676.569-1.136a.953.953 0 00-.404-.806C12.766 8.118 12.384 8 12 8c-.494 0-.814.121-1.03.265zM13 17a1 1 0 11-2 0 1 1 0 012 0z"></path><path fill-rule="evenodd" d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM2.5 12a9.5 9.5 0 1119 0 9.5 9.5 0 01-19 0z"></path>'}}})))}function qr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"></path>'},24:{width:24,path:'<path d="M11.646 15.146L5.854 9.354a.5.5 0 01.353-.854h11.586a.5.5 0 01.353.854l-5.793 5.792a.5.5 0 01-.707 0z"></path>'}}})))}function Yr(e){return n.default.createElement("svg",Wr(Fr({},e,{svgDataByHeight:{16:{width:16,path:'<path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"></path>'},24:{width:24,path:'<path fill-rule="evenodd" d="M5.72 5.72a.75.75 0 011.06 0L12 10.94l5.22-5.22a.75.75 0 111.06 1.06L13.06 12l5.22 5.22a.75.75 0 11-1.06 1.06L12 13.06l-5.22 5.22a.75.75 0 01-1.06-1.06L10.94 12 5.72 6.78a.75.75 0 010-1.06z"></path>'}}})))}jr.defaultProps={className:"octicon octicon-check",size:16,verticalAlign:"text-bottom"},Ur.defaultProps={className:"octicon octicon-git-merge",size:16,verticalAlign:"text-bottom"},_r.defaultProps={className:"octicon octicon-git-pull-request",size:16,verticalAlign:"text-bottom"},Gr.defaultProps={className:"octicon octicon-issue-closed",size:16,verticalAlign:"text-bottom"},Kr.defaultProps={className:"octicon octicon-issue-opened",size:16,verticalAlign:"text-bottom"},Vr.defaultProps={className:"octicon octicon-octoface",size:16,verticalAlign:"text-bottom"},Xr.defaultProps={className:"octicon octicon-question",size:16,verticalAlign:"text-bottom"},qr.defaultProps={className:"octicon octicon-triangle-down",size:16,verticalAlign:"text-bottom"},Yr.defaultProps={className:"octicon octicon-x",size:16,verticalAlign:"text-bottom"};const Jr=i.default.button`
  border: none;
  padding: 0;
  background: transparent;
  outline: none;
  cursor: pointer;
  border-radius: ${vt("radii.2")};
  color: ${vt("colors.text.secondary")};

  &:focus {
    box-shadow: ${vt("shadows.btn.focusShadow")};
  }

  &:hover {
    color: ${vt("colors.text.link")};
  }
  ${yt};
  ${$t};
  ${Vt};
`,Zr=t.forwardRef(((e,t)=>n.default.createElement(Jr,Et({ref:t,"aria-label":"Close"},e),n.default.createElement(Yr,null)))),Qr={"outside-top":["outside-bottom","outside-right","outside-left","outside-bottom"],"outside-bottom":["outside-top","outside-right","outside-left","outside-bottom"],"outside-left":["outside-right","outside-bottom","outside-top","outside-bottom"],"outside-right":["outside-left","outside-bottom","outside-top","outside-bottom"]};function eo(e,t,r={}){const o=function(e){let t=e.parentNode;for(;null!==t;){if(t instanceof HTMLElement&&"static"!==getComputedStyle(t).position)return t;t=t.parentNode}return document.body}(e),a=function(e){let t=e;for(;null!==t&&t!==document.body;){if("visible"!==getComputedStyle(t).overflow)break;t=t.parentNode}const r=t!==document.body&&t instanceof HTMLElement?t:document.body,o=r.getBoundingClientRect(),a=getComputedStyle(r),[n,i,d,l]=[a.borderTopWidth,a.borderLeftWidth,a.borderRightWidth,a.borderBottomWidth].map((e=>parseInt(e,10)||0));return{top:o.top+n,left:o.left+i,width:o.width-d-i,height:Math.max(o.height-n-l,r===document.body?window.innerHeight:-1/0)}}(o),n=getComputedStyle(o),i=o.getBoundingClientRect(),[d,l]=[n.borderTopWidth,n.borderLeftWidth].map((e=>parseInt(e,10)||0));return function(e,t,r,o,{side:a,align:n,allowOutOfBounds:i,anchorOffset:d,alignmentOffset:l}){const c={top:e.top-t.top,left:e.left-t.left,width:e.width,height:e.height};let s=io(r,o,a,n,d,l);if(s.top-=t.top,s.left-=t.left,!i){const i=Qr[a];let f=0;if(i){let e=a;for(;f<i.length&&lo(e,s,c,r);){const a=i[f++];e=a,s=io(r,o,a,n,d,l),s.top-=t.top,s.left-=t.left}}s.top<c.top&&(s.top=c.top),s.left<c.left&&(s.left=c.left),s.left+r.width>e.width+c.left&&(s.left=e.width+c.left-r.width),i&&f<i.length&&s.top+r.height>e.height+c.top&&(s.top=e.height+c.top-r.height)}return s}(a,{top:i.top+d,left:i.left+l},e.getBoundingClientRect(),t instanceof Element?t.getBoundingClientRect():t,function(e={}){var t,r,o,a,n;const i=null!==(t=e.side)&&void 0!==t?t:to,d=null!==(r=e.align)&&void 0!==r?r:ro;return{side:i,align:d,anchorOffset:null!==(o=e.anchorOffset)&&void 0!==o?o:"inside-center"===i?0:oo,alignmentOffset:null!==(a=e.alignmentOffset)&&void 0!==a?a:"center"!==d&&i.startsWith("inside")?ao:0,allowOutOfBounds:null!==(n=e.allowOutOfBounds)&&void 0!==n?n:no}}(r))}const to="outside-bottom",ro="start",oo=4,ao=4,no=!1;function io(e,t,r,o,a,n){const i=t.left+t.width,d=t.top+t.height;let l=-1,c=-1;return"outside-top"===r?l=t.top-a-e.height:"outside-bottom"===r?l=d+a:"outside-left"===r?c=t.left-a-e.width:"outside-right"===r&&(c=i+a),"outside-top"!==r&&"outside-bottom"!==r||(c="start"===o?t.left+n:"center"===o?t.left-(e.width-t.width)/2+n:i-e.width-n),"outside-left"!==r&&"outside-right"!==r||(l="start"===o?t.top+n:"center"===o?t.top-(e.height-t.height)/2+n:d-e.height-n),"inside-top"===r?l=t.top+a:"inside-bottom"===r?l=d-a-e.height:"inside-left"===r?c=t.left+a:"inside-right"===r?c=i-a-e.width:"inside-center"===r&&(c=(i+t.left)/2-e.width/2+a),"inside-top"===r||"inside-bottom"===r?c="start"===o?t.left+n:"center"===o?t.left-(e.width-t.width)/2+n:i-e.width-n:"inside-left"!==r&&"inside-right"!==r&&"inside-center"!==r||(l="start"===o?t.top+n:"center"===o?t.top-(e.height-t.height)/2+n:d-e.height-n),{top:l,left:c}}function lo(e,t,r,o){return"outside-top"===e||"outside-bottom"===e?t.top<r.top||t.top+o.height>r.height+r.top:t.left<r.left||t.left+o.width>r.width+r.left}hr();const co=[];let so;function fo(e,t=!1){return nr(e,{reverse:t,strict:!0,onlyTabbable:!0}).next().value}function bo(e,t,r){const o=new AbortController,a=null!=r?r:o.signal;let n;function i(r){if(r instanceof HTMLElement&&document.contains(e)){if(e.contains(r))return void(n=r);if(n&&dr(n)&&e.contains(n))return void n.focus();{const o=t&&e.contains(t)?t:fo(e);if(o)return void o.focus();r.blur()}}}e.setAttribute("data-focus-trap","active");const d=function(e){const t=new AbortController;return e.addEventListener("abort",(()=>{t.abort()})),t}(a);if(e.addEventListener("keydown",(t=>{if("Tab"!==t.key||t.defaultPrevented)return;const{target:r}=t,o=fo(e),a=fo(e,!0);r===o&&t.shiftKey?(t.preventDefault(),null==a||a.focus()):r!==a||t.shiftKey||(t.preventDefault(),null==o||o.focus())}),{signal:d.signal}),so){const e=so;so.container.setAttribute("data-focus-trap","suspended"),so.controller.abort(),co.push(e)}d.signal.addEventListener("abort",(()=>{so=void 0})),a.addEventListener("abort",(()=>{e.removeAttribute("data-focus-trap");const t=co.findIndex((t=>t.container===e));t>=0&&co.splice(t,1),function(){const e=co.pop();e&&bo(e.container,e.initialFocus,e.originalSignal)}()})),document.addEventListener("focus",(e=>{i(e.target)}),{signal:d.signal,capture:!0}),i(document.activeElement),so={container:e,controller:d,initialFocus:t,originalSignal:a};const l=co.findIndex((t=>t.container===e));if(l>=0&&co.splice(l,1),!r)return o}function go(e,t=[]){const r=br(null==e?void 0:e.containerRef),o=br(null==e?void 0:e.initialFocusRef),a=null==e?void 0:e.disabled,i=n.default.useRef(),d=n.default.useRef(null);function l(){var t;null===(t=i.current)||void 0===t||t.abort(),null!=e&&e.restoreFocusOnCleanUp&&d.current instanceof HTMLElement&&(d.current.focus(),d.current=null)}return d.current||null!=e&&e.disabled||(d.current=document.activeElement),n.default.useEffect((()=>{if(r.current instanceof HTMLElement){var e;if(!a)return i.current=bo(r.current,null!==(e=o.current)&&void 0!==e?e:void 0),()=>{l()};l()}}),[r,o,a,...t]),{containerRef:r,initialFocusRef:o}}function uo({icon:e,...t}){return n.default.createElement(e,t)}uo.displayName="Octicon";const po=i.default(uo)`
  ${yt}
  ${Vt}
`;function mo(e={},r=[]){const o=br(e.containerRef),a=!!e.activeDescendantFocus,i=br("boolean"!=typeof e.activeDescendantFocus&&e.activeDescendantFocus?e.activeDescendantFocus:void 0),d=e.disabled,l=n.default.useRef();return t.useEffect((()=>{if(o.current instanceof HTMLElement&&(!a||i.current instanceof HTMLElement)){if(!d){var t;const r={...e,activeDescendantControl:null!==(t=i.current)&&void 0!==t?t:void 0};return l.current=Er(o.current,r),()=>{var e;null===(e=l.current)||void 0===e||e.abort()}}var r;null===(r=l.current)||void 0===r||r.abort()}}),[d,...r]),{containerRef:o,activeDescendantControlRef:i}}po.defaultProps={size:16};const ho="__primerPortalRoot__",xo="__default__",vo={};function yo(){const e=vo.__default__;if(!e||!document.body.contains(e)){let e=document.getElementById(ho);if(!(e instanceof Element)){e=document.createElement("div"),e.setAttribute("id",ho);const t=document.querySelector("[data-portal-root]");t?t.appendChild(e):document.body.appendChild(e)}!function(e,t="__default__"){vo[t]=e}(e)}}const wo=({children:e,onMount:t,containerName:r})=>{const a=document.createElement("div");a.style.position="relative",a.style.zIndex="1";const i=n.default.useRef(a);return n.default.useLayoutEffect((()=>{let e=r;void 0===e&&(e=xo,yo());const o=vo[e];if(!o)throw new Error(`Portal container '${r}' is not yet registered. Container must be registered with registerPortal before use.`);const a=i.current;return o.appendChild(a),null==t||t(),()=>{o.removeChild(a)}}),[i]),o.createPortal(e,i.current)};function Bo(...e){const r=t.useRef(null);return n.default.useEffect((()=>{function t(t=null){for(const r of e){if(!r)return;"function"==typeof r?r(t):r.current=t}}return t(r.current),()=>{t(r.current)}}),[...e,r.current]),r}const $o="200ms",ko=i.default("div")`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  animation: dialog-backdrop-appear ${$o} ${vt("animation.easeOutCubic")};

  @keyframes dialog-backdrop-appear {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`,So={small:"480px",large:"640px",auto:"auto"},To={small:"296px",medium:"320px",large:"480px",xlarge:"640px"},Eo=i.default.div`
  display: flex;
  flex-direction: column;
  background-color: ${vt("colors.bg.overlay")};
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
  min-width: 296px;
  max-width: calc(100vw - 64px);
  max-height: calc(100vh - 64px);
  width: ${e=>{var t;return To[null!==(t=e.width)&&void 0!==t?t:"xlarge"]}};
  height: ${e=>{var t;return So[null!==(t=e.height)&&void 0!==t?t:"auto"]}};
  border-radius: 12px;
  opacity: 1;
  animation: overlay--dialog-appear ${$o} ${vt("animation.easeOutCubic")};

  @keyframes overlay--dialog-appear {
    0% {
      opacity: 0;
      transform: scale(0.5);
    }
    100% {
      opacity: 1;
      transform: scale(1);
    }
  }

  ${yt};
  ${kt};
  ${Vt};
`,Co=({dialogLabelId:e,title:r,subtitle:o,dialogDescriptionId:a,onClose:i})=>{const d=t.useCallback((()=>{i("close-button")}),[i]);return n.default.createElement(Wo.Header,null,n.default.createElement(Yt,null,n.default.createElement(Yt,{px:2,py:"6px",flexDirection:"column",flexGrow:1},n.default.createElement(Wo.Title,{id:e},null!=r?r:"Dialog"),o&&n.default.createElement(Wo.Subtitle,{id:a},o)),n.default.createElement(Wo.CloseButton,{onClose:d})))};Co.displayName="DefaultHeader";const Io=({children:e})=>n.default.createElement(Wo.Body,null,e);Io.displayName="DefaultBody";const No=({footerButtons:e})=>{const{containerRef:t}=mo({bindKeys:vr.ArrowHorizontal|vr.Tab,focusInStrategy:"closest"});return e?n.default.createElement(Wo.Footer,{ref:t},n.default.createElement(Wo.Buttons,{buttons:e})):null},Ao=n.default.forwardRef(((e,r)=>{const{title:o="Dialog",subtitle:a="",renderHeader:i,renderBody:d,renderFooter:l,onClose:c,role:s="dialog",width:f="xlarge",height:b="auto"}=e,g=wr(),u=wr(),p={...e,title:o,subtitle:a,role:s,dialogLabelId:g,dialogDescriptionId:u},m=t.useRef(null),h=Bo(m,r),x=t.useRef(null);go({containerRef:m,restoreFocusOnCleanUp:!0}),fr((e=>{c("escape"),e.preventDefault()}),[c]);const v=(null!=i?i:Co)(p),y=(null!=d?d:Io)(p),w=(null!=l?l:No)(p);return n.default.createElement(n.default.Fragment,null,n.default.createElement(wo,null,n.default.createElement(ko,{ref:x},n.default.createElement(Eo,{width:f,height:b,ref:h,role:s,"aria-labelledby":g,"aria-describedby":u},v,y,w))))}));Ao.displayName="Dialog";const zo=i.default(Xt).attrs({as:"header"})`
  box-shadow: 0 1px 0 ${vt("colors.border.overlay")};
  padding: ${vt("space.2")};
  z-index: 1;
  flex-shrink: 0;
`,Ro=i.default(Xt)`
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
`,Lo=i.default(Xt)`
  font-size: ${vt("fontSizes.0")};
  margin-top: ${vt("space.1")};
  color: ${vt("colors.text.tertiary")};
`,Mo=i.default(Xt)`
  flex-grow: 1;
  overflow: auto;
  padding: ${vt("space.3")};
`,Oo=i.default(Xt).attrs({as:"footer"})`
  box-shadow: 0 -1px 0 ${vt("colors.border.overlay")};
  padding: ${vt("space.3")};
  display: flex;
  flex-flow: wrap;
  justify-content: flex-end;
  z-index: 1;
  flex-shrink: 0;

  button {
    margin-left: ${vt("space.1")};
    &:first-child {
      margin-left: 0;
    }
  }
`,Ho={normal:zr,primary:Or,danger:Rr},Po=i.default(zr)`
  border-radius: 4px;
  background: transparent;
  border: 0;
  vertical-align: middle;
  color: ${vt("colors.text.secondary")};
  padding: ${vt("space.2")};
  align-self: flex-start;
  line-height: normal;
  box-shadow: none;
`,Do=({onClose:e})=>n.default.createElement(Po,{"aria-label":"Close",onClick:e},n.default.createElement(po,{icon:Yr}));Do.displayName="CloseButton";const Wo=Object.assign(Ao,{Header:zo,Title:Ro,Subtitle:Lo,Body:Mo,Footer:Oo,Buttons:({buttons:e})=>{const r=t.useRef(null);let o=0;const[a,i]=t.useState(0);return t.useEffect((()=>{var e;1===a?null===(e=r.current)||void 0===e||e.focus():i(a+1)}),[a]),n.default.createElement(n.default.Fragment,null,e.map(((e,t)=>{const{content:a,buttonType:i="normal",autoFocus:d=!1,...l}=e,c=Ho[i];return n.default.createElement(c,Et({key:t},l,{ref:d&&0===o?(o++,r):null}),a)})))},CloseButton:Do}),Fo=i.default.header`
  padding: ${vt("space.2")};
  display: flex;
  flex-direction: row;
`,jo=i.default(Xt)`
  font-size: ${vt("fontSizes.3")};
  font-weight: ${vt("fontWeights.bold")};
  padding: 6px ${vt("space.2")};
  flex-grow: 1;
`,Uo=({title:e,onClose:r,dialogLabelId:o})=>{const a=t.useCallback((()=>{r("close-button")}),[r]);return n.default.createElement(Fo,null,n.default.createElement(jo,{id:o},e),n.default.createElement(Wo.CloseButton,{onClose:a}))};Uo.displayName="ConfirmationHeader";const _o=i.default(Xt)`
  font-size: ${vt("fontSizes.1")};
  padding: 0 ${vt("space.3")} ${vt("space.3")} ${vt("space.3")};
  color: ${vt("colors.text.tertiary")};
  flex-grow: 1;
`,Go=({children:e})=>n.default.createElement(_o,null,e);Go.displayName="ConfirmationBody";const Ko=i.default(Xt)`
  display: flex;
  flex-shrink: 0;
  button {
    font-size: ${vt("fontSizes.1")};
    flex: 1 1 0;
    border-bottom: 0;
    margin: 0;
    border-right: 0;
    border-radius: 12px;
    &:first-child {
      border-left: 0;
    }
    &:focus {
      // this is a bit of a hack to get the focus outlines to show up
      z-index: 1;
    }
  }
  button:first-of-type:not(:last-of-type) {
    // All except border-bottom-left-radius, which should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
  button:last-of-type:not(:first-of-type) {
    // All except border-bottom-right-radius, which should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-left-radius: 0;
  }
  button:first-of-type:last-of-type {
    // When the footer contains a single button, it should follow the curvature of the container
    border-top-left-radius: 0;
    border-top-right-radius: 0;
  }
  button:not(:first-of-type):not(:last-of-type) {
    border-radius: 0;
  }
`,Vo=({footerButtons:e})=>{const{containerRef:t}=mo({bindKeys:vr.ArrowHorizontal|vr.Tab,focusInStrategy:"closest"});return n.default.createElement(Ko,{ref:t},n.default.createElement(Wo.Buttons,{buttons:null!=e?e:[]}))};Vo.displayName="ConfirmationFooter";const Xo=e=>{const{onClose:r,title:o,cancelButtonContent:a="Cancel",confirmButtonContent:i="OK",confirmButtonType:d="normal",children:l}=e,c=[{content:a,onClick:t.useCallback((()=>{r("cancel")}),[r]),autoFocus:!0,variant:"large"},{content:i,buttonType:d,variant:"large",onClick:t.useCallback((()=>{r("confirm")}),[r])}];return n.default.createElement(Wo,{onClose:r,title:o,footerButtons:c,role:"alertdialog",width:"medium",renderHeader:Uo,renderBody:Go,renderFooter:Vo},l)};Xo.displayName="ConfirmationDialog";const qo=i.default.div`
   {
    /* 6px vertical padding + 20px line height = 32px total height
     *
     * TODO: When rem-based spacing on a 4px scale lands, replace
     * hardcoded '6px' with 'calc((${vt("space.s32")} - ${vt("space.20")}) / 2)'.
     */
  }
  padding: 6px ${vt("space.3")};
  font-size: ${vt("fontSizes.0")};
  font-weight: ${vt("fontWeights.bold")};
  color: ${vt("colors.text.secondary")};

  ${({variant:e})=>"filled"===e&&r.css(["background:",";margin:"," 0;border-top:1px solid ",";border-bottom:1px solid ",";&:first-child{margin-top:0;}"],vt("colors.bg.tertiary"),vt("space.2"),vt("colors.border.tertiary"),vt("colors.border.tertiary"))}

  ${Vt}
`;function Yo({variant:e="subtle",title:t,auxiliaryText:r,children:o,...a}){return n.default.createElement(qo,Et({role:"heading",variant:e},a),t,r&&n.default.createElement("span",null,"auxiliaryText"))}Yo.displayName="Header";const Jo=i.default.div`
  ${Vt}
`;function Zo({header:e,items:t,...r}){return n.default.createElement(Jo,r,e&&n.default.createElement(Yo,e),t)}Zo.displayName="Group";const Qo=i.default.div`
  ${wt}
  ${yt}
  display: ${e=>e.inline?"inline-block":"inherit"};
  overflow: hidden;
  text-overflow: ellipsis;
  vertical-align: ${e=>e.inline?"top":"initial"};
  white-space: nowrap;
  ${se}
  ${e=>e.expandable?"&:hover { max-width: 10000px; }":""}
  ${Vt};
`;Qo.defaultProps={expandable:!1,inline:!1,maxWidth:125};const ea=i.default.div`
  height: 1px;
  background: ${vt("colors.selectMenu.borderSecondary")};
  margin-top: calc(${vt("space.2")} - 1px);
  margin-bottom: ${vt("space.2")};
`;function ta(){return n.default.createElement(ea,null)}ta.displayName="Divider",ta.renderItem=ta;const ra={default:{hover:{light:"rgba(46, 77, 108, 0.06)",dark:"rgba(201, 206, 212, 0.12)",dark_dimmed:"rgba(201, 206, 212, 0.12)"},focus:{light:"rgba(54, 77, 100, 0.16)",dark:"rgba(201, 206, 212, 0.24)",dark_dimmed:"rgba(201, 206, 212, 0.24)"}},danger:{hover:{light:"rgba(234, 74, 90, 0.08)",dark:"rgba(248, 81, 73, 0.16)",dark_dimmed:"rgba(248, 81, 73, 0.16)"},focus:{light:"rgba(234, 74, 90, 0.14)",dark:"rgba(248, 81, 73, 0.24)",dark_dimmed:"rgba(248, 81, 73, 0.24)"}}},oa=`${wr()}active-descendant`,aa=(e="default",t)=>{if(t)return{color:vt("colors.text.disabled"),iconColor:vt("colors.text.disabled"),annotationColor:vt("colors.text.disabled"),hoverCursor:"default"};switch(e){case"danger":return{color:vt("colors.text.danger"),iconColor:vt("colors.icon.danger"),annotationColor:vt("colors.text.disabled"),hoverCursor:"pointer"};default:return{color:"inherit",iconColor:vt("colors.text.secondary"),annotationColor:vt("colors.text.secondary"),hoverCursor:"pointer"}}},na=i.default.div`
  display: flex;
  min-width: 0;
  flex-grow: 1;
  position: relative;
`,ia=i.default.div`
  /* 6px vertical padding + 20px line height = 32px total height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '6px' with 'calc((${vt("space.s32")} - ${vt("space.20")}) / 2)'.
   */
  padding: 6px ${vt("space.2")};
  display: flex;
  border-radius: ${vt("radii.2")};
  color: ${({variant:e,item:t})=>aa(e,null==t?void 0:t.disabled).color};

  @media (hover: hover) and (pointer: fine) {
    :hover {
      background: ${({hoverBackground:e})=>e};
      cursor: ${({variant:e,item:t})=>aa(e,null==t?void 0:t.disabled).hoverCursor};
    }
  }

  // Item dividers
  :not(:first-of-type):not(${ea} + &):not(${qo} + &) {
    margin-top: ${({showDivider:e})=>e?"1px":"0"};

    ${na}::before {
      content: ' ';
      display: block;
      position: absolute;
      width: 100%;
      top: -7px;
      // NB: This 'get' wont execute if its moved into the arrow function below.
      border: 0 solid ${vt("colors.selectMenu.borderSecondary")};
      border-top-width: ${({showDivider:e})=>e?"1px":"0"};
    }
  }

  // Item dividers should not be visible:
  // - above Hovered
  &:hover ${na}::before,
  // - below Hovered
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :hover + * ${na}::before,
  // - above Focused
  &:focus ${na}::before,
  // - below Focused
  // '*' instead of '&' because '&' maps to separate class names depending on 'variant'
  :focus + * ${na}::before,
  // - above Active Descendent
  &.${oa} ${na}::before,
  // - below Active Descendent
  .${oa} + & ${na}::before {
    // '!important' because all the ':not's above give higher specificity
    border-color: transparent !important;
  }

  // Focused OR Active Descendant
  &:focus,
  &.${oa} {
    background: ${({focusBackground:e})=>e};
    outline: none;
  }

  ${Vt}
`,da=i.default.div`
  display: flex;
  min-width: 0;
  flex-grow: 1;
  flex-direction: ${({descriptionVariant:e})=>"inline"===e?"row":"column"};
  align-items: baseline;
`,la=i.default.div`
  // Match visual height to adjacent text line height.
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '20px' with '${vt("space.s20")}'.
  height: 20px;
  width: ${vt("space.3")};
  flex-shrink: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-right: ${vt("space.2")};
`,ca=i.default(la)`
  svg {
    fill: ${({variant:e,disabled:t})=>aa(e,t).iconColor};
    font-size: ${vt("fontSizes.0")};
  }
`,sa=i.default(ca)``,fa=i.default(ca)`
  color: ${({variant:e,disabled:t})=>aa(e,t).annotationColor}};
  margin-left: ${vt("space.2")};
  margin-right: 0;
  div:nth-child(2) {
    margin-left: ${vt("space.2")};
  }
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`,ba=i.default.span`
  color: ${vt("colors.text.secondary")};
  font-size: ${vt("fontSizes.0")};
  // TODO: When rem-based spacing on a 4px scale lands, replace
  // hardcoded '16px' with '${vt("lh-12")}'.
  line-height: 16px;
  margin-left: ${({descriptionVariant:e})=>"inline"===e?vt("space.2"):0};
  min-width: 0;
  flex-grow: 1;
  flex-basis: ${({descriptionVariant:e})=>"inline"===e?0:"auto"};
`,ga=i.default.input`
  pointer-events: none;
`;function ua(e){const{text:r,description:o,descriptionVariant:a="inline",selected:i,selectionVariant:d,leadingVisual:l,trailingIcon:c,trailingText:s,variant:f="default",showDivider:b,disabled:g,onAction:u,onKeyPress:p,children:m,onClick:h,id:x,...v}=e,y=t.useCallback((t=>{if(g)return;null==p||p(t);t.target instanceof HTMLInputElement&&"checkbox"===t.target.type&&" "===t.key||!t.defaultPrevented&&[" ","Enter"].includes(t.key)&&(null==u||u(e,t))}),[u,g,e,p]),w=t.useCallback((t=>{g||(null==h||h(t),t.defaultPrevented||null==u||u(e,t))}),[u,g,e,h]),B=ra[f],$=Gt(B.hover,"inherit"),k=Gt(B.focus,"inherit"),{theme:S}=_t();return n.default.createElement(ia,Et({tabIndex:g?void 0:-1,variant:f,showDivider:b,"aria-selected":i},v,{"data-id":x,onKeyPress:y,onClick:w,hoverBackground:g?"inherit":$,focusBackground:g?"inherit":k}),!!i===i&&n.default.createElement(la,null,"multiple"===d?n.default.createElement(n.default.Fragment,null,n.default.createElement(ga,{disabled:g,tabIndex:-1,type:"checkbox",checked:i,"aria-label":r,readOnly:!0,"aria-readonly":"false"})):i&&n.default.createElement(jr,{fill:null==S?void 0:S.colors.text.primary})),l&&n.default.createElement(sa,{variant:f,disabled:g},n.default.createElement(l,null)),n.default.createElement(na,null,m,(r||o)&&n.default.createElement(da,{descriptionVariant:a},r&&n.default.createElement("div",null,r),o&&n.default.createElement(ba,{descriptionVariant:a},"block"===a?o:n.default.createElement(Qo,{title:o,inline:!0,maxWidth:"100%"},o))),(c||s)&&n.default.createElement(fa,{variant:f,disabled:g},s&&n.default.createElement("div",null,s),c&&n.default.createElement("div",null,n.default.createElement(c,null)))))}ua.displayName="Item";const pa=i.default.div`
  font-size: ${vt("fontSizes.1")};
  /* 14px font-size * 1.428571429 = 20px line height
   *
   * TODO: When rem-based spacing on a 4px scale lands, replace
   * hardcoded '20px'
   */
  line-height: 20px;
`;function ma(e){const{firstGroupStyle:t,lastGroupStyle:r,headerStyle:o,itemStyle:a}=function(e="inset"){switch(e){case"full":return{headerStyle:{paddingX:vt("space.2")},itemStyle:{borderRadius:0}};default:return{firstGroupStyle:{marginTop:vt("space.2")},lastGroupStyle:{marginBottom:vt("space.2")},itemStyle:{marginX:vt("space.2")}}}}(e.variant),i=(t,r,o)=>{var i,d,l;const c="renderItem"in t&&t.renderItem||e.renderItem||ua,s=null!==(i=null!==(d=t.key)&&void 0!==d?d:null===(l=t.id)||void 0===l?void 0:l.toString())&&void 0!==i?i:o.toString();return n.default.createElement(c,Et({showDivider:e.showItemDividers,selectionVariant:e.selectionVariant},t,{key:s,sx:{...a,...t.sx},item:r}))};let d=[];if(function(e){return"groupMetadata"in e}(e)){const t=e.groupMetadata.reduce(((e,t)=>e.set(t.groupId,t)),new Map);for(const r of e.items){var l,c,s;const e=t.get(r.groupId),o=null!==(l=null==e||null===(c=e.items)||void 0===c?void 0:c.length)&&void 0!==l?l:0;t.set(r.groupId,{...e,items:[...null!==(s=null==e?void 0:e.items)&&void 0!==s?s:[],i({showDivider:null==e?void 0:e.showItemDividers,...e&&"renderItem"in e&&{renderItem:e.renderItem},...r},r,o)]})}d=[...t.values()]}else d=[{items:e.items.map(((e,t)=>i(e,e,t))),groupId:"0"}];return n.default.createElement(pa,e,d.map((({header:a,...i},l)=>{const c="filled"===(null==a?void 0:a.variant),s=l>0&&!c;return n.default.createElement(n.default.Fragment,{key:i.groupId},s?n.default.createElement(ta,{key:`${i.groupId}-divider`}):null,(t=>{var r;const o=(null!==(r="renderGroup"in t&&t.renderGroup)&&void 0!==r?r:e.renderGroup)||Zo;return n.default.createElement(o,Et({},t,{key:t.groupId}))})({sx:{...0===l&&t,...l===d.length-1&&r,...l>0&&!s&&{mt:2}},...a&&{header:{...a,sx:{...o,...a.sx}}},...i}))})))}ma.displayName="List";const ha=Object.assign(ma,{Group:Zo,Item:ua,Divider:ta}),xa={xsmall:"192px",small:"256px",medium:"320px",large:"432px",xlarge:"600px",auto:"auto"},va={small:"256px",medium:"320px",large:"480px",xlarge:"640px",xxlarge:"960px",auto:"auto"},ya=i.default.div`
  background-color: ${vt("colors.bg.overlay")};
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.12), 0px 8px 24px rgba(149, 157, 165, 0.2);
  position: absolute;
  min-width: 192px;
  max-width: 640px;
  height: ${e=>xa[e.height||"auto"]};
  width: ${e=>va[e.width||"auto"]};
  border-radius: 12px;
  overflow: hidden;
  animation: overlay-appear 200ms ${vt("animation.easeOutCubic")};

  @keyframes overlay-appear {
    0% {
      opacity: 0;
      transform: translateY(${vt("space.2")});
    }
    100% {
      opacity: 1;
    }
  }
  visibility: ${e=>e.visibility||"visible"};
  ${yt};
  ${kt};
  ${Vt};
`,wa=n.default.forwardRef((({onClickOutside:e,role:r="dialog",initialFocusRef:o,returnFocusRef:a,ignoreClickRefs:i,onEscape:d,visibility:l,...c},s)=>{const f=t.useRef(null),b=Bo(f,s),g=gr({overlayRef:f,returnFocusRef:a,onEscape:d,ignoreClickRefs:i,onClickOutside:e,initialFocusRef:o});return n.default.createElement(wo,null,n.default.createElement(ya,Et({},g,{"aria-modal":"true",role:r},c,{ref:b,visibility:l})))}));wa.defaultProps={height:"auto",width:"auto"};const Ba=({renderAnchor:e,anchorRef:r,children:o,open:a,onOpen:i,onClose:d,height:l,width:c,overlayProps:s,focusZoneSettings:f})=>{const b=br(r),[g,u]=function(){const[e,r]=t.useState(null),o=t.useRef(null);o.current=e;const a=t.useCallback((e=>{o.current=e,r(e)}),[o]);return[o,a]}(),p=t.useMemo(wr,[]),m=t.useCallback((()=>null==d?void 0:d("click-outside")),[d]),h=t.useCallback((()=>null==d?void 0:d("escape")),[d]),x=t.useCallback((e=>{e.defaultPrevented||!a&&["ArrowDown","ArrowUp"," ","Enter"].includes(e.key)&&(null==i||i("anchor-key-press"),e.preventDefault())}),[a,i]),v=t.useCallback((e=>{e.defaultPrevented||0!==e.button||a||null==i||i("anchor-click")}),[a,i]),{position:y}=function(e,t=[]){const r=br(null==e?void 0:e.floatingElementRef),o=br(null==e?void 0:e.anchorElementRef),[a,i]=n.default.useState(void 0);return n.default.useEffect((()=>{r.current instanceof Element&&o.current instanceof Element?i(eo(r.current,o.current,e)):i(void 0)}),[r,o,...t]),{floatingElementRef:r,anchorElementRef:o,position:a}}({anchorElementRef:b,floatingElementRef:g},[g.current]),w=t.useMemo((()=>y&&{top:`${y.top}px`,left:`${y.left}px`}),[y]);return mo({containerRef:g,disabled:!a||!y,...f}),go({containerRef:g,disabled:!a||!y}),n.default.createElement(n.default.Fragment,null,e&&e({ref:b,id:p,"aria-labelledby":p,"aria-haspopup":"listbox",tabIndex:0,onClick:v,onKeyDown:x}),a?n.default.createElement(wa,Et({returnFocusRef:b,onClickOutside:m,onEscape:h,ref:u,role:"listbox",visibility:y?"visible":"hidden",height:l,width:c},w,s),o):null)};Ba.displayName="AnchoredOverlay";const $a=e=>n.default.createElement(ua,Et({role:"menuitem"},e));$a.displayName="ActionMenuItem",$a.displayName="ActionMenu.Item";const ka=({anchorContent:e,renderAnchor:r=(e=>n.default.createElement(zr,e)),anchorRef:o,onAction:a,open:i,setOpen:d,overlayProps:l,items:c,...s})=>{const[f,b]=function(e,r,o){const[a,n]=t.useState(o);return[null!=e?e:a,t.useCallback((e=>{n(e),r&&r(e)}),[r])]}(i,d,!1),g=br(o),u=t.useCallback((()=>b(!0)),[b]),p=t.useCallback((()=>b(!1)),[b]),m=t.useMemo((()=>null===r?null:t=>r({"aria-label":"menu",children:e,...t})),[e,r]),h=t.useMemo((()=>c.map((e=>({...e,role:"menuitem",onAction:(t,r)=>{var o;const n=null!==(o=e.onAction)&&void 0!==o?o:a;null==n||n(t,r),r.defaultPrevented||p()}})))),[c,a,p]);return n.default.createElement(Ba,{renderAnchor:m,anchorRef:g,open:f,onOpen:u,onClose:p,overlayProps:l},n.default.createElement(ma,Et({},s,{role:"menu",items:h})))};ka.displayName="ActionMenuBase",ka.displayName="ActionMenu";const Sa=Object.assign(ka,{Divider:ta,Item:$a});const Ta=i.default.img.attrs((e=>({height:e.size,width:e.size})))`
  display: inline-block;
  overflow: hidden; // Ensure page layout in Firefox should images fail to load
  line-height: ${vt("lineHeights.condensedUltra")};
  vertical-align: middle;
  border-radius: ${e=>function({size:e,square:t}){return t?e&&e<=24?"4px":"6px":"50%"}(e)};
  ${yt};
  ${Vt}
`;Ta.defaultProps={size:20,alt:"",square:!1};const Ea=i.default(Ta)`
  position: absolute;
  right: -15%;
  bottom: -9%;
  box-shadow: ${vt("shadows.avatar.childShadow")};
`,Ca=({children:e,...t})=>{const r=n.default.Children.map(e,((e,t)=>n.default.isValidElement(e)?0===t?n.default.cloneElement(e,{size:40}):n.default.createElement(Ea,Et({bg:"bg.canvas"},e.props,{size:20})):e));return n.default.createElement(tr,Et({display:"inline-flex"},t),r)};Ca.displayName="AvatarPair",Ca.displayName="AvatarPair";var Ia=function(e,t,r){return e(r={path:t,exports:{},require:function(e,t){return function(){throw new Error("Dynamic requires are not currently supported by @rollup/plugin-commonjs")}(null==t&&r.path)}},r.exports),r.exports}((function(e){
/*!
    Copyright (c) 2018 Jed Watson.
    Licensed under the MIT License (MIT), see
    http://jedwatson.github.io/classnames
  */
!function(){var t={}.hasOwnProperty;function r(){for(var e=[],o=0;o<arguments.length;o++){var a=arguments[o];if(a){var n=typeof a;if("string"===n||"number"===n)e.push(a);else if(Array.isArray(a)){if(a.length){var i=r.apply(null,a);i&&e.push(i)}}else if("object"===n)if(a.toString===Object.prototype.toString)for(var d in a)t.call(a,d)&&a[d]&&e.push(d);else e.push(a.toString())}}return e.join(" ")}e.exports?(r.default=r,e.exports=r):window.classNames=r}()}));const Na=i.default.span`
  display: flex;
  position: relative;
  height: 20px;
  min-width: ${e=>1===e.count?"20px":2===e.count?"30px":"38px"};

  .pc-AvatarItem {
    flex-shrink: 0;
    height: 20px;
    width: 20px;
    box-shadow: 0 0 0 1px ${vt("colors.bg.canvas")};
    margin-left: -11px;
    position: relative;
    overflow: hidden;
    transition: margin 0.2s ease-in-out, opacity 0.2s ease-in-out, visibility 0.2s ease-in-out,
      box-shadow 0.1s ease-in-out;

    &:first-child {
      margin-left: 0;
    }
    &:nth-child(n + 4) {
      display: none;
    }
  }

  &.pc-AvatarStack--two {
    min-width: 30px;
    .pc-AvatarItem {
      &:nth-child(n + 3) {
        display: none;
      }
    }
  }

  &.pc-AvatarStack--three-plus {
    min-width: 38px;
    .pc-AvatarItem {
      &:nth-child(3) {
        opacity: ${55}%;
        margin-left: -17px;
      }
      &:nth-child(4) {
        opacity: ${40}%;
        margin-left: -17px;
      }
      &:nth-child(5) {
        opacity: ${25}%;
        margin-left: -17px;
      }
      &:nth-child(n + 4) {
        display: block;
      }
      &:nth-child(n + 6) {
        opacity: 0;
        visibility: hidden;
      }
    }
  }

  &.pc-AvatarStack--right {
    justify-content: flex-end;
    .pc-AvatarItem {
      margin-left: 0 !important;
      margin-right: -11px;

      &:first-child {
        margin-right: 0;
      }
    }

    .pc-AvatarStackBody {
      flex-direction: row-reverse;

      &:hover {
        .pc-AvatarItem {
          margin-right: ${vt("space.1")}!important;
          margin-left: 0 !important;

          &:first-child {
            margin-right: 0 !important;
          }
        }
      }
    }
  }

  &.pc-AvatarStack--three-plus.pc-AvatarStack--right {
    .pc-AvatarItem {
      &:nth-child(3) {
        margin-right: -17px;
      }
      &:nth-child(4) {
        margin-right: -17px;
      }
      &:nth-child(5) {
        margin-right: -17px;
      }
    }
  }

  .pc-AvatarStackBody:hover {
    width: auto;

    .pc-AvatarItem {
      margin-left: ${vt("space.1")};
      opacity: 100%;
      visibility: visible;
      box-shadow: 0 0 0 4px ${vt("colors.bg.canvas")};
      &:first-child {
        margin-left: 0;
      }
    }
  }

  ${yt}
  ${Vt};
`,Aa=({children:e,alignRight:t,...r})=>{const o=n.default.Children.count(e),a=Ia({"pc-AvatarStack--two":2===o,"pc-AvatarStack--three-plus":o>2,"pc-AvatarStack--right":t});return n.default.createElement(Na,Et({count:o,className:a},r),n.default.createElement(Qt,{display:"flex",width:"38px",className:"pc-AvatarStackBody"},(e=>n.default.Children.map(e,((e,t)=>n.default.isValidElement(e)?n.default.cloneElement(e,{className:Ia(e.props.className,"pc-AvatarItem"),sx:{zIndex:10-t,...e.props.sx}}):e)))(e)))};Aa.displayName="AvatarStack";const za=i.default.a`
  display: inline-block;
  padding: 2px 6px;
  font-size: ${vt("fontSizes.0")};
  font-family: ${vt("fonts.mono")};
  color: ${vt("colors.branchName.text")};
  background-color: ${vt("colors.branchName.bg")};
  border-radius: ${vt("radii.2")};
  ${yt};
  ${Vt};
`,Ra=i.default.li`
  display: inline-block;
  white-space: nowrap;
  list-style: none;
  &::after {
    padding-right: 0.5em;
    padding-left: 0.5em;
    color: ${vt("colors.text.disabled")};
    font-size: ${vt("fontSizes.1")};
    content: '/';
  }
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    &::after {
      content: none;
    }
  }
`,La=i.default.nav`
  display: flex;
  justify-content: space-between;
  ${yt};
  ${St};
  ${Vt};
`;function Ma({className:e,children:t,theme:r,...o}){const a=Ia(e,"Breadcrumb"),i=n.default.Children.map(t,(e=>n.default.createElement(Ra,{theme:r},e)));return n.default.createElement(La,Et({className:a,"aria-label":"breadcrumb",theme:r},o),n.default.createElement(Xt,{as:"ol",my:0,pl:0},i))}Ma.displayName="Breadcrumb";const Oa=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Ia(e.selected&&"selected",e.className),"aria-current":e.selected?"page":null})))`
  color: ${vt("colors.text.link")};
  display: inline-block;
  font-size: ${vt("fontSizes.1")};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &.selected {
    color: ${vt("colors.text.primary")};
    pointer-events: none;
  }
  ${yt}
  ${Vt};
`;Ma.displayName="Breadcrumb",Oa.displayName="Breadcrumb.Item";var Ha=Object.assign(Ma,{Item:Oa});const Pa={top:"Bottom",right:"Left",bottom:"Top",left:"Right"},Da={top:"Left",right:"Top",bottom:"Left",left:"Top"};function Wa(e,t,r){const o=Pa[e].toLowerCase(),a=Da[e].toLowerCase();return{[o]:"100%",[t||a]:t?r:"50%"}}const Fa=ut({prop:"bg",key:"colors"}),ja=ut({prop:"borderColor",key:"colors"}),Ua=ut({prop:"borderWidth",key:"borderWidths",scale:[0,1]});function _a(e){var t;const o=n.default.useContext(r.ThemeContext),a={...e,theme:null!==(t=e.theme)&&void 0!==t?t:o},{bg:i}=Fa(a),{borderColor:d}=ja(a),{borderWidth:l}=Ua(a),{size:c=8,location:s="bottom"}=e,[f,b]=function(e){const[t,r]=e.split("-");return[t,r]}(s),g=Da[f],u=[-c,0],p=[0,c],m=[c,0],h=`M${u}L${p}L${m}L${u}Z`,x=`M${u}L${p}L${m}`,v={top:`translate(${[c,2*c]}) rotate(180)`,right:`translate(${[0,c]}) rotate(-90)`,bottom:`translate(${[c,0]})`,left:`translate(${[2*c,c]}) rotate(90)`}[f];return n.default.createElement("svg",{width:2*c,height:2*c,style:{pointerEvents:"none",position:"absolute",...Wa(f,b,c),[`margin${g}`]:b?null:-c}},n.default.createElement("g",{transform:v},n.default.createElement("path",{d:h,fill:i}),n.default.createElement("path",{d:x,fill:"none",stroke:d,strokeWidth:l})))}_a.displayName="Caret",_a.locations=["top","top-left","top-right","right","right-top","right-bottom","bottom","bottom-left","bottom-right","left","left-top","left-bottom"],_a.defaultProps={bg:"bg.canvas",borderColor:"border.primary",borderWidth:1};const Ga={small:56,medium:96,large:128},Ka=i.default.div`
  display: ${e=>e.inline?"inline-flex":"flex"};
  align-items: center;
  justify-content: center;
  background-color: ${vt("colors.bg.canvas")};
  border-radius: 50%;
  box-shadow: ${vt("shadows.shadow.medium")};
  ${yt};
  ${({size:e,variant:t="medium"})=>{const r=(o=e,!isNaN(parseFloat(o))&&isFinite(o)?e:Ga[t]);var o;return{width:r,height:r}}};
  ${Vt};
`,Va=i.default(po)`
  height: auto;
  max-width: 60%;
  max-height: 55%;
`;Ka.defaultProps={inline:!1},Va.displayName="CircleBadge.Icon";var Xa=Object.assign(Ka,{Icon:Va});function qa(e){const{size:t,as:r}=e,{icon:o,bg:a,...i}=e;return n.default.createElement(qt,{as:r,bg:a,overflow:"hidden",borderWidth:0,size:t,borderRadius:"50%"},n.default.createElement(Yt,Et({},i,{alignItems:"center",justifyContent:"center"}),n.default.createElement(o,{size:t})))}qa.displayName="CircleOcticon",qa.defaultProps={...Yt.defaultProps,size:32};const Ya=i.default.span`
  display: inline-block;
  padding: 2px 5px;
  font-size: ${vt("fontSizes.0")};
  font-weight: ${vt("fontWeights.bold")};
  line-height: ${vt("lineHeights.condensedUltra")};
  border-radius: 20px;
  ${({scheme:e,...t})=>({color:"secondary"===e?vt("colors.counter.text")(t):"primary"===e?vt("colors.counter.primary.text")(t):vt("colors.counter.text")(t)})};
  ${({scheme:e,...t})=>({backgroundColor:"secondary"===e?vt("colors.counter.bg")(t):"primary"===e?vt("colors.counter.primary.bg")(t):vt("colors.counter.bg")(t)})};
  ${yt};

  &:empty {
    display: none;
  }

  ${Vt};
`,Ja=i.default.details`
  & > summary {
    list-style: none;
  }
  & > summary::-webkit-details-marker {
    display: none;
  }

  ${yt}
  ${Vt};
`;Ja.displayName="Details";const Za=()=>null;function Qa(e){const t=e;return t.tabIndex>=0&&!t.disabled&&function(e){return!e.hidden&&(!e.type||"hidden"!==e.type)&&(e.offsetWidth>0||e.offsetHeight>0)}(t)}const en=i.default.span`
  ${wt};
  ${yt};
  ${Vt};
`,tn=()=>null,rn=i.default.div`
  box-shadow: ${vt("shadows.shadow.large")};
  border-radius: ${vt("radii.2")};
  position: fixed;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  max-height: 80vh;
  z-index: 999;
  margin: 10vh auto;
  background-color: ${vt("colors.bg.primary")};
  width: ${e=>e.narrow?"320px":e.wide?"640px":"440px"};
  outline: none;

  @media screen and (max-width: 750px) {
    width: 100vw;
    margin: 0;
    border-radius: 0;
    height: 100vh;
  }

  ${$t};
  ${yt};
  ${Vt};
`,on=i.default(Yt)`
  border-radius: ${vt("radii.2")} ${vt("radii.2")} 0px 0px;
  border-bottom: 1px solid ${vt("colors.border.primary")};

  @media screen and (max-width: 750px) {
    border-radius: 0px;
  }

  ${Vt};
`;function an({theme:e,children:t,backgroundColor:r="gray.1",...o}){return n.default.Children.toArray(t).every((e=>"string"==typeof e))&&(t=n.default.createElement(en,{theme:e,color:"text.primary",fontSize:1,fontWeight:"bold",fontFamily:"sans-serif"},t)),n.default.createElement(on,Et({theme:e,p:3,backgroundColor:r},o),t)}an.displayName="DialogHeader";const nn=i.default.span`
  &:before {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    cursor: default;
    content: ' ';
    background: transparent;
    z-index: 99;
    background: ${vt("colors.fade.black50")};
  }
`,dn=t.forwardRef((({children:e,onDismiss:r=tn,isOpen:o,initialFocusRef:a,returnFocusRef:i,...d},l)=>{const c=t.useRef(null),s=Bo(l),f=t.useRef(null),b=()=>{r(),i&&i.current&&i.current.focus()},{getDialogProps:g}=function({modalRef:e,overlayRef:r,isOpen:o,onDismiss:a=Za,initialFocusRef:n,closeButtonRef:i}){const d=t.useCallback((t=>{e.current&&r.current&&!e.current.contains(t.target)&&r.current.contains(t.target)&&a()}),[a,e,r]);t.useEffect((()=>{if(o)return document.addEventListener("click",d),()=>{document.removeEventListener("click",d)}}),[o,d]),t.useEffect((()=>{o&&(n&&n.current?n.current.focus():i&&i.current&&i.current.focus())}),[o,n,i]);const l=t.useCallback(((t,r)=>{if(e.current){const o=Array.from(e.current.querySelectorAll("*")).filter(Qa);if(0===o.length)return;t.preventDefault();const a=document.activeElement;if(!a)return;const n=o.indexOf(a)+r,i=1===r?0:o.length-1;return o[n]||o[i]}}),[e]),c=t.useCallback((e=>{const t=e.shiftKey?-1:1,r=l(e,t);r&&r.focus()}),[l]),s=t.useCallback((e=>{switch(e.key){case"Tab":c(e);break;case"Escape":a(),e.stopPropagation()}}),[c,a]);return{getDialogProps:()=>({onKeyDown:s})}}({modalRef:s,onDismiss:b,isOpen:o,initialFocusRef:a,closeButtonRef:f,returnFocusRef:i,overlayRef:c});return o?n.default.createElement(n.default.Fragment,null,n.default.createElement(nn,{ref:c}),n.default.createElement(rn,Et({tabIndex:-1,ref:s,role:"dialog","aria-modal":"true"},d,g()),n.default.createElement(Zr,{ref:f,onClick:b,sx:{position:"absolute",top:"16px",right:"16px"}}),e)):null}));an.defaultProps={backgroundColor:"bg.tertiary"},an.propTypes={...Yt.propTypes},an.displayName="Dialog.Header",dn.displayName="Dialog";var ln=Object.assign(dn,{Header:an});const cn=i.default(Ja)`
  position: relative;
  display: inline-block;
`,sn=({children:e,className:t,...r})=>{const{getDetailsProps:o}=or({closeOnOutsideClick:!0});return n.default.createElement(cn,Et({className:t},o(),r),e)};sn.displayName="Dropdown";const fn=({children:e,...t})=>n.default.createElement(zr,Et({as:"summary","aria-haspopup":"true"},t),e,n.default.createElement(bn,null));fn.displayName="DropdownButton";const bn=i.default.div`
  border: 4px solid transparent;
  margin-left: 12px;
  border-top-color: currentcolor;
  border-bottom-width: 0;
  content: '';
  display: inline-block;
  height: 0;
  vertical-align: middle;
  width: 0;
  ${yt};
  ${Vt};
`,gn=i.default.ul`
  background-clip: padding-box;
  background-color: ${vt("colors.bg.overlay")};
  border: 1px solid ${vt("colors.border.overlay")};
  border-radius: ${vt("radii.2")};
  box-shadow: ${vt("shadows.dropdown.shadow")};
  left: 0;
  list-style: none;
  margin-top: 2px;
  padding: 5px 0 5px 0 !important; //TODO: fix this override on our markdown styles
  position: absolute;
  top: 100%;
  width: 160px;
  z-index: 100;

  &::before {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::after {
    position: absolute;
    display: inline-block;
    content: '';
  }

  &::before {
    border: 8px solid transparent;
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  &::after {
    border: 7px solid transparent;
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  // stylelint-disable-next-line selector-max-type
  > ul {
    list-style: none;
  }
  ${e=>{return e.direction?(t=e.theme,r=e.direction,{w:`\n      top: 0;\n      right: 100%;\n      left: auto;\n      width: auto;\n      margin-top: 0;\n      margin-right: 10px;\n\n      &::before {\n        top: 10px;\n        right: -16px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${vt("colors.border.overlay")(t)};\n      }\n\n      &::after {\n        top: 11px;\n        right: -14px;\n        left: auto;\n        border-color: transparent;\n        border-left-color: ${vt("colors.border.overlay")(t)};\n      }\n    `,e:`\n      top: 0;\n      left: 100%;\n      width: auto;\n      margin-top: 0;\n      margin-left: 10px;\n\n      &::before {\n        top: 10px;\n        left: -16px;\n        border-color: transparent;\n        border-right-color: ${vt("colors.border.overlay")(t)};\n      }\n\n      &::after {\n        top: 11px;\n        left: -14px;\n        border-color: transparent;\n        border-right-color: ${vt("colors.border.overlay")(t)};\n      }\n    `,ne:`\n      top: auto;\n      bottom: 100%;\n      left: 0;\n      margin-bottom: 3px;\n\n      &::before,\n      &::after {\n        top: auto;\n        right: auto;\n      }\n\n      &::before {\n        bottom: -8px;\n        left: 9px;\n        border-top: 8px solid ${vt("colors.border.overlay")(t)};\n        border-bottom: 0;\n        border-left: 8px solid transparent;\n      }\n\n      &::after {\n        bottom: -7px;\n        left: 10px;\n        border-top: 7px solid ${vt("colors.border.overlay")(t)};\n        border-right: 7px solid transparent;\n        border-bottom: 0;\n        border-left: 7px solid transparent;\n      }\n    `,s:"\n      right: 50%;\n      left: auto;\n      transform: translateX(50%);\n\n      &::before {\n        top: -16px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n\n      &::after {\n        top: -14px;\n        right: 50%;\n        transform: translateX(50%);\n      }\n    ",sw:"\n      right: 0;\n      left: auto;\n\n      &::before {\n        top: -16px;\n        right: 9px;\n        left: auto;\n      }\n\n      &::after {\n        top: -14px;\n        right: 10px;\n        left: auto;\n      }\n    ",se:"\n      &::before {\n        top: -16px;\n        left: 9px;\n      }\n\n      &::after {\n        top: -14px;\n        left: 10px;\n      }\n    "}[r]):"";var t,r}};
  ${yt};
  ${Vt};
`,un=i.default.li`
  display: block;
  padding: ${vt("space.1")} 10px ${vt("space.1")} 15px;
  overflow: hidden;
  color: ${vt("colors.text.primary")};
  text-overflow: ellipsis;
  white-space: nowrap;
  a {
    color: ${vt("colors.text.primary")};
    text-decoration: none;
    display: block;
    overflow: hidden;
    color: ${vt("colors.text.primary")};
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &:focus,
  a:focus {
    color: ${vt("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${vt("colors.state.hover.primaryBg")};
  }

  &:hover,
  &:hover a {
    color: ${vt("colors.state.hover.primaryText")};
    text-decoration: none;
    background-color: ${vt("colors.state.hover.primaryBg")};
    outline: none;
  }
  ${yt};
  ${Vt};
`;gn.defaultProps={direction:"sw"},gn.displayName="Dropdown.Menu",un.displayName="Dropdown.Item",fn.defaultProps=zr.defaultProps,fn.displayName="Dropdown.Button",bn.displayName="Dropdown.Caret",sn.defaultProps=Ja.defaultProps;var pn=Object.assign(sn,{Caret:bn,Menu:gn,Item:un,Button:fn});const mn=n.default.forwardRef((({children:e,...t},r)=>n.default.createElement(zr,Et({ref:r},t),e,n.default.createElement(po,{icon:qr,ml:1}))));function hn({renderAnchor:e=(e=>n.default.createElement(mn,e)),placeholder:r,selectedItem:o,onChange:a,overlayProps:i,items:d,...l}){const[c,s]=t.useState(!1),f=t.useCallback((()=>s(!0)),[]),b=t.useCallback((()=>s(!1)),[]),g=t.useCallback((t=>{var a;return e({...t,children:null!==(a=null==o?void 0:o.text)&&void 0!==a?a:r})}),[r,e,null==o?void 0:o.text]),u=t.useMemo((()=>d.map((e=>({...e,role:"option",selected:e===o,onAction:(t,r)=>{var n;null===(n=e.onAction)||void 0===n||n.call(e,t,r),r.defaultPrevented||(b(),null==a||a(e===o?void 0:e))}})))),[d,a,b,o]);return n.default.createElement(Ba,{renderAnchor:g,open:c,onOpen:f,onClose:b,overlayProps:i},n.default.createElement(ma,Et({},l,{role:"listbox",items:u})))}hn.displayName="DropdownMenu",hn.displayName="DropdownMenu";const xn=i.default.div`
  ${yt};
  display: flex;
  align-items: stretch;

  summary {
    border-radius: 0;
    border-top-left-radius: ${vt("radii.2")};
    border-bottom-left-radius: ${vt("radii.2")};
    border-right: 0;
  }
  .TextInput-wrapper {
    border-radius: 0;
    border-top-right-radius: ${vt("radii.2")};
    border-bottom-right-radius: ${vt("radii.2")};
    z-index: 1; // Allows the focus outline to show on top of the dropdown.
  }

  ${Vt}
`,vn=i.default.ul`
  list-style-type: none;
  ${yt};
  ${Vt};
`,yn=({children:e,...t})=>{const r=n.default.Children.map(e,(e=>n.default.createElement("li",null,e)));return n.default.createElement(vn,t,r)};yn.displayName="FilterList";const wn=i.default.a`
  position: relative;
  display: block;
  padding: ${e=>e.small?`${vt("space.1")(e)} 10px`:`${vt("space.2")(e)} 11px`};
  margin: ${e=>e.small?"0 0 2px":"0 0 5px 0"};
  overflow: hidden;
  font-size: ${vt("fontSizes.1")};
  color: ${e=>e.selected?vt("colors.state.selected.primaryText"):vt("colors.text.secondary")};
  background-color: ${e=>e.selected?vt("colors.state.selected.primaryBg"):""}!important;
  text-decoration: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
  border-radius: ${vt("radii.1")};
  &:hover {
    text-decoration: none;
    background-color: ${vt("colors.bg.tertiary")};
  }
  &:active {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
  }
  .count {
    float: right;
    font-weight: ${vt("fontWeights.bold")};
  }
  ${yt};
  ${Vt};
`;function Bn({children:e,count:t,...r}){return n.default.createElement(wn,r,t&&n.default.createElement("span",{title:"results",className:"count"},t),e)}Bn.displayName="FilterListItem",yn.defaultProps={m:0,p:0},Bn.displayName="FilterList.Item";var $n=Object.assign(yn,{Item:Bn});const kn=re({variants:{default:{color:"alert.info.text",backgroundColor:"alert.info.bg",borderColor:"alert.info.border",svg:{color:"alert.info.icon"}},success:{color:"alert.success.text",backgroundColor:"alert.success.bg",borderColor:"alert.success.border",svg:{color:"alert.success.icon"}},danger:{color:"alert.error.text",backgroundColor:"alert.error.bg",borderColor:"alert.error.border",svg:{color:"alert.error.icon"}},warning:{color:"alert.warn.text",backgroundColor:"alert.warn.bg",borderColor:"alert.warn.border",svg:{color:"alert.warn.icon"}}}}),Sn=i.default.div`
  position: relative;
  color: ${vt("colors.text.primary")};
  padding: ${vt("space.3")};
  border-style: solid;
  border-width: ${e=>e.full?"1px 0px":"1px"};
  border-radius: ${e=>e.full?"0":vt("radii.2")};
  margin-top: ${e=>e.full?"-1px":"0"};

  p:last-child {
    margin-bottom: 0;
  }

  svg {
    margin-right: ${vt("space.2")};
  }

  ${yt};
  ${kn};
  ${Vt};
`;Sn.defaultProps={variant:"default"};const Tn=i.default.div`
  margin: ${vt("space.3")} 0;
  font-weight: ${vt("fontWeights.normal")};
  ${yt};
  ${Vt};
`,En=i.default.label`
  display: block;
  margin: 0 0 ${vt("space.2")};
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
  ${wt};
  ${yt};
  ${Vt};
`;En.displayName="FormGroup.Label";var Cn=Object.assign(Tn,{Label:En});const In=i.default.div`
  z-index: 32;
  display: flex;
  padding: ${vt("space.3")};
  font-size: ${vt("fontSizes.1")};
  line-height: ${vt("lineHeights.default")};
  color: ${vt("colors.header.text")};
  background-color: ${vt("colors.header.bg")};
  align-items: center;
  flex-wrap: nowrap;

  ${yt}
  ${Bt}
  ${Vt};
`,Nn=i.default.div`
  display: flex;
  margin-right: ${vt("space.3")};
  align-self: stretch;
  align-items: center;
  flex-wrap: nowrap;

  ${({full:e})=>e&&r.css(["flex:auto;"])};

  ${yt};
  ${Bt};
  ${Vt};
`;Nn.displayName="Header.Item";const An=i.default.a.attrs((({to:e})=>"string"==typeof e?{"aria-current":"page"}:{}))`
  font-weight: ${vt("fontWeights.bold")};
  color: ${vt("colors.header.logo")};
  white-space: nowrap;
  cursor: pointer;
  text-decoration: none;
  display: flex;
  align-items: center;

  &:hover,
  &:focus {
    color: ${vt("colors.header.text")};
  }

  ${yt};
  ${Bt};
  ${wt};
  ${Vt};
`;An.displayName="Header.Link";var zn=Object.assign(In,{Link:An,Item:Nn});const Rn=i.default.h2`
  font-weight: ${vt("fontWeights.bold")};
  font-size: ${vt("fontSizes.5")};
  margin: 0;
  ${wt};
  ${yt};
  ${Vt};
`;Rn.defaultProps={theme:l.theme};const Ln=i.default.span`
  ${yt}
  & * {
    margin-right: ${vt("space.1")};
  }
  & *:last-child {
    margin-right: 0;
  }
  ${Vt};
`,Mn=r.css(["margin-top:-1px;margin-bottom:-1px;color:",";border:"," solid ",";box-shadow:none;",";",";background-color:transparent;"],vt("colors.label.secondary.text"),vt("borderWidths.1"),vt("colors.label.border"),Qe,yt),On=re({variants:{small:{fontSize:0,lineHeight:"16px",padding:"0px 8px"},medium:{fontSize:0,lineHeight:"20px",padding:"0 8px"},large:{fontSize:0,lineHeight:"24px",padding:"0 12px"},xl:{fontSize:1,lineHeight:"16px",padding:"8px 12px"}}}),Hn=i.default.span`
  display: inline-block;
  font-weight: ${vt("fontWeights.semibold")};
  color: ${vt("colors.text.inverse")};
  border-radius: ${vt("radii.3")};

  &:hover {
    text-decoration: none;
  }

  ${On}
  ${yt}
  ${e=>e.dropshadow?"box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12)":""}
  ${e=>e.outline?Mn:""} // must be last to override other values
  ${Vt}
`;Hn.defaultProps={bg:"label.primary.border",variant:"medium"};const Pn=$({hoverColor:{property:"color",scale:"colors"}}),Dn=i.default.a`
  color: ${e=>e.muted?vt("colors.text.secondary")(e):vt("colors.text.link")(e)};
  text-decoration: ${e=>e.underline?"underline":"none"};
  &:hover {
    text-decoration: ${e=>e.muted?"none":"underline"};
    ${e=>e.hoverColor?Pn:e.muted?`color: ${vt("colors.text.link")(e)}`:""};
  }
  &:is(button) {
    display: inline-block;
    padding: 0;
    font-size: inherit;
    white-space: nowrap;
    cursor: pointer;
    user-select: none;
    background-color: transparent;
    border: 0;
    appearance: none;
  }
  ${wt};
  ${yt};
  ${Vt};
`,Wn=i.default.div`
  position: relative;
  padding-top: ${vt("space.4")};
  padding-bottom: ${vt("space.4")};
  margin-bottom: ${vt("space.4")};
  border-bottom: 1px solid ${vt("colors.border.primary")};
  ${yt};
  ${Vt};
`;const Fn=i.default.a`
  display: inline-block;
  min-width: 32px;
  padding: 5px 10px;
  font-style: normal;
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  text-decoration: none;

  margin-right: ${vt("space.1")};

  &:last-child {
    margin-right: 0;
  }

  border: ${vt("borderWidths.1")} solid transparent;
  border-radius: ${vt("radii.2")};
  transition: border-color 0.2s cubic-bezier(0.3, 0, 0.5, 1);

  &:hover,
  &:focus {
    text-decoration: none;
    border-color: ${vt("colors.border.primary")};
    outline: 0;
    transition-duration: 0.1s;
  }

  &:active {
    border-color: ${vt("colors.border.secondary")};
  }

  &[rel='prev'],
  &[rel='next'] {
    color: ${vt("colors.text.link")};
  }

  &[aria-current],
  &[aria-current]:hover {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
    border-color: transparent;
  }

  &[aria-disabled],
  &[aria-disabled]:hover {
    color: ${vt("colors.text.disabled")}; // check
    cursor: default;
    border-color: transparent;
  }

  @supports (clip-path: polygon(50% 0, 100% 50%, 50% 100%)) {
    &[rel='prev']::before,
    &[rel='next']::after {
      display: inline-block;
      width: 16px;
      height: 16px;
      vertical-align: text-bottom;
      content: '';
      background-color: currentColor;
    }

    // chevron-left
    &[rel='prev']::before {
      margin-right: ${vt("space.1")};
      clip-path: polygon(
        9.8px 12.8px,
        8.7px 12.8px,
        4.5px 8.5px,
        4.5px 7.5px,
        8.7px 3.2px,
        9.8px 4.3px,
        6.1px 8px,
        9.8px 11.7px,
        9.8px 12.8px
      );
    }

    // chevron-right
    &[rel='next']::after {
      margin-left: ${vt("space.1")};
      clip-path: polygon(
        6.2px 3.2px,
        7.3px 3.2px,
        11.5px 7.5px,
        11.5px 8.5px,
        7.3px 12.8px,
        6.2px 11.7px,
        9.9px 8px,
        6.2px 4.3px,
        6.2px 3.2px
      );
    }
  }

  ${yt};
`;function jn({theme:e,pageCount:t,currentPage:r,onPageChange:o,hrefBuilder:a,marginPageCount:i,showPages:d,surroundingPageCount:l}){const c=n.default.useCallback((e=>t=>o(t,e)),[o]),s=n.default.useMemo((()=>function(e,t,r,o,a){const n=[];if(r){const r=[],i=t=>{t>=1&&t<=e&&r.push(t)};let d=t-a,l=t+a;if(d<1&&l>e)d=1,l=e;else if(d<1)for(;d<1;)d++,l++;else if(l>e)for(;l>e;)d--,l--;for(let t=1;t<=o;t++){const r=t,o=e-(t-1);r>=d?l++:i(r),o<=l?d--:i(o)}for(let e=d;e<=l;e++)i(e);const c=r.slice().sort(((e,t)=>e-t)).filter(((e,t,r)=>!t||e!==r[t-1]));for(let e=0;e<c.length;e++){const r=c[e],o=r===t;0===e?(1!==r&&n.push({type:"BREAK",num:1}),n.push({type:"NUM",num:r,selected:o})):(1==r-c[e-1]||n.push({type:"BREAK",num:r-1}),n.push({type:"NUM",num:r,selected:o}))}const s=n[n.length-1];"NUM"===s.type&&s.num!==e&&n.push({type:"BREAK",num:e})}const i={type:"NEXT",num:t+1,disabled:t===e};return[{type:"PREV",num:t-1,disabled:1===t},...n,i]}(t,r,!!d,i,l)),[t,r,d,i,l]);return n.default.useMemo((()=>s.map((t=>{const{props:r,key:o,content:i}=function(e,t,r){const o={};let a="",n="";switch(e.type){case"PREV":n="page-prev",a="Previous",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"prev",href:t(e.num),"aria-label":"Previous Page",onClick:r});break;case"NEXT":n="page-next",a="Next",e.disabled?Object.assign(o,{as:"span","aria-disabled":"true"}):Object.assign(o,{rel:"next",href:t(e.num),"aria-label":"Next Page",onClick:r});break;case"NUM":n=`page-${e.num}`,a=String(e.num),e.selected?Object.assign(o,{as:"em","aria-current":"page"}):Object.assign(o,{href:t(e.num),"aria-label":`Page ${e.num}`,onClick:r});break;case"BREAK":n=`page-${e.num}-break`,a="",Object.assign(o,{as:"span","aria-disabled":!0})}return{props:o,key:n,content:a}}(t,a,c(t.num));return n.default.createElement(Fn,Et({},r,{key:o,theme:e}),i)}))),[s,a,c,e])}const Un=i.default.nav`
  margin-top: 20px;
  margin-bottom: 15px;
  text-align: center;
  ${Vt};
`;function _n({theme:e,pageCount:t,currentPage:r,onPageChange:o=Kn,hrefBuilder:a=Gn,marginPageCount:i=1,showPages:d=!0,surroundingPageCount:l=2,...c}){const s=jn({theme:e,pageCount:t,currentPage:r,onPageChange:o,hrefBuilder:a,marginPageCount:i,showPages:d,surroundingPageCount:l});return n.default.createElement(Un,Et({"aria-label":"Pagination"},c,{theme:e}),n.default.createElement(Xt,{display:"inline-block",theme:e},s))}function Gn(e){return`#${e}`}function Kn(){}function Vn(e){const{bg:t,border:r,borderColor:o,theme:a}=e,{caret:i,children:d,...l}=e,c={bg:t,borderColor:o,borderWidth:r,location:i,theme:a};return n.default.createElement(qt,Et({sx:{position:"relative"}},l),d,n.default.createElement(_a,c))}_n.displayName="Pagination",_n.defaultProps={hrefBuilder:Gn,marginPageCount:1,onPageChange:Kn,showPages:!0,surroundingPageCount:2},Vn.displayName="PointerBox";const Xn=i.default.div.attrs((({className:e,caret:t})=>({className:Ia(e,`caret-pos--${t}`)})))`
  position: ${e=>e.relative?"relative":"absolute"};
  z-index: 100;
  display: ${e=>e.open?"block":"none"};

  ${yt};
  ${$t};
  ${kt};
  ${Vt};
`,qn=i.default(qt)`
  position: relative;
  width: 232px;
  margin-right: auto;
  margin-left: auto;
  padding: ${vt("space.4")};
  background-color: ${vt("colors.bg.overlay")};

  ${yt};
  ${$t};

  // Carets
  &::before,
  &::after {
    position: absolute;
    left: 50%;
    display: inline-block;
    content: '';
  }

  &::before {
    top: -${vt("space.3")};
    margin-left: -9px;
    border: ${vt("space.2")} solid transparent; // TODO: solid?
    border-bottom-color: ${vt("colors.border.overlay")};
  }

  &::after {
    top: -14px;
    margin-left: -${vt("space.2")};
    border: 7px solid transparent; // todo: solid
    border-bottom-color: ${vt("colors.bg.overlay")};
  }

  // Bottom-oriented carets
  ${Xn}.caret-pos--bottom & ,
  ${Xn}.caret-pos--bottom-right & ,
  ${Xn}.caret-pos--bottom-left & {
    &::before,
    &::after {
      top: auto;
      border-bottom-color: transparent;
    }

    &::before {
      bottom: -${vt("space.3")};
      border-top-color: ${vt("colors.border.overlay")};
    }

    &::after {
      bottom: -14px;
      // stylelint-disable-next-line primer/borders
      border-top-color: ${vt("colors.bg.overlay")};
    }
  }

  // Top & Bottom: Right-oriented carets
  ${Xn}.caret-pos--top-right & ,
  ${Xn}.caret-pos--bottom-right & {
    right: -9px;
    margin-right: 0;

    &::before,
    &::after {
      left: auto;
      margin-left: 0;
    }

    &::before {
      right: 20px;
    }

    &::after {
      right: 21px;
    }
  }

  // Top & Bottom: Left-oriented carets
  ${Xn}.caret-pos--top-left & ,
  ${Xn}.caret-pos--bottom-left & {
    left: -9px;
    margin-left: 0;

    &::before,
    &::after {
      left: ${vt("space.4")};
      margin-left: 0;
    }

    &::after {
      left: calc(${vt("space.4")} + 1px);
    }
  }

  // Right- & Left-oriented carets
  ${Xn}.caret-pos--right & ,
  ${Xn}.caret-pos--right-top & ,
  ${Xn}.caret-pos--right-bottom & ,
  ${Xn}.caret-pos--left & ,
  ${Xn}.caret-pos--left-top & ,
  ${Xn}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: 50%;
      left: auto;
      margin-left: 0;
      border-bottom-color: transparent;
    }

    &::before {
      // stylelint-disable-next-line primer/spacing
      margin-top: calc((${vt("space.2")} + 1px) * -1);
    }

    &::after {
      margin-top: -${vt("space.2")};
    }
  }

  // Right-oriented carets
  ${Xn}.caret-pos--right & ,
  ${Xn}.caret-pos--right-top & ,
  ${Xn}.caret-pos--right-bottom & {
    &::before {
      right: -${vt("space.3")};
      border-left-color: ${vt("colors.border.overlay")};
    }

    &::after {
      right: -14px;
      // stylelint-disable-next-line primer/borders
      border-left-color: ${vt("colors.bg.overlay")};
    }
  }

  // Left-oriented carets
  ${Xn}.caret-pos--left & ,
  ${Xn}.caret-pos--left-top & ,
  ${Xn}.caret-pos--left-bottom & {
    &::before {
      left: -${vt("space.3")};
      border-right-color: ${vt("colors.border.overlay")};
    }

    &::after {
      left: -14px;
      // stylelint-disable-next-line primer/borders
      border-right-color: ${vt("colors.bg.overlay")};
    }
  }

  // Right & Left: Top-oriented carets
  ${Xn}.caret-pos--right-top & ,
  ${Xn}.caret-pos--left-top & {
    &::before,
    &::after {
      top: ${vt("space.4")};
    }
  }

  // Right & Left: Bottom-oriented carets
  ${Xn}.caret-pos--right-bottom & ,
  ${Xn}.caret-pos--left-bottom & {
    &::before,
    &::after {
      top: auto;
    }

    &::before {
      bottom: ${vt("space.3")};
    }

    &::after {
      bottom: calc(${vt("space.3")} + 1px);
    }
  }

  ${Vt};
`;Xn.defaultProps={caret:"top"},qn.displayName="Popover.Content";var Yn=Object.assign(Xn,{Content:qn});const Jn=i.default.span`
  width: ${e=>e.progress?`${e.progress}%`:0};
  ${yt}
`,Zn={small:"5px",large:"10px",default:"8px"},Qn=i.default.span`
  display: ${e=>e.inline?"inline-flex":"flex"};
  overflow: hidden;
  background-color: ${vt("colors.border.primary")};
  border-radius: ${vt("radii.1")};
  height: ${e=>Zn[e.barSize||"default"]};
  ${yt}
  ${ie}
  ${Vt};
`;function ei({progress:e,bg:t,theme:r,...o}){return n.default.createElement(Qn,Et({theme:r},o),n.default.createElement(Jn,{progress:e,bg:t,theme:r}))}function ti(e,r,o){const a=t.useCallback((t=>{const a=()=>{o(!0)},n=t=>{const r=Array.from(e.current.querySelectorAll('[role^="menuitem"]:not([hidden]):not([disabled]):not([aria-disabled="true"])')),o=document.activeElement,a=r.indexOf(o),n=t?r[a+1]:r[a-1],i=t?r[0]:r[r.length-1];return n||i};if(!(t instanceof KeyboardEvent))return;const i=t.target instanceof Element&&"SUMMARY"===t.target.tagName;switch(t.key){case"Escape":r&&((()=>{o(!1);const t=e.current.querySelector("summary");t&&t.focus()})(),t.preventDefault(),t.stopPropagation());break;case"ArrowDown":{i&&!r&&a();const e=n(!0);e&&e.focus(),t.preventDefault()}break;case"ArrowUp":{i&&!r&&a();const e=n(!1);e&&e.focus(),t.preventDefault()}break;case" ":case"Enter":{const r=document.activeElement;r&&(e=>{const t=e.getAttribute("role");return"menuitem"===t||"menuitemcheckbox"===t||"menuitemradio"===t})(r)&&r.closest("details")===e&&(t.preventDefault(),t.stopPropagation(),r.click())}}}),[e,r,o]);t.useEffect((()=>{const t=e.current;if(t)return t.addEventListener("keydown",a),()=>{t.removeEventListener("keydown",a)}}),[e,a])}ei.displayName="ProgressBar",ei.defaultProps={bg:"bg.successInverse",barSize:"default"};const ri=t.createContext({}),oi=r.css(["padding:"," ",";margin:0;font-size:",";font-weight:",";color:",";background-color:",";border-bottom:"," solid ",";"],vt("space.1"),vt("space.3"),vt("fontSizes.0"),vt("fontWeights.bold"),vt("colors.text.tertiary"),vt("colors.bg.tertiary"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary")),ai=i.default.div`
  ${oi}
  ${yt}
  ${Vt};
`;ai.displayName="SelectMenu.Divider";var ni=k(_,C,E,S,I,R,M,H,A,G,oe,ae,ne),ii=new RegExp("^("+ni.propNames.join("|")+")$");const di=re({variants:{small:{minHeight:"28px",px:2,py:"3px",fontSize:0,lineHeight:"20px"},large:{px:2,py:"10px",fontSize:3}}}),li=i.default.input`
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
  &:focus {
    outline: 0;
  }
`,ci=i.default.span`
  display: inline-flex;
  align-items: stretch;
  min-height: 34px;
  font-size: ${vt("fontSizes.1")};
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  vertical-align: middle;
  background-repeat: no-repeat; // Repeat and position set for form states (success, error, etc)
  background-position: right 8px center; // For form validation. This keeps images 8px from right and centered vertically.
  border: 1px solid ${vt("colors.input.border")};
  border-radius: ${vt("radii.2")};
  outline: none;
  box-shadow: ${vt("shadows.shadow.inset")};

  ${e=>e.hasIcon?r.css(["padding:0;"]):r.css(["padding:6px 12px;"])}

  .TextInput-icon {
    align-self: center;
    color: ${vt("colors.icon.tertiary")};
    margin: 0 ${vt("space.2")};
    flex-shrink: 0;
  }

  &:focus-within {
    border-color: ${vt("colors.state.focus.border")};
    box-shadow: ${vt("shadows.state.focus.shadow")};
  }

  ${e=>e.contrast&&r.css(["background-color:",";"],vt("colors.input.contrastBg"))}

  ${e=>e.disabled&&r.css(["color:",";background-color:",";border-color:",";"],vt("colors.text.secondary"),vt("colors.input.disabledBg"),vt("colors.input.disabledBorder"))}

  ${e=>e.block&&r.css(["display:block;width:100%;"])}

  // Ensures inputs don't zoom on mobile but are body-font size on desktop
  @media (min-width: ${vt("breakpoints.1")}) {
    font-size: ${vt("fontSizes.1")};
  }
  ${yt}
  ${ie}
  ${le}
  ${se}
  ${di}
  ${Vt};
`,si=n.default.forwardRef((({icon:e,contrast:t,className:r,block:o,disabled:a,theme:i,sx:d,...l},c)=>{const s=Ia(r,"TextInput-wrapper"),f=function(e){var t={};for(var r in e)ii.test(r)&&(t[r]=e[r]);return t}(l),b=function(e){var t={};for(var r in e)ii.test(r)||(t[r]=e[r]);return t}(l);return n.default.createElement(ci,Et({className:s,hasIcon:!!e,block:o,theme:i,disabled:a,contrast:t,sx:d},f),e&&n.default.createElement(e,{className:"TextInput-icon"}),n.default.createElement(li,Et({ref:c,disabled:a},b)))}));si.defaultProps={type:"text"},si.displayName="TextInput";const fi=i.default.form`
  padding: ${vt("space.3")};
  margin: 0;
  border-bottom: ${vt("borderWidths.1")} solid ${vt("colors.selectMenu.borderSecondary")};
  background-color: ${vt("colors.bg.overlay")};
  ${yt};

  @media (min-width: ${vt("breakpoints.0")}) {
    padding: ${vt("space.2")};
  }

  ${Vt};
`,bi=t.forwardRef((({theme:e,value:r,sx:o,...a},i)=>{const d=t.useRef(null),l=null!=i?i:d,{open:c}=t.useContext(ri);return t.useEffect((()=>{var e;c&&(null===(e=d.current)||void 0===e||e.focus())}),[c]),n.default.createElement(fi,{theme:e,sx:o},n.default.createElement(si,Et({theme:e,ref:l,width:"100%",block:!0,value:r,contrast:!0},a)))}));bi.displayName="SelectMenu.Filter";const gi=r.css(["margin-top:-1px;padding:"," ",";font-size:",";color:",";text-align:center;border-top:"," solid ",";@media (min-width:","){padding:"," ",";}"],vt("space.2"),vt("space.3"),vt("fontSizes.0"),vt("colors.text.tertiary"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("space.1"),vt("space.2")),ui=i.default.footer`
  ${gi}
  ${yt}
  ${Vt};
`;ui.displayName="SelectMenu.Footer";const pi=i.default.h3`
  color: ${vt("colors.text.primary")};
  flex: auto;
  font-size: ${vt("fontSizes.1")};
  font-weight: ${vt("fontWeights.bold")};
  margin: 0;

  @media (min-width: ${vt("breakpoints.0")}) {
    font-size: inherit;
  }
`,mi=i.default.header`
  display: flex;
  flex: none; // fixes header from getting squeezed in Safari iOS
  padding: ${vt("space.3")};
  border-bottom: ${vt("borderWidths")} solid ${vt("colors.selectMenu.borderSecondary")};
  ${yt}
  ${wt}

  @media (min-width: ${vt("breakpoints.0")}) {
    padding-top: ${vt("space.2")};
    padding-bottom: ${vt("space.2")};
  }

  ${Vt};
`,hi=({children:e,theme:t,...r})=>n.default.createElement(mi,Et({theme:t},r),n.default.createElement(pi,{theme:t},e));hi.displayName="SelectMenuHeader",hi.displayName="SelectMenu.Header";const xi=r.css(["display:flex;align-items:center;padding:",";overflow:hidden;text-align:left;cursor:pointer;background-color:",";border:0;border-bottom:"," solid ",";color:",";text-decoration:none;font-size:",";font-family:inherit;width:100%;&:hover{text-decoration:none;}&:focus{outline:none;}&[hidden]{display:none !important;}@media (min-width:","){padding-top:",";padding-bottom:",";}.SelectMenu-icon{width:",";margin-right:",";flex-shrink:0;}.SelectMenu-selected-icon{visibility:hidden;transition:transform 0.12s cubic-bezier(0.5,0.1,1,0.5),visibility 0s 0.12s linear;transform:scale(0);}&[aria-checked='true']{font-weight:500;color:",";.SelectMenu-selected-icon{visibility:visible;transition:transform 0.12s cubic-bezier(0,0,0.2,1),visibility 0s linear;transform:scale(1);}}@media (hover:hover){&:hover,&:active,&:focus{background-color:",";}}@media (hover:none){&:focus,&:active{background-color:",";}-webkit-tap-highlight-color:",";}"],vt("space.3"),vt("colors.bg.overlay"),vt("borderWidths.1"),vt("colors.selectMenu.borderSecondary"),vt("colors.text.secondary"),vt("fontSizes.0"),vt("breakpoints.0"),vt("space.2"),vt("space.2"),vt("space.3"),vt("space.2"),vt("colors.text.primary"),vt("colors.state.hover.secondaryBg"),vt("colors.bg.secondary"),vt("colors.selectMenu.tapHighlight")),vi=i.default.a.attrs((()=>({role:"menuitemcheckbox"})))`
  ${xi}
  ${yt}
  ${Vt};
`,yi=t.forwardRef((({children:e,selected:r,theme:o,onClick:a,...i},d)=>{const l=t.useContext(ri),c=t.useRef(null),s=null!=d?d:c;return n.default.createElement(vi,Et({ref:s},i,{theme:o,onClick:e=>{var t;(a&&a(e),e.defaultPrevented)||(null===(t=l.setOpen)||void 0===t||t.call(l,!1))},"aria-checked":r}),n.default.createElement(po,{theme:o,className:"SelectMenu-icon SelectMenu-selected-icon",icon:jr}),e)}));yi.defaultProps={selected:!1},yi.displayName="SelectMenu.Item";const wi=r.css(["position:relative;padding:0;margin:0;flex:auto;overflow-x:hidden;overflow-y:auto;background-color:",";-webkit-overflow-scrolling:touch;@media (hover:hover){.SelectMenuTab:focus{background-color:",";}.SelectMenuTab:not([aria-checked='true']):hover{color:",";background-color:",";}.SelectMenuTab:not([aria-checked='true']):active{color:",";background-color:",";}}"],vt("colors.bg.overlay"),vt("colors.selectMenu.tapFocusBg"),vt("colors.text.primary"),vt("colors.bg.tertiary"),vt("colors.text.primary"),vt("colors.bg.tertiary")),Bi=i.default.div`
  ${wi}
  ${yt}
  ${Vt};
`;Bi.displayName="SelectMenu.List";const $i=r.keyframes(["0%{opacity:0.3;}10%{opacity:1;}100%{opacity:0.3;}"]),ki=i.default.div`
  padding: ${vt("space.6")} ${vt("space.4")};
  text-align: center;
  background-color: ${vt("colors.bg.overlay")};
  animation-name: ${$i};
  animation-duration: 2s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  ${yt}
`,Si=e=>n.default.createElement(ki,e,n.default.createElement(po,{size:32,icon:Vr}));Si.displayName="SelectMenuLoadingAnimation";const Ti=r.keyframes(["0%{opacity:0;transform:scale(0.9);}"]),Ei=r.css(["position:relative;z-index:99;display:flex;",";max-height:",";margin:auto 0;",";overflow:hidden;pointer-events:auto;flex-direction:column;background-color:",";border-radius:",";box-shadow:",";animation:"," 0.12s cubic-bezier(0,0.1,0.1,1) backwards;@media (min-width:","){height:auto;max-height:350px;margin:"," 0 "," 0;font-size:",";border:"," solid ",";border-radius:",";box-shadow:",";}"],(e=>e.filter?"height: 80%":""),(e=>e.filter?"none":"66%"),(e=>e.filter?"margin-top: 0":""),vt("colors.bg.overlay"),vt("radii.2"),vt("shadows.shadow.small"),Ti,vt("breakpoints.0"),vt("space.1"),vt("space.3"),vt("fontSizes.0"),vt("borderWidths.1"),vt("colors.border.overlay"),vt("radii.2"),vt("shadows.shadow.small")),Ci=r.css(["position:fixed;top:0;right:0;bottom:0;left:0;z-index:99;display:flex;padding:",";pointer-events:none;flex-direction:column;&::before{position:absolute;top:0;right:0;bottom:0;left:0;pointer-events:none;content:'';background-color:",";@media (min-width:","){display:none;}}@media (min-width:","){position:absolute;top:auto;right:",";bottom:auto;left:auto;padding:0;}"],vt("space.3"),vt("colors.selectMenu.backdropBg"),vt("breakpoints.0"),vt("breakpoints.0"),(e=>"right"===e.align?"0":"auto")),Ii=i.default.div`
  ${Ei}
  ${ie}
`,Ni=i.default.div`
  ${Ci}
  ${yt}
  ${Vt};
`,Ai=n.default.forwardRef((({children:e,theme:t,width:r,...o},a)=>n.default.createElement(Ni,Et({theme:t},o,{role:"menu",ref:a}),n.default.createElement(Ii,{theme:t,width:r},e))));Ai.defaultProps={align:"left",width:"300px"},Ai.displayName="SelectMenu.Modal";const zi=r.css(["flex:1;padding:"," ",";font-size:",";font-weight:500;color:",";text-align:center;background-color:transparent;border:0;box-shadow:inset 0 -1px 0 ",";@media (min-width:","){flex:none;padding:"," ",";border:"," solid transparent;border-bottom-width:0;border-top-left-radius:",";border-top-right-radius:",";}&[aria-selected='true']{z-index:1;color:",";background-color:",";box-shadow:0 0 0 1px ",";@media (min-width:","){border-color:",";box-shadow:none;}}&:focus{background-color:",";}"],vt("space.2"),vt("space.3"),vt("fontSizes.0"),vt("colors.text.tertiary"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("space.1"),vt("space.3"),vt("borderWidths.1"),vt("radii.2"),vt("radii.2"),vt("colors.text-primary"),vt("colors.bg.overlay"),vt("colors.selectMenu.borderSecondary"),vt("breakpoints.0"),vt("colors.selectMenu.borderSecondary"),vt("colors.state.hover.secondaryBg")),Ri=i.default.button`
  ${zi}
  ${yt}
  ${Vt};
`,Li=({tabName:e="",index:r,className:o,onClick:a,...i})=>{const d=t.useContext(ri);t.useEffect((()=>{var t;d.selectedTab||0!==r||(null===(t=d.setSelectedTab)||void 0===t||t.call(d,e))}),[r,d,e]);const l=d.selectedTab===e;return n.default.createElement(Ri,Et({role:"tab",className:Ia("SelectMenuTab",o),"aria-selected":l,onClick:t=>{var r;(a&&a(t),t.defaultPrevented)||(null===(r=d.setSelectedTab)||void 0===r||r.call(d,e))}},i),e)};Li.displayName="SelectMenuTab",Li.displayName="SelectMenu.Tab";const Mi=i.default.div`
  border-top: ${vt("borderWidths.1")} solid ${vt("colors.selectMenu.borderSecondary")};
  ${yt}
  ${Vt};
`,Oi=({tabName:e,className:r,children:o,...a})=>{const i=t.useContext(ri);return n.default.createElement(Mi,Et({role:"tabpanel",className:r,hidden:i.selectedTab!==e},a),n.default.createElement(Bi,null,o))};Oi.displayName="TabPanel",Oi.displayName="SelectMenu.TabPanel";const Hi=r.css(["display:flex;flex-shrink:0;margin-bottom:-1px;-webkit-overflow-scrolling:touch;overflow-x:auto;overflow-y:hidden;&::-webkit-scrollbar{display:none;}@media (min-width:","){padding:0 ",";margin-top:",";}"],vt("breakpoints.0"),vt("space.2"),vt("space.3")),Pi=i.default.div`
  ${Hi}
  ${yt}
  ${Vt};
`,Di=({children:e,...t})=>n.default.createElement(Pi,Et({role:"tablist"},t),e);Di.displayName="SelectMenuTabs",Di.displayName="SelectMenu.Tabs";const Wi=i.default.details`
  ${"\n  // Remove marker added by the display: list-item browser default\n  > summary {\n    list-style: none;\n  }\n  // Remove marker added by details polyfill\n  > summary::before {\n    display: none;\n  }\n  // Remove marker added by Chrome\n  > summary::-webkit-details-marker {\n    display: none;\n  }\n"}
  ${yt}
  ${Vt};
`,Fi=n.default.forwardRef((({children:e,initialTab:r="",as:o,...a},i)=>{const d=t.useRef(null),l=null!=i?i:d,[c,s]=t.useState(r),[f,b]=t.useState(!1),g={selectedTab:c,setSelectedTab:s,setOpen:b,open:f,initialTab:r},u=t.useCallback((e=>{"current"in l&&l.current&&!l.current.contains(e.target)&&(e.defaultPrevented||b(!1))}),[l,b]);return t.useEffect((()=>{if(f)return document.addEventListener("click",u),()=>{document.removeEventListener("click",u)}}),[f,u]),ti(l,f,b),n.default.createElement(ri.Provider,{value:g},n.default.createElement(Wi,Et({ref:l},a,{open:f,onToggle:function(e){b(e.target.open)}}),e))}));Fi.displayName="SelectMenu";var ji=Object.assign(Fi,{MenuContext:ri,List:Bi,Divider:ai,Filter:bi,Footer:ui,Item:yi,Modal:Ai,Tabs:Di,Tab:Li,TabPanel:Oi,Header:hi,LoadingAnimation:Si});function Ui({variant:e,className:t,bordered:r,children:o,...a}){const i=Ia(t,`variant-${"lightweight"===e?"lightweight":"normal"}`);return r||(a={...a,borderWidth:0}),n.default.createElement(qt,Et({as:"nav",className:i},a),o)}Ui.displayName="SideNavBase";const _i=i.default(Ui)`
  background-color: ${vt("colors.bg.secondary")};

  ${e=>e.bordered&&r.css(["& > &{border-left:0;border-right:0;border-bottom:0;}"])}

  ${yt};
  ${Vt};
`,Gi=r.css(["background-color:",";outline:none;text-decoration:none;"],vt("colors.state.hover.secondaryBg")),Ki=r.css(["color:",";text-decoration:none;outline:none;"],vt("colors.text.primary")),Vi=i.default(Dn).attrs((e=>"string"==typeof e.to||e.selected?{"aria-current":"page"}:{}))`
  position: relative;
  display: block;
  ${e=>"full"===e.variant&&r.css(["display:flex;align-items:center;justify-content:space-between;"])}
  width: 100%;
  text-align: left;
  font-size: ${vt("fontSizes.1")};

  & > ${_i} {
    border-bottom: none;
  }

  ${_i}.variant-normal > & {
    color: ${vt("colors.text.primary")};
    padding: ${vt("space.3")};
    border: 0;
    border-top: ${vt("borderWidths.1")} solid ${vt("colors.border.secondary")};

    &:first-child {
      border-top: 0;
      border-top-right-radius: ${vt("radii.2")};
      border-top-left-radius: ${vt("radii.2")};
    }

    &:last-child {
      border-bottom-right-radius: ${vt("radii.2")};
      border-bottom-left-radius: ${vt("radii.2")};
    }

    // Bar on the left
    &::before {
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      z-index: 1;
      width: 3px;
      pointer-events: none;
      content: '';
    }

    &:hover {
      ${Gi}
    }

    &:focus {
      ${Gi}
      box-shadow: ${vt("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      background-color: ${vt("colors.sidenav.selectedBg")};

      // Bar on the left
      &::before {
        background-color: ${vt("colors.sidenav.borderActive")};
      }
    }
  }

  ${_i}.variant-lightweight > & {
    padding: ${vt("space.1")} 0;
    color: ${vt("colors.text.link")};

    &:hover {
      ${Ki}
    }

    &:focus {
      ${Ki}
      box-shadow: ${vt("shadows.state.focus.shadow")};
      z-index: 1;
    }

    &[aria-current='page'],
    &[aria-selected='true'] {
      color: ${vt("colors.text.primary")};
      font-weight: ${vt("fontWeights.semibold")};
    }
  }

  ${Vt};
`;_i.defaultProps={variant:"normal"},Vi.defaultProps={variant:"normal"},Vi.displayName="SideNav.Link";var Xi=Object.assign(_i,{Link:Vi});const qi={small:"16px",medium:"32px",large:"64px"};function Yi({size:e,...t}){var r;const o=null!==(r=e&&qi[e])&&void 0!==r?r:qi.medium;return n.default.createElement("svg",Et({height:o,width:o,viewBox:"0 0 16 16",fill:"none"},t),n.default.createElement("circle",{cx:"8",cy:"8",r:"7",stroke:"currentColor",strokeOpacity:"0.25",strokeWidth:"2",vectorEffect:"non-scaling-stroke"}),n.default.createElement("path",{d:"M15 8a7.002 7.002 0 00-7-7",stroke:"currentColor",strokeWidth:"2",strokeLinecap:"round",vectorEffect:"non-scaling-stroke"}))}Yi.displayName="Spinner";const Ji=i.default(Yi)`
  @keyframes rotate-keyframes {
    100% {
      transform: rotate(360deg);
    }
  }

  animation: rotate-keyframes 1s linear infinite;

  ${yt}
  ${Vt}
`;Ji.displayName="Spinner";const Zi={issueOpened:Kr,pullOpened:_r,issueClosed:Gr,pullClosed:_r,pullMerged:Ur,draft:_r},Qi=re({prop:"status",variants:{issueClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullClosed:{backgroundColor:"prState.closed.bg",color:"prState.closed.text",borderColor:"prState.closed.border"},pullMerged:{backgroundColor:"prState.merged.bg",color:"prState.merged.text",borderColor:"prState.merged.border"},issueOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},pullOpened:{backgroundColor:"prState.open.bg",color:"prState.open.text",borderColor:"prState.open.border"},draft:{backgroundColor:"prState.draft.bg",color:"prState.draft.text",borderColor:"prState.draft.border"}}}),ed=re({prop:"variant",variants:{small:{paddingX:2,paddingY:1,fontSize:0},normal:{paddingX:"12px",paddingY:2,fontSize:1}}}),td=i.default.span`
  display: inline-flex;
  align-items: center;
  font-weight: ${vt("fontWeights.bold")};
  line-height: 16px;
  color: ${vt("colors.bg.primary")};
  text-align: center;
  border-radius: ${vt("radii.3")};
  border-width: 1px;
  border-style: solid;
  ${Qi};
  ${ed};
  ${yt};
  ${Vt};
`;function rd({children:e,status:t,variant:r,...o}){const a="small"===r?{width:"1em"}:{};return n.default.createElement(td,Et({},o,{variant:r,status:t}),t&&n.default.createElement(po,Et({mr:1},a,{icon:Zi[t]||Xr})),e)}rd.displayName="StateLabel",rd.defaultProps={variant:"normal"};const od=i.default.nav`
  display: flex;
  justify-content: space-between;

  .SubNav-body {
    display: flex;
    margin-bottom: -1px;

    > * {
      margin-left: ${vt("space.2")};
    }

    > *:first-child {
      margin-left: 0;
    }
  }

  .SubNav-actions {
    align-self: center;
  }

  ${yt};
  ${St};
  ${Vt};
`;function ad({actions:e,className:t,children:r,label:o,...a}){const i=Ia(t,"SubNav");return n.default.createElement(od,Et({className:i,"aria-label":o},a),n.default.createElement("div",{className:"SubNav-body"},r),e&&n.default.createElement("div",{className:"SubNav-actions"},e))}function nd(e){return n.default.createElement(Yt,e)}ad.displayName="SubNav",nd.displayName="SubNavLinks";const id=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Ia("SubNav-item",e.selected&&"selected",e.className)})))`
  padding-left: ${vt("space.3")};
  padding-right: ${vt("space.3")};
  font-weight: ${vt("fontWeights.semibold")};
  font-size: ${vt("fontSizes.1")};
  line-height: 20px; //custom value for SubNav
  min-height: 34px; //custom value for SubNav
  color: ${vt("colors.text.primary")};
  text-align: center;
  text-decoration: none;
  border-top: 1px solid ${vt("colors.border.primary")};
  border-bottom: 1px solid ${vt("colors.border.primary")};
  border-right: 1px solid ${vt("colors.border.primary")};
  display: flex;
  align-items: center;

  &:first-of-type {
    border-top-left-radius: ${vt("radii.2")};
    border-bottom-left-radius: ${vt("radii.2")};
    border-left: 1px solid ${vt("colors.border.primary")};
  }

  &:last-of-type {
    border-top-right-radius: ${vt("radii.2")};
    border-bottom-right-radius: ${vt("radii.2")};
  }

  &:hover,
  &:focus {
    text-decoration: none;
    background-color: ${vt("colors.bg.tertiary")};
    transition: 0.2s ease;

    .SubNav-octicon {
      color: ${vt("colors.icon.secondary")};
    }
  }

  &.selected {
    color: ${vt("colors.state.selected.primaryText")};
    background-color: ${vt("colors.state.selected.primaryBg")};
    border-color: ${vt("colors.state.selected.primaryBorder")};
    .SubNav-octicon {
      color: ${vt("colors.state.selected.primaryText")};
    }
  }

  ${yt};
  ${Vt};
`;id.displayName="SubNav.Link",nd.displayName="SubNav.Links";var dd=Object.assign(ad,{Link:id,Links:nd});const ld=i.default.div`
  margin-top: 0;
  border-bottom: 1px solid ${vt("colors.border.primary")};
  ${yt}
  ${Vt}
`,cd=i.default.nav`
  display: flex;
  margin-bottom: -1px;
  overflow: auto;
`;function sd({children:e,"aria-label":t,...r}){return n.default.createElement(ld,r,n.default.createElement(cd,{"aria-label":t},e))}sd.displayName="TabNav";const fd=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Ia("TabNav-item",e.selected&&"selected",e.className)})))`
  padding: 8px 12px;
  font-size: ${vt("fontSizes.1")};
  line-height: 20px;
  color: ${vt("colors.text.primary")};
  text-decoration: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-bottom: 0;

  &:hover,
  &:focus {
    color: ${vt("colors.text.primary")};
    text-decoration: none;
  }

  &.selected {
    color: ${vt("colors.text.primary")};
    border-color: ${vt("colors.border.primary")};
    border-top-right-radius: ${vt("radii.2")};
    border-top-left-radius: ${vt("radii.2")};
    background-color: ${vt("colors.bg.canvas")};
  }

  ${yt};
  ${Vt};
`;fd.displayName="TabNav.Link";var bd=Object.assign(sd,{Link:fd});const gd=i.default(Yt).withConfig({displayName:"Timeline",componentId:"c4dq2e-0"})(["flex-direction:column;"," ",";"],(e=>e.clipSidebar&&r.css([".Timeline-Item:first-child{padding-top:0;}.Timeline-Item:last-child{padding-bottom:0;}"])),Vt),ud=i.default(Yt).attrs((e=>({className:Ia("Timeline-Item",e.className)}))).withConfig({displayName:"Timeline__TimelineItem",componentId:"c4dq2e-1"})(["position:relative;padding:"," 0;margin-left:",";&::before{position:absolute;top:0;bottom:0;left:0;display:block;width:2px;content:'';background-color:",";}"," ",";",";"],vt("space.3"),vt("space.3"),vt("colors.border.secondary"),(e=>e.condensed&&r.css(["padding-top:",";padding-bottom:0;&:last-child{padding-bottom:",";}.TimelineItem-Badge{height:16px;margin-top:",";margin-bottom:",";color:",";background-color:",";border:0;}"],vt("space.1"),vt("space.3"),vt("space.2"),vt("space.2"),vt("colors.icon.tertiary"),vt("colors.bg.canvas"))),yt,Vt);var pd=i.default(Yt).withConfig({displayName:"Timeline___StyledFlex",componentId:"c4dq2e-2"})(["border-radius:50%;border:2px solid ",";"],(e=>e._css));const md=e=>n.default.createElement(tr,{zIndex:1},n.default.createElement(pd,Et({className:Ia(e.className,"TimelineItem-Badge"),flexShrink:0,overflow:"hidden",color:"icon.secondary",bg:"timeline.badgeBg",width:"32px",height:"32px",mr:2,ml:"-15px",alignItems:"center",justifyContent:"center"},e,{_css:vt("colors.bg.canvas")}),e.children));md.displayName="TimelineBadge";const hd=i.default(Xt).withConfig({displayName:"Timeline__TimelineBody",componentId:"c4dq2e-3"})(["min-width:0;max-width:100%;margin-top:",";color:",";flex:auto;font-size:",";",";"],vt("space.1"),vt("colors.timeline.text"),vt("fontSizes.1"),Vt),xd=i.default(tr).withConfig({displayName:"Timeline__TimelineBreak",componentId:"c4dq2e-4"})(["z-index:1;height:24px;margin:0;margin-bottom:-",";margin-left:0;background-color:",";border:0;border-top:"," solid ",";",";"],vt("space.3"),vt("colors.bg.canvas"),vt("space.1"),vt("colors.border.primary"),Vt);ud.displayName="Timeline.Item",md.displayName="Timeline.Badge",hd.displayName="Timeline.Body",xd.displayName="Timeline.Break";var vd=Object.assign(gd,{Item:ud,Badge:md,Body:hd,Break:xd});const yd=i.default.span`
  position: relative;

  &::before {
    position: absolute;
    z-index: 1000001;
    display: none;
    width: 0px;
    height: 0px;
    color: ${vt("colors.tooltip.bg")};
    pointer-events: none;
    content: '';
    border: 6px solid transparent;
    opacity: 0;
  }

  &::after {
    position: absolute;
    z-index: 1000000;
    display: none;
    padding: 0.5em 0.75em;
    font: normal normal 11px/1.5 ${vt("fonts.normal")};
    -webkit-font-smoothing: subpixel-antialiased;
    color: ${vt("colors.tooltip.text")};
    text-align: center;
    text-decoration: none;
    text-shadow: none;
    text-transform: none;
    letter-spacing: normal;
    word-wrap: break-word;
    white-space: pre;
    pointer-events: none;
    content: attr(aria-label);
    background: ${vt("colors.tooltip.bg")};
    border-radius: ${vt("radii.1")};
    opacity: 0;
  }

  // delay animation for tooltip
  @keyframes tooltip-appear {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }

  &:hover,
  &:active,
  &:focus {
    &::before,
    &::after {
      display: inline-block;
      text-decoration: none;
      animation-name: tooltip-appear;
      animation-duration: 0.1s;
      animation-fill-mode: forwards;
      animation-timing-function: ease-in;
      animation-delay: 0.4s;
    }
  }

  &.tooltipped-no-delay:hover,
  &.tooltipped-no-delay:active,
  &.tooltipped-no-delay:focus {
    &::before,
    &::after {
      animation-delay: 0s;
    }
  }

  &.tooltipped-multiline:hover,
  &.tooltipped-multiline:active,
  &.tooltipped-multiline:focus {
    &::after {
      display: table-cell;
    }
  }

  // Tooltipped south
  &.tooltipped-s,
  &.tooltipped-se,
  &.tooltipped-sw {
    &::after {
      top: 100%;
      right: 50%;
      margin-top: 6px;
    }

    &::before {
      top: auto;
      right: 50%;
      bottom: -7px;
      margin-right: -6px;
      border-bottom-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-se {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${vt("space.3")};
    }
  }

  &.tooltipped-sw::after {
    margin-right: -${vt("space.3")};
  }

  // Tooltips above the object
  &.tooltipped-n,
  &.tooltipped-ne,
  &.tooltipped-nw {
    &::after {
      right: 50%;
      bottom: 100%;
      margin-bottom: 6px;
    }

    &::before {
      top: -7px;
      right: 50%;
      bottom: auto;
      margin-right: -6px;
      border-top-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-ne {
    &::after {
      right: auto;
      left: 50%;
      margin-left: -${vt("space.3")};
    }
  }

  &.tooltipped-nw::after {
    margin-right: -${vt("space.3")};
  }

  // Move the tooltip body to the center of the object.
  &.tooltipped-s::after,
  &.tooltipped-n::after {
    transform: translateX(50%);
  }

  // Tooltipped to the left
  &.tooltipped-w {
    &::after {
      right: 100%;
      bottom: 50%;
      margin-right: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      bottom: 50%;
      left: -7px;
      margin-top: -6px;
      border-left-color: ${vt("colors.tooltip.bg")};
    }
  }

  // tooltipped to the right
  &.tooltipped-e {
    &::after {
      bottom: 50%;
      left: 100%;
      margin-left: 6px;
      transform: translateY(50%);
    }

    &::before {
      top: 50%;
      right: -7px;
      bottom: 50%;
      margin-top: -6px;
      border-right-color: ${vt("colors.tooltip.bg")};
    }
  }

  &.tooltipped-multiline {
    &::after {
      width: max-content;
      max-width: 250px;
      word-wrap: break-word;
      white-space: pre-line;
      border-collapse: separate;
    }

    &.tooltipped-s::after,
    &.tooltipped-n::after {
      right: auto;
      left: 50%;
      transform: translateX(-50%);
    }

    &.tooltipped-w::after,
    &.tooltipped-e::after {
      right: 100%;
    }
  }

  &.tooltipped-align-right-2::after {
    right: 0;
    margin-right: 0;
  }

  &.tooltipped-align-right-2::before {
    right: 15px;
  }

  &.tooltipped-align-left-2::after {
    left: 0;
    margin-left: 0;
  }

  &.tooltipped-align-left-2::before {
    left: 10px;
  }
  ${yt};
  ${Vt};
`;function wd({direction:e="n",children:t,className:r,text:o,noDelay:a,align:i,wrap:d,...l}){const c=Ia(r,`tooltipped-${e}`,i&&`tooltipped-align-${i}-2`,a&&"tooltipped-no-delay",d&&"tooltipped-multiline");return n.default.createElement(yd,Et({role:"tooltip","aria-label":o},l,{className:c}),t)}wd.displayName="Tooltip",wd.alignments=["left","right"],wd.directions=["n","ne","e","se","s","sw","w","nw"];const Bd=i.default.nav`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid ${vt("colors.border.secondary")};
  &.UnderlineNav--right {
    justify-content: flex-end;

    .UnderlineNav-item {
      margin-right: 0;
      margin-left: ${vt("space.3")};
    }

    .UnderlineNav-actions {
      flex: 1 1 auto;
    }
  }
  &.UnderlineNav--full {
    display: block;
  }

  .UnderlineNav-body {
    display: flex;
    margin-bottom: -1px;
  }

  .UnderlineNav-actions {
    align-self: center;
  }

  ${yt};
  ${Vt};
`;function $d({actions:e,className:t,align:r,children:o,full:a,label:i,theme:d,...l}){const c=Ia(t,"UnderlineNav",r&&`UnderlineNav--${r}`,a&&"UnderlineNav--full");return n.default.createElement(Bd,Et({className:c,"aria-label":i,theme:d},l),n.default.createElement("div",{className:"UnderlineNav-body"},o),e&&n.default.createElement("div",{className:"UnderlineNav-actions"},e))}$d.displayName="UnderlineNav";const kd=i.default.a.attrs((e=>({activeClassName:"string"==typeof e.to?"selected":"",className:Ia("UnderlineNav-item",e.selected&&"selected",e.className)})))`
  padding: ${vt("space.3")} ${vt("space.2")};
  margin-right: ${vt("space.3")};
  font-size: ${vt("fontSizes.1")};
  line-height: ${vt("lineHeights.default")};
  color: ${vt("colors.underlinenav.text")};
  text-align: center;
  border-bottom: 2px solid transparent;
  text-decoration: none;

  &:hover,
  &:focus {
    color: ${vt("colors.underlinenav.textHover")};
    text-decoration: none;
    border-bottom-color: ${vt("colors.border.tertiary")};
    transition: 0.2s ease;

    .UnderlineNav-octicon {
      color: ${vt("colors.text.tertiary")};
    }
  }

  &.selected {
    color: ${vt("colors.underlinenav.textActive")};
    border-bottom-color: ${vt("colors.underlinenav.borderActive")};

    .UnderlineNav-octicon {
      color: ${vt("colors.underlinenav.iconActive")};
    }
  }

  ${yt};
  ${Vt};
`;kd.displayName="UnderlineNav.Link";var Sd=Object.assign($d,{Link:kd});e.Absolute=Qt,e.ActionList=ha,e.ActionMenu=Sa,e.Avatar=Ta,e.AvatarPair=Ca,e.AvatarStack=Aa,e.BaseStyles=Nt,e.BorderBox=qt,e.Box=Xt,e.BranchName=za,e.Breadcrumb=Ha,e.Button=zr,e.ButtonClose=Zr,e.ButtonDanger=Rr,e.ButtonGroup=Lr,e.ButtonInvisible=Hr,e.ButtonOutline=Mr,e.ButtonPrimary=Or,e.ButtonTableList=Pr,e.Caret=_a,e.CircleBadge=Xa,e.CircleOcticon=qa,e.ConfirmationDialog=Xo,e.CounterLabel=Ya,e.Details=Ja,e.Dialog=ln,e.Dropdown=pn,e.DropdownButton=mn,e.DropdownMenu=hn,e.FilterList=$n,e.FilteredSearch=xn,e.Fixed=er,e.Flash=Sn,e.Flex=Yt,e.FormGroup=Cn,e.Grid=Jt,e.Header=zn,e.Heading=Rn,e.Label=Hn,e.LabelGroup=Ln,e.Link=Dn,e.Overlay=wa,e.Pagehead=Wn,e.Pagination=_n,e.PointerBox=Vn,e.Popover=Yn,e.Position=Zt,e.ProgressBar=ei,e.Relative=tr,e.SelectMenu=ji,e.SideNav=Xi,e.Spinner=Ji,e.StateLabel=rd,e.Sticky=rr,e.StyledOcticon=po,e.SubNav=dd,e.TabNav=bd,e.Text=en,e.TextInput=si,e.ThemeProvider=Ut,e.Timeline=vd,e.Tooltip=wd,e.Truncate=Qo,e.UnderlineNav=Sd,e.theme=l.theme,e.themeGet=vt,e.useColorSchemeVar=Gt,e.useConfirm=function(){const{theme:e,colorMode:r,dayScheme:o,nightScheme:a}=_t();return t.useCallback((t=>async function(e,t){const{content:r,...o}=t;return new Promise((t=>{const a=document.createElement("div");d.default.render(n.default.createElement(Ut,e,n.default.createElement(Xo,Et({},o,{onClose:e=>{d.default.unmountComponentAtNode(a),t("confirm"===e)}}),r)),a)}))}({theme:e,colorMode:r,dayScheme:o,nightScheme:a},t)),[e,r,o,a])},e.useDetails=or,e.useOnEscapePress=fr,e.useOnOutsideClick=ar,e.useOpenAndCloseFocus=lr,e.useOverlay=gr,e.useSafeTimeout=function(){const e=t.useRef(new Set),r=t.useCallback(((t,r,...o)=>{const a=window.setTimeout(t,r,...o);return e.current.add(a),a}),[]),o=t.useCallback((t=>{clearTimeout(t),e.current.delete(t)}),[]);return t.useEffect((()=>()=>{for(const t of e.current)clearTimeout(t)}),[]),{safeSetTimeout:r,safeClearTimeout:o}},e.useTheme=_t,Object.defineProperty(e,"__esModule",{value:!0})}));
